--- vpxtable_scripts/4roses (Williams 1962)v1.03.vbs
+++ todo_extracted_tables/4roses (Williams 1962)v1.03.vbs
@@ -174,8 +174,8 @@
     If Tilt=0 then
       If GameInProgress=1 then
         LeftFlipper.RotateToEnd
-        PlaySoundAtVol SoundFXDOF("FlipperUp",101,DOFOn,DOFContactors), LeftFlipper, 1
-        PlayLoopSoundAtVol "buzzl",LeftFlipper, 1
+        PlaySound SoundFXDOF("FlipperUp",101,DOFOn,DOFFlippers)
+        playsound "buzzl",-1
       End If
     End If
   End If
@@ -183,8 +183,8 @@
     If Tilt=0 then
       If GameinProgress=1 then
         RightFlipper.RotateToEnd
-        PlaySoundAtVol SoundFXDOF("FlipperUp",102,DOFOn,DOFContactors), RightFlipper, 1
-        PlayLoopSoundAtVol"buzz", RightFlipper, 1
+        PlaySound SoundFXDOF("FlipperUp",102,DOFOn,DOFFlippers)
+        playsound"buzz",-1
       End If
     End If
   End If
@@ -200,9 +200,6 @@
     Nudge 0, 2
     TiltSensor=TiltSensor+50
   End If
-  If keycode = MechanicalTilt Then
-    TiltSensor=TiltSensor+50
-  End If
 
   If Keycode = StartGameKey And Not HSEnterMode=true then
     If GameinProgress=0 and credits>0 then
@@ -244,13 +241,13 @@
 Sub Table1_KeyUp(ByVal keycode)
   If keycode = PlungerKey Then
     Plunger.Fire
-    PlaySoundAtVol "Plunger", Plunger, 1
+    PlaySound "Plunger"
   End If
   If keycode = LeftFlipperKey Then
     If Tilt=0 then
       If GameInProgress=1 then
         LeftFlipper.RotateToStart
-        PlaySoundAtVol SoundFXDOF("FlipperDown",101,DOFOff,DOFContactors), LeftFlipper, 1
+        PlaySound SoundFXDOF("FlipperDown",101,DOFOff,DOFFlippers)
         stopsound"buzzl"
       End If
     End If
@@ -259,7 +256,7 @@
     If GameinProgress=1 then
       If Tilt=0 then
         RightFlipper.RotateToStart
-        PlaySoundAtVol SoundFXDOF("FlipperDown",102,DOFOff,DOFContactors), RightFlipper, 1
+        PlaySound SoundFXDOF("FlipperDown",102,DOFOff,DOFFlippers)
         stopsound "buzz"
       End If
     End If
@@ -537,7 +534,7 @@
 End Sub
 
 sub trigger13_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
     If lbumper2.state=0 And lbumper3.state=0 Then
     DOF 214, DOFPulse
 
@@ -557,7 +554,7 @@
 
 
 sub trigger3_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light409.state=1 then
      DOF 210, DOFPulse
     addscore 10
@@ -567,7 +564,7 @@
   end if
 end sub
 sub trigger4_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light410.state=1 then
      DOF 210, DOFPulse
     addscore 10
@@ -577,7 +574,7 @@
   end if
 end sub
 sub trigger5_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light411.state=1 then
      DOF 210, DOFPulse
     addscore 10
@@ -587,7 +584,7 @@
   end if
 end sub
 sub trigger6_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light412.state=1 then
      DOF 210, DOFPulse
     addscore 10
@@ -597,7 +594,7 @@
   end if
 end sub
 sub trigger7_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light413.state=1 then
      DOF 210, DOFPulse
     addscore 10
@@ -609,7 +606,7 @@
   end if
 end sub
 sub trigger8_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light414.state=1 then
      DOF 211, DOFPulse
     addscore 10
@@ -619,7 +616,7 @@
   end if
 end sub
 sub trigger9_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light415.state=1 then
      DOF 211, DOFPulse
     addscore 10
@@ -629,7 +626,7 @@
   end if
 end sub
 sub trigger10_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light416.state=1 then
      DOF 211, DOFPulse
     addscore 10
@@ -639,7 +636,7 @@
   end if
 end sub
 sub trigger11_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light417.state=1 then
      DOF 211, DOFPulse
     addscore 10
@@ -649,7 +646,7 @@
   end if
 end sub
 sub trigger12_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
   if light418.state=1 then
      DOF 211, DOFPulse
     addscore 10
@@ -659,13 +656,13 @@
   end if
 end sub
 sub trigger15_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
     DOF 212, DOFPulse
   addscore 10
   yellowslights
 end sub
 sub trigger16_hit()
-  PlaysoundAtVol "fx_sensor", ActiveBall, 1
+  Playsound "fx_sensor"
     DOF 213, DOFPulse
   addscore 10
   redslights
@@ -1056,12 +1053,12 @@
 
 
 Sub Gate_hit
-  PlaysoundAtVol "gate", ActiveBall, 1
+  Playsound "gate"
 End Sub
 
 
 Sub RightSlingShot_Slingshot
-    playsoundAtVol SoundFXDOF("right_slingshot",104,DOFPulse,DOFContactors),sling1, 1
+    playsound SoundFXDOF("right_slingshot",104,DOFPulse,DOFContactors),0,1,-0.05,0.05, 0, 1, 0.05, 0.05
   DOF 204,DOFPulse
     RSling0.Visible = 0
     RSling1.Visible = 1
@@ -1082,7 +1079,7 @@
 End Sub
 
 Sub LeftSlingShot_Slingshot
-    playsoundAtVol SoundFXDOF("left_slingshot",103,DOFPulse,DOFContactors), sling2, 1
+    playsound SoundFXDOF("left_slingshot",103,DOFPulse,DOFContactors),0,1,-0.05,0.05
   DOF 203,DOFPulse
     LSling0.Visible = 0
     LSling1.Visible = 1
@@ -1170,109 +1167,17 @@
   If B2SOn Then Controller.Stop
 End Sub
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
-
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
+' *********************************************************************
 '                      Supporting Ball & Sound Functions
-'*********************************************************************
-
-Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / table1.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
+' *********************************************************************
 
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / table1.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
+Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
+    Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
     Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / table1.width-1
+    tmp = ball.x * 2 / Table1.width-1
     If tmp > 0 Then
         Pan = Csng(tmp ^10)
     Else
@@ -1280,23 +1185,6 @@
     End If
 End Function
 
-Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
-End Function
-
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-End Function
-
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
-End Function
-
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
-End Function
-
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
     Pitch = BallVel(ball) * 20
 End Function
@@ -1305,31 +1193,6 @@
     BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
 End Function
 
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
-End Function
-
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
-End Function
-
-
 '*****************************************
 '      JP's VP10 Rolling Sounds
 '*****************************************
@@ -1360,13 +1223,9 @@
 
   ' play the rolling sound for each ball
     For b = 0 to UBound(BOT)
-      If BallVel(BOT(b) ) > 1 Then
+        If BallVel(BOT(b) ) > 1 AND BOT(b).z < 30 Then
             rolling(b) = True
-        if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-        Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-        End If
+            PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*50, Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0
         Else
             If rolling(b) = True Then
                 StopSound("fx_ballrolling" & b)
@@ -1381,23 +1240,23 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, Pan(ball1), 0, Pitch(ball1), 0, 0
 End Sub
 
 Sub Rubbers_Hit(idx)
   Select Case Int(Rnd*2)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 1 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 1 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
 Sub Slings1_slingshot(idx)
-  PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   Addscore(1)
 End Sub
 Sub Slings10_slingshot(idx)
-  PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   Addscore(10)
 End Sub
 
@@ -1411,9 +1270,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*400, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
@@ -1421,15 +1280,15 @@
 ' Diverse Collection Hit Sounds
 '******************************
 
-Sub aMetals_Hit(idx):PlaySound "fx_MetalHit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aRubber_Bands_Hit(idx):PlaySound "fx_rubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aRubber_Posts_Hit(idx):PlaySound "fx_postrubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aRubber_Pins_Hit(idx):PlaySound "fx_postrubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aYellowPins_Hit(idx):PlaySound "fx_postrubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aPlastics_Hit(idx):PlaySound "fx_PlasticHit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aGates_Hit(idx):PlaySound "fx_Gate", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aWoods_Hit(idx):PlaySound "fx_Woodhit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aCaptiveWalls_Hit(idx):PlaySound "fx_collide", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aMetals_Hit(idx):PlaySound "fx_MetalHit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aRubber_Bands_Hit(idx):PlaySound "fx_rubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aRubber_Posts_Hit(idx):PlaySound "fx_postrubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aRubber_Pins_Hit(idx):PlaySound "fx_postrubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aYellowPins_Hit(idx):PlaySound "fx_postrubber", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aPlastics_Hit(idx):PlaySound "fx_PlasticHit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aGates_Hit(idx):PlaySound "fx_Gate", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aWoods_Hit(idx):PlaySound "fx_Woodhit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
+Sub aCaptiveWalls_Hit(idx):PlaySound "fx_collide", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0:End Sub
 
 
 
