--- vpxtable_scripts/Airborne Avenger (Atari 1977) 1.4.vbs
+++ todo_extracted_tables/Airborne Avenger (Atari 1977) 1.4.vbs
@@ -33,8 +33,7 @@
 Const VolFlip   = 1    ' Flipper volume.
 
 
-Dim BallSound, Decal, AltColor, PlayNums, GatePos
-
+Dim BallSound, Decal, AltColor, PlayNums
 'Dim F1Xon, f1xoff, f1on, f1off, F2Xon, f2xoff, f2on, f2off, F3Xon, f3xoff, f3on, f3off
 
 '***************************REMOVE DECAL IN CABINET MODE*****************
@@ -56,21 +55,14 @@
 '************************************************************************
 
 
-' Thalamus - this sub was called twice - I'm disabling this one, since that
-' is what is going to happen anyway.
+Sub Table1_init()
 
-' Sub Table1_init()
-'
-' GatePos=0
-'
-' FlashTimer
-'   BallSound=1
-'   FlashSync.Enabled=True
-' BumpTimer.Enabled=False
-' End Sub
 
-' Thalamus : Was missing 'vpminit me'
-  vpminit me
+FlashTimer
+  BallSound=1
+  FlashSync.Enabled=True
+BumpTimer.Enabled=False
+End Sub
 
 Const cGameName     = "aavenger"   ' PinMAME short name
 Const cCredits      = "VPX table by DarthMarino"
@@ -125,23 +117,15 @@
 
 'Smoothed Solenoid routines for gates
 
-' Sub SolGate(Enabled)
-'   If Not LeftDelay.Enabled Then vpmSolDiverter Gate,True,True:PlaySound "savegate"
-'   LeftDelay.Enabled=0
-'   LeftDelay.Enabled=1
-' End Sub
-'
-' Sub LeftDelay_Timer:LeftDelay.Enabled=0:vpmSolDiverter Gate,True,False:PlaySound "savegate":End Sub
-' 'End of solenoid smoothing for gates
-
 Sub SolGate(Enabled)
-  If GatePos=0 then
-    vpmSolDiverter Gate,True,True:GatePos=1:playsound "savegate"
-  Else
-    vpmSolDiverter Gate,True,False:GatePos=0:playsound "savegate"
-  End If
+  If Not LeftDelay.Enabled Then vpmSolDiverter Gate,True,True:PlaySound "savegate"
+  LeftDelay.Enabled=0
+  LeftDelay.Enabled=1
 End Sub
 
+Sub LeftDelay_Timer:LeftDelay.Enabled=0:vpmSolDiverter Gate,True,False:PlaySound "savegate":End Sub
+'End of solenoid smoothing for gates
+
 Dim bsTrough,bsSaucer1,bsSaucer2,bsEject,bsREject,cbCaptive, wallsoundrandom
 
 Sub Table1_Init
@@ -175,7 +159,6 @@
     bsTrough.InitKick BallRelease,90,5
     bsTrough.InitExitSnd SoundFX("Ballrel",DOFContactors),SoundFX("solon",DOFContactors)
     bsTrough.Balls=1
-    GatePos=0
 
   Set bsSaucer1=New cvpmBallStack
     bsSaucer1.InitSaucer lmkicker,41,65,10
@@ -646,28 +629,26 @@
 char(103) = chr(&HCE) ' 9 without bottom line
 char(231) = chr(&HCF) ' 9, without bottom line
 
-' Thalamus : This sub is used twice - this means ... this one IS NOT USED
-
-' Sub DisplayTimer_Timer
-'   Dim ChgLED,ii,num,chg,stat,obj
-' ' ChgLED = Controller.ChangedLEDs(&HFF, &HFFFF) 'hex of binary (display 111111, or first 6 digits)
-'   ChgLED = Controller.ChangedLEDs(&H0000003f, &Hffffffff)
-'   If Not IsEmpty(ChgLED) Then
-'     For ii = 0 To UBound(chgLED)
-'       num = chgLED(ii, 0) : chg = chgLED(ii, 1) : stat = chgLED(ii, 2)
-'       if (num < 39) then
-' '       MsgBox "Status Code is:" & vbNewLine & stat
-'         if char(stat) > "" then msg(num) = char(stat)
-'       end if
-'     next
-'   end if
-'   ScoreText1.Text = msg(0) & msg(1) & msg(2) & msg(3) & msg(4) & msg(5)
-'   ScoreText2.Text = msg(6) & msg(7) & msg(8) & msg(9) & msg(10) & msg(11)
-'   ScoreText3.Text = msg(12) & msg(13) & msg(14) & msg(15) & msg(16)& msg(17)
-'   ScoreText4.Text = msg(18) & msg(19) & msg(20) & msg(21) & msg(22) & msg(23)
-'   BallText.Text = msg(24) & msg(25)
-'   CreditText.Text=msg(26) & msg(27)
-' End Sub
+Sub DisplayTimer_Timer
+  Dim ChgLED,ii,num,chg,stat,obj
+' ChgLED = Controller.ChangedLEDs(&HFF, &HFFFF) 'hex of binary (display 111111, or first 6 digits)
+  ChgLED = Controller.ChangedLEDs(&H0000003f, &Hffffffff)
+  If Not IsEmpty(ChgLED) Then
+    For ii = 0 To UBound(chgLED)
+      num = chgLED(ii, 0) : chg = chgLED(ii, 1) : stat = chgLED(ii, 2)
+      if (num < 39) then
+'       MsgBox "Status Code is:" & vbNewLine & stat
+        if char(stat) > "" then msg(num) = char(stat)
+      end if
+    next
+  end if
+  ScoreText1.Text = msg(0) & msg(1) & msg(2) & msg(3) & msg(4) & msg(5)
+  ScoreText2.Text = msg(6) & msg(7) & msg(8) & msg(9) & msg(10) & msg(11)
+  ScoreText3.Text = msg(12) & msg(13) & msg(14) & msg(15) & msg(16)& msg(17)
+  ScoreText4.Text = msg(18) & msg(19) & msg(20) & msg(21) & msg(22) & msg(23)
+  BallText.Text = msg(24) & msg(25)
+  CreditText.Text=msg(26) & msg(27)
+End Sub
 
 'Atari Airborne Avenger
 'added by Inkochnito
@@ -888,17 +869,17 @@
 
 Sub Rubbers_Hit(idx)
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*100, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*100, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*100, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*10, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*10, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*10, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
 Sub Metals_Hit(idx)
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "metal_hit1", 0, Vol(ActiveBall)*100, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "metal_hit2", 0, Vol(ActiveBall)*100, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "metal_hit3", 0, Vol(ActiveBall)*100, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "metal_hit1", 0, Vol(ActiveBall)*30, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "metal_hit2", 0, Vol(ActiveBall)*30, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "metal_hit3", 0, Vol(ActiveBall)*30, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
