--- vpxtable_scripts/Attack from Mars (Midway 1995) v600.vbs
+++ todo_extracted_tables/Attack from Mars (Midway 1995) v600.vbs
@@ -268,7 +268,6 @@
 
 Sub table1_KeyDown(ByVal Keycode)
     If keycode = PlungerKey Then Controller.Switch(11) = 1
-    If keycode = LockBarKey Then Controller.Switch(11) = 1
     If keycode = LeftTiltKey Then Nudge 90, 6:PlaySound SoundFX("fx_nudge", 0), 0, 1, -0.1, 0.25:aSaucerShake:a3BankShake2
     If keycode = RightTiltKey Then Nudge 270, 6:PlaySound SoundFX("fx_nudge", 0), 0, 1, 0.1, 0.25:aSaucerShake:a3BankShake2
     If keycode = CenterTiltKey Then Nudge 0, 7:PlaySound SoundFX("fx_nudge", 0), 0, 1, 0, 0.25:aSaucerShake:a3BankShake2
@@ -277,7 +276,6 @@
 
 Sub table1_KeyUp(ByVal Keycode)
     If keycode = PlungerKey Then Controller.Switch(11) = 0
-    If keycode = LockBarKey Then Controller.Switch(11) = 0
     If vpmKeyUp(keycode) Then Exit Sub
 End Sub
 
@@ -1095,10 +1093,3 @@
         Case 21:table1.ColorGradeImage = "LUT Warm 10"
     End Select
 End Sub
-
-' Thalamus : Exit in a clean and proper way
-Sub Table1_exit
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
