--- vpxtable_scripts/Back To The Future - The Pinball (Data East 1990) v1.07.vbs
+++ todo_extracted_tables/Back To The Future - The Pinball (Data East 1990) v1.07.vbs
@@ -4430,8 +4430,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 '                 Applies the coef to x and y velocities
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     if debugOn then TBPout.text = str
   End Sub
 
@@ -4441,8 +4440,7 @@
     RealCOR = BallSpeed(aBall) / (cor.ballvel(aBall.id)+0.0001)
     coef = desiredcor / realcor
     If abs(aball.velx) < 2 and aball.vely < 0 and aball.vely > -3.75 then
-' Thalamus - patched :       aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+      aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     End If
   End Sub
 
