--- vpxtable_scripts/Bad Girls (Gottlieb 1988) Vifico.vbs
+++ todo_extracted_tables/Bad Girls (Gottlieb 1988) Vifico.vbs
@@ -39,17 +39,17 @@
 
 Sub SolLFlipper(Enabled)
      If Enabled Then
-         PlaySoundAtVol SoundFX("fx_Flipperup",DOFContactors), Leftflipper, 1:LeftFlipper.RotateToEnd
+         PlaySound SoundFX("fx_Flipperup",DOFContactors):LeftFlipper.RotateToEnd
      Else
-         PlaySoundatVol SoundFX("fx_Flipperdown",DOFContactors), Leftflipper, 1:LeftFlipper.RotateToStart
+         PlaySound SoundFX("fx_Flipperdown",DOFContactors):LeftFlipper.RotateToStart
      End If
   End Sub
 
 Sub SolRFlipper(Enabled)
      If Enabled Then
-         PlaySoundAtVol SoundFX("fx_Flipperup",DOFContactors), RightFlipper, 1:RightFlipper.RotateToEnd:RightFlipper1.RotateToEnd
+         PlaySound SoundFX("fx_Flipperup",DOFContactors):RightFlipper.RotateToEnd:RightFlipper1.RotateToEnd
      Else
-         PlaySoundAtVol SoundFX("fx_Flipperdown",DOFContactors), RightFlipper, 1:RightFlipper.RotateToStart:RightFlipper1.RotateToStart
+         PlaySound SoundFX("fx_Flipperdown",DOFContactors):RightFlipper.RotateToStart:RightFlipper1.RotateToStart
      End If
 End Sub
 '**********************************************************************************************************
@@ -78,7 +78,7 @@
     SetLamp 152, Enabled  ' Flasher Drop Targets X2
   Else
     if Controller.switch (45) then
-      PlaySoundAtVol SoundFX("Popper",DOFContactors), TopVuk, 1
+      playsound SoundFX("Popper",DOFContactors)
       TopVUK.DestroyBall
       Set raiseball = TopVUK.CreateBall
       raiseballsw = True
@@ -86,7 +86,7 @@
       TopVUK.Enabled=TRUE
       Controller.switch (45) = False
     else
-      PlaySoundAtVol "Popper", TopVuk, 1
+      PlaySound "Popper"
     end if
   End If
  End Sub
@@ -123,7 +123,7 @@
     SetLamp 156,Enabled'Dome Flasher X2
   Else
     if Controller.switch (55) then
-      PlaySoundAtVol SoundFX("Popper",DOFContactors), MidVukTop, 1
+      playsound SoundFX("Popper",DOFContactors)
       MidVUKTop.DestroyBall
       Set dropball = MidVUKTop.CreateBall
       dropballsw = True
@@ -131,7 +131,7 @@
       MidVUKTop.Enabled=TRUE
       Controller.switch (55) = False
     else
-      PlaySoundAtVol "Popper", MidVukTop, 1
+      PlaySound "Popper"
     end if
   End If
  End Sub
@@ -292,14 +292,14 @@
 '**********************************************************************************************************
 
 Sub Table1_KeyDown(ByVal KeyCode)
-  If keycode = PlungerKey Then Plunger.Pullback:PlaySoundAtVol "plungerpull", Plunger, 1
+  If keycode = PlungerKey Then Plunger.Pullback:playsound"plungerpull"
   If keycode = RightFlipperkey Then Controller.Switch(15) = 1
 
   If KeyDownHandler(keycode) Then Exit Sub
 End Sub
 
 Sub Table1_KeyUp(ByVal KeyCode)
-  If keycode = PlungerKey Then Plunger.Fire:PlaySoundAtVol "plunger", Plunger, 1
+  If keycode = PlungerKey Then Plunger.Fire:PlaySound"plunger"
   If keycode = RightFlipperkey Then Controller.Switch(15) = 0
 
   If KeyUpHandler(keycode) Then Exit Sub
@@ -308,7 +308,7 @@
 '**********************************************************************************************************
 
  ' Drain hole and kickers
-Sub Drain_Hit:bsTrough.addball me : PlaySoundAtVol "drain", Drain, 1 : End Sub
+Sub Drain_Hit:bsTrough.addball me : playsound"drain" : End Sub
 
  '***********************************
  'Top Left Raising VUK
@@ -318,7 +318,7 @@
  Sub TopVUK_Hit()
   TopVUK.Enabled=FALSE
   Controller.switch (45) = True
-  PlaySoundAtVol "popper_ball", ActiveBall, 1
+  playsound "popper_ball"
  End Sub
 
  Sub TopVukraiseballtimer_Timer()
@@ -341,7 +341,7 @@
 Sub MidVUKTop_Hit()
   MidVUKTop.Enabled=FALSE
   Controller.switch (55) = True
-  PlaySoundAtVol "popper_ball", ActiveBall, 1
+  playsound "popper_ball"
 End Sub
 
  Sub MidVukdropballtimer_Timer()
@@ -350,7 +350,7 @@
     If dropball.z > 100 then
       MidVUKTop.Kick 45, 10
       Set dropball = Nothing
-      PlaySoundAtVol "fx_ballrampdrop", MidVukTop, 1
+      PlaySound "fx_ballrampdrop"
       MidVukdropballtimer.Enabled = False
       dropballsw = False
     End If
@@ -377,44 +377,44 @@
 Sub sw51_hit:vpmTimer.pulseSw 51 : End Sub
 
 'Wire Triggers
- Sub sw3_Hit:Controller.Switch(3)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw3_Hit:Controller.Switch(3)=1 : playsound"rollover" : End Sub
  Sub sw3_UnHit:Controller.Switch(3)=0:End Sub
- Sub sw13_Hit:Controller.Switch(13)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw13_Hit:Controller.Switch(13)=1 : playsound"rollover" : End Sub
  Sub sw13_UnHit:Controller.Switch(13)=0:End Sub
- Sub sw23_Hit:Controller.Switch(23)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw23_Hit:Controller.Switch(23)=1 : playsound"rollover" : End Sub
  Sub sw23_UnHit:Controller.Switch(23)=0:End Sub
- Sub sw33_Hit:Controller.Switch(33)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw33_Hit:Controller.Switch(33)=1 : playsound"rollover" : End Sub
  Sub sw33_UnHit:Controller.Switch(33)=0:End Sub
- Sub sw43_Hit:Controller.Switch(43)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw43_Hit:Controller.Switch(43)=1 : playsound"rollover" : End Sub
  Sub sw43_UnHit:Controller.Switch(43)=0:End Sub
- Sub sw53_Hit:Controller.Switch(53)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw53_Hit:Controller.Switch(53)=1 : playsound"rollover" : End Sub
  Sub sw53_UnHit:Controller.Switch(53)=0:End Sub
- Sub sw63_Hit:Controller.Switch(63)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw63_Hit:Controller.Switch(63)=1 : playsound"rollover" : End Sub
  Sub sw63_UnHit:Controller.Switch(63)=0:End Sub
- Sub sw73_Hit:Controller.Switch(73)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw73_Hit:Controller.Switch(73)=1 : playsound"rollover" : End Sub
  Sub sw73_UnHit:Controller.Switch(73)=0:End Sub
 
 'Star Rollover
- Sub sw72a_Hit:Controller.Switch(72)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw72a_Hit:Controller.Switch(72)=1 : playsound"rollover" : End Sub
  Sub sw72a_UnHit:Controller.Switch(72)=0:End Sub
- Sub sw72b_Hit:Controller.Switch(72)=1 : PlaySoundAtVol "rollover", ActiveBall, 1 : End Sub
+ Sub sw72b_Hit:Controller.Switch(72)=1 : playsound"rollover" : End Sub
  Sub sw72b_UnHit:Controller.Switch(72)=0:End Sub
 
 'Bumpers
-Sub Bumper1_Hit : vpmTimer.PulseSw(52) : PlaySoundAtVol SoundFX("fx_bumper1",DOFContactors), ActiveBall, 1: End Sub
-Sub Bumper2_Hit : vpmTimer.PulseSw(42) : PlaySoundAtVol SoundFX("fx_bumper1",DOFContactors), ActiveBall, 1: End Sub
-Sub Bumper3_Hit : vpmTimer.PulseSw(62) : PlaySoundatVol SoundFX("fx_bumper1",DOFContactors), ActiveBall, 1: End Sub
+Sub Bumper1_Hit : vpmTimer.PulseSw(52) : playsound SoundFX("fx_bumper1",DOFContactors): End Sub
+Sub Bumper2_Hit : vpmTimer.PulseSw(42) : playsound SoundFX("fx_bumper1",DOFContactors): End Sub
+Sub Bumper3_Hit : vpmTimer.PulseSw(62) : playsound SoundFX("fx_bumper1",DOFContactors): End Sub
 
 'Spinners
-Sub sw71_Spin:vpmTimer.PulseSw 71 : PlaySoundAtVol "fx_spinner", sw71, 1 : End Sub
-Sub sw61_Spin:vpmTimer.PulseSw 61 : PlaySoundAtVol "fx_spinner", sw61, 1 : End Sub
+Sub sw71_Spin:vpmTimer.PulseSw 71 : playsound"fx_spinner" : End Sub
+Sub sw61_Spin:vpmTimer.PulseSw 61 : playsound"fx_spinner" : End Sub
 
 'Generic Sounds
-Sub Trigger1_Hit:PlaySoundAtVol "fx_ballrampdrop", ActiveBall, 1:End Sub
-Sub Trigger2_Hit:PlaySoundAtVol "fx_ballrampdrop", ActiveBall, 1:End Sub
+Sub Trigger1_Hit:PlaySound "fx_ballrampdrop":End Sub
+Sub Trigger2_Hit:PlaySound "fx_ballrampdrop":End Sub
 
-Sub Trigger3_Hit:PlaySoundAtVol "Wire Ramp", ActiveBall, 1:End Sub
-Sub Trigger4_Hit:PlaySoundAtVol "Wire Ramp", ActiveBall, 1:End Sub
+Sub Trigger3_Hit:PlaySound "Wire Ramp":End Sub
+Sub Trigger4_Hit:PlaySound "Wire Ramp":End Sub
 
 '**********************************************************************************************************
  'VariTarget
@@ -488,12 +488,12 @@
   VariDir = -1:VariTimer.enabled = 1
  End Sub
 
-' Function dsin(angle)
+Function dsin(angle)
 '
 ' Calculates the sine of an angle given in degrees
 '
-' dsin = Sin(angle * 3.14159 / 180)
-' End Function
+dsin = Sin(angle * 3.14159 / 180)
+End Function
 
 Function dcos(angle)
 '
@@ -895,7 +895,7 @@
 
 Sub RightSlingShot_Slingshot
   vpmTimer.PulseSw 35
-    PlaySoundAtVol SoundFX("right_slingshot",DOFContactors), ActiveBall, 1
+    PlaySound SoundFX("right_slingshot",DOFContactors), 0,1, 0.05,0.05 '0,1, AudioPan(RightSlingShot), 0.05,0,0,1,AudioFade(RightSlingShot)
     RSling.Visible = 0
     RSling1.Visible = 1
     sling1.TransZ = -20
@@ -913,7 +913,7 @@
 
 Sub LeftSlingShot_Slingshot
   vpmTimer.PulseSw 25
-    PlaySoundAtVol SoundFX("left_slingshot",DOFContactors), ActiveBall, 1
+    PlaySound SoundFX("left_slingshot",DOFContactors), 0,1, -0.05,0.05 '0,1, AudioPan(LeftSlingShot), 0.05,0,0,1,AudioFade(LeftSlingShot)
     LSling.Visible = 0
     LSling1.Visible = 1
     sling2.TransZ = -20
@@ -930,173 +930,61 @@
 End Sub
 
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
+'*********************************************************************
+'                 Positional Sound Playback Functions
+'*********************************************************************
 
 ' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
 ' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
 ' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
 Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
   PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
 End Sub
 
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
+' Similar subroutines that are less complicated to use (e.g. simply use standard parameters for the PlaySound call)
 Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+    PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
 Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
+    PlaySoundAt soundname, ActiveBall
 End Sub
 
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
 
 '*********************************************************************
-'            Supporting Ball, Sound Functions and Math
+'                     Supporting Ball & Sound Functions
 '*********************************************************************
 
-Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Const Pi = 3.1415927
-
-Function dSin(degrees)
-  dsin = sin(degrees * Pi/180)
-End Function
-
-Function dCos(degrees)
-  dcos = cos(degrees * Pi/180)
-End Function
-
 Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
   Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / table1.height-1
-  If tmp > 0 Then
+    tmp = tableobj.y * 2 / table1.height-1
+    If tmp > 0 Then
     AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
+    Else
+        AudioFade = Csng(-((- tmp) ^10) )
+    End If
 End Function
 
 Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / table1.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / table1.width-1
-  If tmp > 0 Then
-    Pan = Csng(tmp ^10)
-  Else
-    Pan = Csng(-((- tmp) ^10) )
-  End If
+    Dim tmp
+    tmp = tableobj.x * 2 / table1.width-1
+    If tmp > 0 Then
+        AudioPan = Csng(tmp ^10)
+    Else
+        AudioPan = Csng(-((- tmp) ^10) )
+    End If
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
-End Function
-
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-End Function
-
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
-End Function
-
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
+    Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
+    Pitch = BallVel(ball) * 20
 End Function
 
 Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
-End Function
-
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
+    BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
 End Function
 
 '*****************************************
@@ -1128,23 +1016,16 @@
     If UBound(BOT) = -1 Then Exit Sub
 
   ' play the rolling sound for each ball
-
     For b = 0 to UBound(BOT)
-      If BallVel(BOT(b) ) > 1 Then
-        rolling(b) = True
-        if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-          ' PlaySound("fx_ballrolling" & b), -1, BallRollVol(BOT(b) )*.8, AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-        Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-          ' PlaySound("fx_ballrolling" & b), -1, BallRollVol(BOT(b) )*.2, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-        End If
-      Else
-        If rolling(b) = True Then
-          StopSound("fx_ballrolling" & b)
-          rolling(b) = False
+        If BallVel(BOT(b) ) > 1 AND BOT(b).z < 30 Then
+            rolling(b) = True
+            PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b)), AudioPan(BOT(b)), 0, Pitch(BOT(b)), 1, 0, AudioFade(BOT(b))
+        Else
+            If rolling(b) = True Then
+                StopSound("fx_ballrolling" & b)
+                rolling(b) = False
+            End If
         End If
-      End If
     Next
 End Sub
 
@@ -1153,11 +1034,10 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
 End Sub
 
 
-
 '*****************************************
 ' ninuzzu's FLIPPER SHADOWS
 '*****************************************
@@ -1192,11 +1072,11 @@
     ' render the shadow for each ball
     For b = 0 to UBound(BOT)
         If BOT(b).X < Table1.Width/2 Then
-            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) + 6
+            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) + 6
         Else
-            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) - 6
+            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) - 6
         End If
-        ballShadow(b).Y = BOT(b).Y + 4
+        ballShadow(b).Y = BOT(b).Y + 12
         If BOT(b).Z > 20 Then
             BallShadow(b).visible = 1
         Else
