--- vpxtable_scripts/Beavis and Butt-head_Pinballed.vbs
+++ todo_extracted_tables/Beavis and Butt-head_Pinballed.vbs
@@ -73,8 +73,6 @@
 Dim bsSaucer,bsScoop, bsTK, dtBankL, dtBankR, xx
 
 Sub Table1_Init
-    ' Thalamus : Was missing 'vpminit me'
-    vpminit me
          Controller.Games("beav_butt").Settings.Value("sound_mode") =1
          With Controller
     .GameName = cGameName
@@ -105,16 +103,12 @@
 
     Set bsScoop = New cvpmBallStack
     With bsScoop
-        .KickForceVar = 3
-        .KickAngleVar = 3
         .InitSaucer Sw52, 52, 120, 0
         .InitExitSnd SoundFX("CenterEject", DOFContactors), SoundFX("CenterEject", DOFContactors)
     End With
 
     Set bsTK = New cvpmBallStack
     With bsTK
-        .KickForceVar = 3
-        .KickAngleVar = 3
         .InitSaucer Sw53k, 53, 20, 50
         .InitExitSnd SoundFX("LeftEject", DOFContactors), SoundFX("LeftEject", DOFContactors)
         .KickZ = 3.1415926/3
