--- vpxtable_scripts/Big Bang Bar (Capcom 1996 ).vbs
+++ todo_extracted_tables/Big Bang Bar (Capcom 1996 ).vbs
@@ -800,10 +800,6 @@
 
 '********Table Init************
   Sub Table_Init
-
-' Thalamus : Was missing 'vpminit me'
-  vpminit me
-
   With Controller
     .GameName = cGameName
     .SplashInfoLine = "Big Bang Bar, Capcom 1996"
@@ -2594,10 +2590,10 @@
     RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
 End Function
 
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "Table" is the name of the table
+Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
   Dim tmp
   On Error Resume Next
-  tmp = tableobj.y * 2 / Table.height-1
+  tmp = tableobj.y * 2 / table1.height-1
   If tmp > 0 Then
     AudioFade = Csng(tmp ^10)
   Else
@@ -2605,10 +2601,10 @@
   End If
 End Function
 
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "Table" is the name of the table
+Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
   Dim tmp
   On Error Resume Next
-  tmp = tableobj.x * 2 / Table.width-1
+  tmp = tableobj.x * 2 / table1.width-1
   If tmp > 0 Then
     AudioPan = Csng(tmp ^10)
   Else
@@ -2616,10 +2612,10 @@
   End If
 End Function
 
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "Table" is the name of the table
+Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
   Dim tmp
   On Error Resume Next
-  tmp = ball.x * 2 / Table.width-1
+  tmp = ball.x * 2 / table1.width-1
   If tmp > 0 Then
     Pan = Csng(tmp ^10)
   Else
@@ -2803,9 +2799,3 @@
     PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
 End Sub
 
-' Thalamus : Exit in a clean and proper way
-Sub Table_exit
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
