--- vpxtable_scripts/Big Star (Williams 1972) v600.vbs
+++ todo_extracted_tables/Big Star (Williams 1972) v600.vbs
@@ -164,7 +164,6 @@
     If keycode = LeftTiltKey Then Nudge 90, 6:PlaySound "fx_nudge", 0, 1, -0.1, 0.25:CheckTilt
     If keycode = RightTiltKey Then Nudge 270, 6:PlaySound "fx_nudge", 0, 1, 0.1, 0.25:CheckTilt
     If keycode = CenterTiltKey Then Nudge 0, 6:PlaySound "fx_nudge", 0, 1, 1, 0.25:CheckTilt
-    If keycode = MechanicalTilt Then CheckTilt
 
     ' Funcionamiento normal de los flipers y otras teclas durante el juego
 
