--- vpxtable_scripts/Black and Red (Inder 1975) v600.vbs
+++ todo_extracted_tables/Black and Red (Inder 1975) v600.vbs
@@ -205,7 +205,6 @@
     If keycode = LeftTiltKey Then Nudge 90, 8:PlaySound "fx_nudge", 0, 1, -0.1, 0.25:CheckTilt
     If keycode = RightTiltKey Then Nudge 270, 8:PlaySound "fx_nudge", 0, 1, 0.1, 0.25:CheckTilt
     If keycode = CenterTiltKey Then Nudge 0, 9:PlaySound "fx_nudge", 0, 1, 1, 0.25:CheckTilt
-    If keycode = MechanicalTilt Then CheckTilt
 
     ' keys during game
 
