--- vpxtable_scripts/Bram Stokers Dracula (Williams 1993)_Thal_Mod_vpx8_v1.1.vbs
+++ todo_extracted_tables/Bram Stokers Dracula (Williams 1993)_Thal_Mod_vpx8_v1.1.vbs
@@ -2224,8 +2224,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     'playsound "fx_knocker"
     if debugOn then TBPout.text = str
   End Sub
