--- vpxtable_scripts/Bram Stokers Dracula (Williams 1993) VPW 1.0.1.vbs
+++ todo_extracted_tables/Bram Stokers Dracula (Williams 1993) VPW 1.0.1.vbs
@@ -563,7 +563,7 @@
   If keycode = RightTiltKey Then Nudge 270, 2:SoundNudgeRight
   If keycode = CenterTiltKey Then Nudge 0, 2:SoundNudgeCenter
   If keycode = keyInsertCoin1 or keycode = keyInsertCoin2 or keycode = keyInsertCoin3 or keycode = keyInsertCoin4 Then SoundCoinIn
-  If keycode = PlungerKey or keycode = LockBarKey Then
+  If keycode = PlungerKey Then
     Plunger.FireSpeed = 95 + 5*rnd
     Controller.Switch(34) = 1
     LaunchBut.Y = LaunchBut.Y - 7
@@ -583,7 +583,7 @@
     FlipperDeActivate RightFlipper, RFPress
     RightBut.X = RightBut.X + 10
   End If
-  If keycode = PlungerKey or keycode = LockBarKey Then
+  If keycode = PlungerKey Then
     Controller.Switch(34) = 0
     LaunchBut.Y = LaunchBut.Y + 7
     Flasher1.Y = Flasher1.Y + 7
@@ -2885,8 +2885,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     if debugOn then TBPout.text = str
   End Sub
 
@@ -2896,8 +2895,7 @@
     RealCOR = BallSpeed(aBall) / (cor.ballvel(aBall.id)+0.0001)
     coef = desiredcor / realcor
     If abs(aball.velx) < 2 and aball.vely < 0 and aball.vely > -3.75 then
-' Thalamus - patched :       aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+      aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     End If
   End Sub
 
@@ -3553,11 +3551,10 @@
     tmp =  - 7000
   End If
 
-' Thalamus, AudioFade patched
   If tmp > 0 Then
-    AudioFade = CSng(tmp ^ 5) 'was 10
+    AudioFade = CSng(tmp ^ 10)
   Else
-    AudioFade = CSng( - (( - tmp) ^ 5) ) 'was 10
+    AudioFade = CSng( - (( - tmp) ^ 10) )
   End If
 End Function
 
@@ -3571,11 +3568,10 @@
     tmp =  - 7000
   End If
 
-' Thalamus, AudioPan patched
   If tmp > 0 Then
-    AudioPan = CSng(tmp ^ 5) 'was 10
+    AudioPan = CSng(tmp ^ 10)
   Else
-    AudioPan = CSng( - (( - tmp) ^ 5) ) 'was 10
+    AudioPan = CSng( - (( - tmp) ^ 10) )
   End If
 End Function
 
