--- vpxtable_scripts/Buccaneer (Gottlieb 1976) v1.00.vbs
+++ todo_extracted_tables/Buccaneer (Gottlieb 1976) v1.00.vbs
@@ -16,35 +16,13 @@
 ' 101,102 flippers
 ' 126 ballrelease
 ' -------------------------------------------------
-Option Explicit
-Randomize
 
 ' Thalamus 2018-07-19
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
-' Thalamus 2018-08-09 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
+' No special SSF tweaks yet.
 
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolKick   = 1    ' Kicker volume.
-Const VolSpin   = 1.5  ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
+Option Explicit
+Randomize
 
 Const cGameName = "buccaneer"
 Dim PI: PI = Round(4 * Atn(1), 6)
@@ -244,7 +222,7 @@
 Sub Buccaneer_KeyDown(ByVal keycode)
 
   if keycode=AddCreditKey then
-    playsoundAtVol "coinin", drain, 1
+    playsound "coinin"
     coindelay.enabled=true
     end if
 
@@ -293,7 +271,7 @@
 
   If keycode = PlungerKey Then
     Plunger.PullBack
-    PlaySoundAtVol "fx_plungerpull", Plunger ,1
+    PlaySound "plungerpull",0,1,0.25,0.25
   End If
 
   If keycode=LeftFlipperKey and State = false and OperatorMenu=0 then
@@ -360,14 +338,14 @@
   if tilt=false and state=true then
   If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToEnd
-    PlaySoundAtVol SoundFXDOF("flipperup",101,DOFOn,DOFContactors), LeftFlipper, VolFlip
-    PlaySoundAtVol "Buzz", LeftFlipper, VolFlip
+    PlaySound SoundFXDOF("flipperup",101,DOFOn,DOFContactors), 0, .67, -0.05, 0.05
+    PlaySound "Buzz",-1,.05,-0.05, 0.05
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToEnd
-    PlaySoundAtVol SoundFXDOF("flipperup",102,DOFOn,DOFContactors), RightFlipper, VolFlip
-    PlaySoundAtVol "Buzz1", RightFlipper, VolFlip
+    PlaySound SoundFXDOF("flipperup",102,DOFOn,DOFContactors), 0, .67, 0.05, 0.05
+    PlaySound "Buzz1",-1,.05,0.05,0.05
   End If
 
   If keycode = LeftTiltKey Then
@@ -422,7 +400,7 @@
 
   If keycode = PlungerKey Then
     Plunger.Fire
-    PlaySoundAtVol "plunger", Plunger, 1
+    PlaySound "plunger",0,1,0.25,0.25
   End If
 
   if keycode = LeftFlipperKey then
@@ -435,7 +413,7 @@
         LeftFlipper.TimerEnabled = 1    ' nFozzy Flipper Code
         LeftFlipper.TimerInterval = 16
         LeftFlipper.return = returnspeed * 0.5
-    PlaySoundAtVol SoundFXDOF("flipperdown",101,DOFOff,DOFContactors), LeftFlipper, VolFlip
+    PlaySound SoundFXDOF("flipperdown",101,DOFOff,DOFContactors), 0, 1, -0.05, 0.05
     StopSound "Buzz"
   End If
 
@@ -444,7 +422,7 @@
         Rightflipper.TimerEnabled = 1  ' nFozzy Flipper Code
         Rightflipper.TimerInterval = 16
         Rightflipper.return = returnspeed * 0.5
-    PlaySoundAtVol SoundFXDOF("flipperdown",102,DOFOff,DOFContactors), RightFlipper, VolFlip
+    PlaySound SoundFXDOF("flipperdown",102,DOFOff,DOFContactors), 0, 1, 0.05, 0.05
     StopSound "Buzz1"
   End If
    End if
@@ -479,7 +457,7 @@
 
 Sub Drain_Hit()
   DOF 130, DOFPulse
-  PlaySoundAtVol "drain", Drain, 1
+  PlaySound "drain",0,1,0,0.25
   for each light in GIlights:light.state=0:next
   'for each light in lights:light.state=0:next
   me.timerenabled=1
@@ -603,12 +581,12 @@
     End If
     For each light in GIlights:light.state=0:next
   else
-  Drain.kick 60,38,0
+  Drain.kick 60,28,0
   biptext.text=ballinplay
     If B2SOn then Controller.B2ssetballinplay 32, Ballinplay
     ballreltimer.enabled=false
-  playsoundAtVol SoundFXDOF("kickerkick",126,DOFPulse,DOFContactors), Drain, 1
-    playsoundAtVol "kickerkick", Drain, 1
+  playsound SoundFXDOF("kickerkick",126,DOFPulse,DOFContactors)
+    playsound "kickerkick"
   end if
   ballreltimer.enabled=false
 end sub
@@ -633,7 +611,7 @@
 
 Sub Bumper1_Hit
    if tilt=false then
-  playsoundAtVol SoundFXDOF("fx_bumper4",105,DOFPulse,DOFContactors), Bumper1, VolBump
+  playsound SoundFXDOF("fx_bumper4",105,DOFPulse,DOFContactors)
   if BumperLight1.state=1 then
     addscore (bumperlitscore)
     else
@@ -660,7 +638,7 @@
 
 Sub Bumper2_Hit
    if tilt=false then
-  playsoundAtVol SoundFXDOF("fx_bumper4",106,DOFPulse,DOFContactors), Bumper2, VolBump
+  playsound SoundFXDOF("fx_bumper4",106,DOFPulse,DOFContactors)
   if BumperLight2.state=1 then
     addscore 1000  'Center Red Bumper
     else
@@ -686,7 +664,7 @@
 
 Sub Bumper3_Hit
    if tilt=false then
-  playsoundAtVol SoundFXDOF("fx_bumper4",107,DOFPulse,DOFContactors), Bumper3, VolBump
+  playsound SoundFXDOF("fx_bumper4",107,DOFPulse,DOFContactors)
   if BumperLight3.state=1 then
     addscore (bumperlitscore)
     else
@@ -712,7 +690,7 @@
 
 Sub Bumper4_Hit
    if tilt=false then
-  playsoundAtVol SoundFXDOF("fx_bumper4",108,DOFPulse,DOFContactors), Bumper4, VolBump
+  playsound SoundFXDOF("fx_bumper4",108,DOFPulse,DOFContactors)
   if balls = 3 then
     addscore 1000
     else
@@ -747,7 +725,7 @@
 
 Sub Bumper5_Hit
    if tilt=false then
-  playsoundAtVol SoundFXDOF("fx_bumper4",109,DOFPulse,DOFContactors), Bumper5, VolBump
+  playsound SoundFXDOF("fx_bumper4",109,DOFPulse,DOFContactors)
   if balls = 3 then
     addscore 1000
     else
@@ -938,7 +916,7 @@
 
 Sub RightSlingShot_Slingshot
   addscore 10
-  playsoundAtVol SoundFXDOF("right_slingshot",104,DOFPulse,DOFContactors), sling1, 1
+  playsound SoundFXDOF("right_slingshot",104,DOFPulse,DOFContactors)
     RSling.Visible = 0
     RSling1.Visible = 1
     sling1.TransZ = -10
@@ -957,7 +935,7 @@
 
 Sub LeftSlingShot_Slingshot
   addscore 10
-  playsoundAtVol SoundFXDOF("left_slingshot",103,DOFPulse,DOFContactors), sling2, 1
+  playsound SoundFXDOF("left_slingshot",103,DOFPulse,DOFContactors)
     LSling.Visible = 0
     LSling1.Visible = 1
     sling2.TransZ = -10
@@ -1347,38 +1325,38 @@
 ' depending of the speed of the collision.
 
 Sub a_Pins_Hit (idx)
-  PlaySound "pinhit_low", 0, Vol(ActiveBall)*VolPi, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "pinhit_low", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Targets_Hit (idx)
-  PlaySound "target", 0, Vol(ActiveBall)*VolTarg, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "target", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall)*VolGates, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Spinner_Spin(idx)
-    PlaySound SoundFXDOF("fx_spinner",129,DOFPulse,DOFKnocker),0,VolSpin,0,0.25, AudioFade(SpinnerRod)
+    PlaySound SoundFXDOF("fx_spinner",129,DOFPulse,DOFKnocker),0,.25,0,0.25
 End Sub
 
 Sub a_Rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1389,7 +1367,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1400,7 +1378,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolPo, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -1409,9 +1387,9 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -1425,9 +1403,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -1494,7 +1472,7 @@
 
 ' *******************************************************************************************************
 ' Positional Sound Playback Functions by DJRobX
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
+' PlaySound sound, 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
 ' *******************************************************************************************************
 
 ' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
@@ -1519,20 +1497,20 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
 
 Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
+  PlaySound sound, 0, Vol(ActiveBall) * VolMult, Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
 End Sub
 
 'Set position as bumperX and Vol manually.
 
 Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,1, 1, AudioFade(tableobj)
 End Sub
 
 '*********************************************************************
@@ -1580,7 +1558,7 @@
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -1626,9 +1604,9 @@
       If BallVel(BOT(b) ) > 1 Then
         rolling(b) = True
         if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
+          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
         Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
+          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, Pan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
         End If
       Else
         If rolling(b) = True Then
@@ -1644,7 +1622,11 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  If Buccaneer.VersionMinor > 3 OR Buccaneer.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
 
 ' Thalamus : Exit in a clean and proper way
