--- vpxtable_scripts/Contest (Gottlieb 1958) 52 Final 1.5.vbs
+++ todo_extracted_tables/Contest (Gottlieb 1958) 52 Final 1.5.vbs
@@ -3,26 +3,7 @@
 
 ' Thalamus 2018-07-20
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
-' Thalamus 2018-08-09 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolKick   = 1    ' Kicker volume.
-Const VolFlip   = 1    ' Flipper volume.
+' No special SSF tweaks yet.
 
 Dim Score(4)
 dim truesc(4)
@@ -228,7 +209,7 @@
 
 
     if keycode = AddCreditKey then
-  playsoundAtVol "coin3", drain, 1
+  playsound "coin3"
   coindelay.enabled=true
   end if
 
@@ -278,7 +259,7 @@
     activeballs=activeballs+1
   nb.CreateSizedballWithMass BallRadius,BallMass 'old nb.createball
     nb.kick 270,10
-playsoundAtVol SoundFXDOF("ballout",115,DOFPulse,DOFContactors), nb, VolKick
+playsound SoundFXDOF("ballout",115,DOFPulse,DOFContactors)
     end if
 
     if tilt=false and state=true then
@@ -286,17 +267,15 @@
     If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToEnd
     Flipper1.RotateToEnd
-PlaySoundAtVol SoundFXDOF("FlipperUp",101,DOFOn,DOFFlippers), LeftFlipper, VolFlip
-PlaySoundAtVol "FlipperUp", Flipper1, VolFlip
-    playsoundAtVol "buzz", LeftFlipper, VolFlip
+PlaySound SoundFXDOF("FlipperUp",101,DOFOn,DOFFlippers)
+    playsound "buzz"
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToEnd
         Flipper2.RotateToEnd
-PlaySoundAtVol SoundFXDOF("FlipperUp",102,DOFOn,DOFFlippers), RightFlipper, VolFlip
-PlaySoundAtVol "FlipperUp", Flipper2, VolFlip
-    playsoundAtVol "buzz1", RightFlipper, VolFlip
+PlaySound SoundFXDOF("FlipperUp",102,DOFOn,DOFFlippers)
+    playsound "buzz1"
   End If
   End If
 
@@ -327,22 +306,20 @@
 
   If keycode = PlungerKey Then
     Plunger.Fire
-    playsoundAtVol "plunger", plunger, 1
+    playsound "plunger"
   End If
 
   If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToStart
     Flipper1.RotateToStart
-if state=true and tilt=false then PlaySoundAtVol SoundFXDOF("FlipperDown",101,DOFOff,DOFFlippers), LeftFlipper, VolFlip
-if state=true and tilt=false then PlaySoundAtVol "FlipperDown", Flipper1, VolFlip
+if state=true and tilt=false then PlaySound SoundFXDOF("FlipperDown",101,DOFOff,DOFFlippers)
     stopsound "buzz"
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToStart
         Flipper2.RotateToStart
-if state=true and tilt=false then PlaySoundAtVol SoundFXDOF("FlipperDown",102,DOFOff,DOFFlippers), RightFlipper, VolFlip
-if state=true and tilt=false then PlaySoundAtVol "FlipperDown", Flipper2, VolFlip
+if state=true and tilt=false then PlaySound SoundFXDOF("FlipperDown",102,DOFOff,DOFFlippers)
     stopsound "buzz1"
   End If
 
@@ -666,7 +643,7 @@
 end sub
 
 Sub Drain_Hit()
-    playsoundAtVol "drainshort", drain, 1
+    playsound "drainshort"
 DOF 116, DOFPulse
   Drain.DestroyBall
   if ballinplay<5 then playsound "motorshort"
@@ -880,7 +857,7 @@
 '************** Slingshots ***********************
 
 Sub RightSlingShot_Slingshot
-  playsoundAtVol SoundFXDOF("rightslingshot",104,DOFPulse,DOFContactors), slingR, 1
+  playsound SoundFXDOF("rightslingshot",104,DOFPulse,DOFContactors)
   DOF 106, DOFPulse
   if GI1.state=1 then
   addscore 10
@@ -908,7 +885,7 @@
 End Sub
 
 Sub LeftSlingShot_Slingshot
-  playsoundAtVol SoundFXDOF("leftslingshot",103,DOFPulse,DOFContactors), slingL, 1
+  playsound SoundFXDOF("leftslingshot",103,DOFPulse,DOFContactors)
   DOF 105, DOFPulse
   if GI01.state=1 then
   addscore 10
@@ -936,7 +913,7 @@
 End Sub
 
 Sub LeftSlingShotb_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFcontactors), slingLb, 1
+  playsound SoundFX("leftslingshot",DOFcontactors)
   DOF 124, DOFPulse
   addscore 1
   if GL = "L" then
@@ -960,7 +937,7 @@
 End Sub
 
 Sub LeftSlingShotc_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFcontactors), slingLc, 1
+  playsound SoundFX("leftslingshot",DOFcontactors)
   DOF 122, DOFPulse
     addscore 5
     if motion=0 and tilt=false and c=0 then
@@ -991,7 +968,7 @@
 End Sub
 
 Sub LeftSlingShotd_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFContactors), slingLd, 1
+  playsound SoundFX("leftslingshot",DOFContactors)
   DOF 120, DOFPulse
   addscore 1
   if GL = "L" then
@@ -1015,7 +992,7 @@
 End Sub
 
 Sub LeftSlingShote_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFcontactors), slingLe, 1
+  playsound SoundFX("leftslingshot",DOFcontactors)
   DOF 126, DOFPulse
   addscore 1
   if GL = "L" then
@@ -1039,7 +1016,7 @@
 End Sub
 
 Sub RightSlingShotb_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFContactors), slingRb, 1
+  playsound SoundFX("leftslingshot",DOFContactors)
   DOF 125, DOFPulse
   addscore 1
   if GL = "L" then
@@ -1063,7 +1040,7 @@
 End Sub
 
 Sub RightSlingShotc_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFcontactors), slingRc, 1
+  playsound SoundFX("leftslingshot",DOFcontactors)
   DOF 123, DOFPulse
     addscore 5
     if motion=0 and tilt=false and c=0 then
@@ -1094,7 +1071,7 @@
 End Sub
 
 Sub RightSlingShotd_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFcontactors),slingRd, 1
+  playsound SoundFX("leftslingshot",DOFcontactors)
   DOF 121, DOFPulse
   addscore 1
   if GL = "L" then
@@ -1118,7 +1095,7 @@
 End Sub
 
 Sub RightSlingShote_Slingshot
-  playsoundAtVol SoundFX("leftslingshot",DOFcontactors), slingRe, 1
+  playsound SoundFX("leftslingshot",DOFcontactors)
   DOF 126, DOFPulse
   addscore 1
   if GL = "L" then
@@ -1151,7 +1128,7 @@
     addscore 1
   end if
     if tilt=false then
-playsoundAtVol SoundFXDOF("jet1",107,DOFPulse,DOFContactors), bumper1, VolBump
+playsound SoundFXDOF("jet1",107,DOFPulse,DOFContactors)
 DOF 109, DOFPulse
     ring1.isdropped=false
     end if
@@ -1165,7 +1142,7 @@
     addscore 1
   end if
     if tilt=false then
-playsoundAtVol SoundFXDOF("jet1",108,DOFPulse,DOFContactors), bumper2, VolBump
+playsound SoundFXDOF("jet1",108,DOFPulse,DOFContactors)
 DOF 110, DOFPulse
     ring2.isdropped=false
     end if
@@ -1309,15 +1286,15 @@
 end Sub
 
 sub gate_hit
-playsoundAtVol "gate", gate, VolGates
+playsound "gate"
 end Sub
 
 sub gate1_hit
-playsoundAtVol "gate", gate1,VolGates
+playsound "gate"
 end Sub
 
 sub gate2_hit
-playsoundAtVol "gate", gate2,VolGates
+playsound "gate"
 end Sub
 
 '******************************************************
@@ -1331,13 +1308,13 @@
     addscore 100
 end sub
 
-Sub SW24_hit():sw24p.transx = -10:Me.TimerEnabled = 1:playsoundAtVol "metal4", ActiveBall, 1:ltarga:DOF 111, DOFPulse:End Sub
+Sub SW24_hit():sw24p.transx = -10:Me.TimerEnabled = 1:playsound "metal4":ltarga:DOF 111, DOFPulse:End Sub
 Sub SW24_Timer():sw24p.transx = 0:Me.TimerEnabled = 0:End Sub
 
-Sub SW25_hit():sw25p.transx = -10:Me.TimerEnabled = 1:playsoundAtVol "metal4", ActiveBall, 1:ctarga:DOF 112, DOFPulse:End Sub
+Sub SW25_hit():sw25p.transx = -10:Me.TimerEnabled = 1:playsound "metal4":ctarga:DOF 112, DOFPulse:End Sub
 Sub SW25_Timer():sw25p.transx = 0:Me.TimerEnabled = 0:End Sub
 
-Sub SW26_hit():sw26p.transx = -10:Me.TimerEnabled = 1:playsoundAtVol "metal4", ActiveBall, 1:rtarga:DOF 113, DOFPulse:End Sub
+Sub SW26_hit():sw26p.transx = -10:Me.TimerEnabled = 1:playsound "metal4":rtarga:DOF 113, DOFPulse:End Sub
 Sub SW26_Timer():sw26p.transx = 0:Me.TimerEnabled = 0:End Sub
 
 sub rtarga
@@ -1358,7 +1335,7 @@
 '******         Wheel price et score            *******
 '******************************************************
 
-Sub SWheelPrice_hit():Wheel.transx = 10:Me.TimerEnabled = 1:playsoundAtVol "metal4", Wheel, 1:swheelscore:DOF 114, DOFPulse:End Sub
+Sub SWheelPrice_hit():Wheel.transx = 10:Me.TimerEnabled = 1:playsound "metal4":swheelscore:DOF 114, DOFPulse:End Sub
 Sub SWheelPrice_Timer():Wheel.transx = 0:Me.TimerEnabled = 0:End Sub
 
 sub swheelscore
@@ -1383,12 +1360,12 @@
 '****                 Holes            ****
 '************************************************
 sub lgob_hit
-    playsoundAtVol "drain", drain, 1
+    playsound "drain"
     if lgl.state=1 then addscore (rsv*10) else addscore (rsv) end if
 end sub
 
 sub rgob_hit
-    playsoundAtVol "drain", rgob, 1
+    playsound "drain"
     if rgl.state=1 then addscore (rsv*10) else addscore (rsv) end if
 end sub
 
@@ -1494,7 +1471,7 @@
 
 sub ballrel_hit
     if ballrelenabled=1 then
-    playsoundAtVol "launchball", ballrel, 1
+    playsound "launchball"
     ballrelenabled=0
     end if
 end sub
@@ -1907,7 +1884,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*10*VolRH , AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+    PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*10*GlobalSoundLevel , pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 1 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1916,20 +1893,20 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*10*VolRH , AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*10*VolRH , AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*10*VolRH , AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*10*GlobalSoundLevel , pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*10*GlobalSoundLevel , pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*10*GlobalSoundLevel , pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
 'Metals Collision
 Sub MetalsThin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*GlobalSoundLevel, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 'Post Collision
 Sub wood_hit(idx)
-PlaySound "Post1", 0, Vol(ActiveBall)*10*VolWood , AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+PlaySound "Post1", 0, Vol(ActiveBall)*10*GlobalSoundLevel , pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 'Flippers Collision
@@ -1938,9 +1915,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 :PlaySound "flip_hit_1", 0, VolRH * ballvel(ActiveBall) / 50, -0.1, 0.25
-    Case 2 :PlaySound "flip_hit_2", 0, VolRH * ballvel(ActiveBall) / 50, -0.1, 0.25
-    Case 3 :PlaySound "flip_hit_3", 0, VolRH * ballvel(ActiveBall) / 50, -0.1, 0.25
+    Case 1 :PlaySound "flip_hit_1", 0, GlobalSoundLevel * ballvel(ActiveBall) / 50, -0.1, 0.25
+    Case 2 :PlaySound "flip_hit_2", 0, GlobalSoundLevel * ballvel(ActiveBall) / 50, -0.1, 0.25
+    Case 3 :PlaySound "flip_hit_3", 0, GlobalSoundLevel * ballvel(ActiveBall) / 50, -0.1, 0.25
   End Select
 End Sub
 
@@ -2044,10 +2021,10 @@
 
 ' *******************************************************************************************************
 ' Positional Sound Playback Functions by DJRobX
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
+' PlaySound sound, 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
 ' *******************************************************************************************************
 
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo AudioPanning only)
+' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
 ' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
 ' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
 
@@ -2069,20 +2046,20 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
 
 Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
+  PlaySound sound, 0, Vol(ActiveBall) * VolMult, Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
 End Sub
 
 'Set position as bumperX and Vol manually.
 
 Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,1, 1, AudioFade(tableobj)
 End Sub
 
 '*********************************************************************
@@ -2099,7 +2076,7 @@
   End If
 End Function
 
-Function AudioPan(tableobj) ' Calculates the AudioPan for a tableobj based on the X position on the table. "table1" is the name of the table
+Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
   Dim tmp
   tmp = tableobj.x * 2 / table1.width-1
   If tmp > 0 Then
@@ -2109,8 +2086,28 @@
   End If
 End Function
 
+Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
+    Dim tmp
+    tmp = ball.x * 2 / table1.width-1
+    If tmp > 0 Then
+        Pan = Csng(tmp ^10)
+    Else
+        Pan = Csng(-((- tmp) ^10) )
+    End If
+End Function
+
+Function AudioFade(ball) ' Can this be together with the above function ?
+  Dim tmp
+  tmp = ball.y * 2 / Table1.height-1
+  If tmp > 0 Then
+    AudioFade = Csng(tmp ^10)
+  Else
+    AudioFade = Csng(-((- tmp) ^10) )
+  End If
+End Function
+
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -2145,9 +2142,9 @@
       If BallVel(BOT(b) ) > 1 Then
         rolling(b) = True
         if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
+          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
         Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
+          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, Pan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
         End If
       Else
         If rolling(b) = True Then
@@ -2163,7 +2160,11 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  If Table1.VersionMinor > 3 OR Table1.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
 
 ' Thalamus : Exit in a clean and proper way
