--- vpxtable_scripts/Count-Down (Gottlieb 1979) VPW v1.1.6.vbs
+++ todo_extracted_tables/Count-Down (Gottlieb 1979) VPW v1.1.6.vbs
@@ -409,9 +409,7 @@
 
 Sub Table_Paused:Controller.Pause = 1:End Sub
 Sub Table_unPaused:Controller.Pause = 0:End Sub
-
-' Thalamus - table is named Table1 - not Table !
-Sub Table1_exit()
+Sub Table_exit()
   debug.print "Exiting"
   HighScoreSave
   Controller.Pause = False
