--- vpxtable_scripts/Cuphead Pro (D. Goblett 2020)_vpx8.vbs
+++ todo_extracted_tables/Cuphead Pro (D. Goblett 2020)_vpx8.vbs
@@ -4619,8 +4619,7 @@
 '   tb.text = "Coef = " & coef
 
 '               Applies the coef to x and y velocities
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
   End Sub
 
 '***********This Sub sets the values for Sleeves (or any other future objects) to 85% (or whatever is passed in) of Posts
