--- vpxtable_scripts/Demolition Man (Williams 1994)_Bigus(MOD)1.2.vbs
+++ todo_extracted_tables/Demolition Man (Williams 1994)_Bigus(MOD)1.2.vbs
@@ -49,34 +49,7 @@
     'Bug Fixes
 
 
-Option Explicit
-Randomize
 
-' Thalamus 2018-11-01 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-' Has it's own fast flip routine - consider disabling it and using the new from core.vbs ??
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolKick   = 1    ' Kicker volume.
-Const VolSpin   = 1.5  ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
 
 
 Option Explicit
@@ -96,7 +69,7 @@
 
 
  '***ROM***ROM***ROM***ROM***
- Const cGameName = "dm_lx4" ' use dm_h6 for profanity
+ Const cGameName = "dm_lx4"
 '***************************
 
  Const UseSolenoids = 2
@@ -143,7 +116,7 @@
       PinMAMETimer.Interval = PinMAMEInterval
       PinMAMETimer.Enabled = true
       vpmNudge.TiltSwitch = 14
-      vpmNudge.Sensitivity = 3
+      vpmNudge.Sensitivity = 2
      End With
 
 
@@ -391,7 +364,7 @@
 
 Sub SolClawMotorLeft(Enabled)
   If enabled then
-    PlaySoundAtVol SoundFX("Motor",DofGear), Claw, 20
+    PlaySound SoundFX("Motor",DofGear)
   Else
     StopSound SoundFX("Motor",DofGear)
   End if
@@ -412,7 +385,7 @@
 End Sub
 
 Sub UpdateElevator(aNewPos,aSpeed,aLastPos)
-  PlaySoundAtVol SoundFX("ElevatorMotor",DOFGear), BallP, 10
+  PlaySound SoundFX("ElevatorMotor",DOFGear)
   Elevator.TransY = elevatormech.Position
   BallP.TransY = elevatormech.Position
 End Sub
@@ -479,7 +452,7 @@
 Dim aBall
 
 Sub sw73_Hit()
-    PlaySoundAtVol "HeadquarterHit2", sw73, 1
+    PlaySound "HeadquarterHit2", 0, 1, pan(ActiveBall)
     Set aBall = ActiveBall:Me.TimerEnabled = 1
     bsTopPopper.AddBall Me
 End Sub
@@ -510,13 +483,13 @@
 End Sub
 
 Sub sw66_Hit()
-  PlaySoundAtVol "EjectKick", sw66, 1
+  PlaySound ""
   bsEject.AddBall Me
 End Sub
 
 Sub Drain_Hit
   bsTrough.AddBall Me
-  PlaySoundAtVol "drain", drain, 1
+  PlaySound "drain",0,0.5,0
 End Sub
 
 
@@ -532,8 +505,6 @@
 
   If KeyCode = LeftFlipperKey then FastFlips.FlipL True' :  FastFlips.FlipUL True
   If KeyCode = RightFlipperKey then FastFlips.FlipR True' :  FastFlips.FlipUR True
-    If keycode = LockBarKey Then Controller.Switch(11) = 1: Controller.Switch(12) = 1
-    If keycode = RightMagnaSave Then Controller.Switch(11) = 1: Controller.Switch(12) = 1
     If keycode = PlungerKey Then Controller.Switch(11) = 1: Controller.Switch(12) = 1
 '    If keycode = LeftMagnaSave Or RightMagnaSave Then Controller.Switch(12) = 1
     If keycode = keyFront Then Controller.Switch(23) = 1
@@ -549,8 +520,7 @@
 
   If KeyCode = LeftFlipperKey then FastFlips.FlipL False' :  FastFlips.FlipUL False
   If KeyCode = RightFlipperKey then FastFlips.FlipR False' :  FastFlips.FlipUR False
-    If keycode = LockBarKey Then Controller.Switch(11) = 0: Controller.Switch(12) = 0
-     If keycode = RightMagnaSave Then Controller.Switch(11) = 0: Controller.Switch(12) = 0
+
 
      If vpmKeyUp(keycode) Then Exit Sub
      If keycode = PlungerKey Then Controller.Switch(11) = 0: Controller.Switch(12) = 0
@@ -570,19 +540,17 @@
 
  Sub SolLFlipper(Enabled)
      If Enabled Then
-         PlaySoundAtVol SoundFX("FlipperUpLeftBoth",DOFContactors), LeftFlipper, VolFlip:LeftFlipper.RotateToEnd:LeftFlipper1.RotateToEnd
-         PlaySoundAtVol "FlipperUpLeftBoth", LeftFlipper1, VolFlip
+         PlaySound SoundFX("FlipperUpLeftBoth",DOFContactors):LeftFlipper.RotateToEnd:LeftFlipper1.RotateToEnd
      Else
-         PlaySoundAtVol SoundFX("FlipperDown",DOFContactors), LeftFlipper, VolFlip:LeftFlipper.RotateToStart:LeftFlipper1.RotateToStart
-         PlaySoundAtVol "FlipperDown",LeftFlipper1, VolFlip
+         PlaySound SoundFX("FlipperDown",DOFContactors):LeftFlipper.RotateToStart:LeftFlipper1.RotateToStart
      End If
  End Sub
 
  Sub SolRFlipper(Enabled)
      If Enabled Then
-         PlaySoundAtVol SoundFX("FlipperUpRight",DOFContactors), RightFlipper, VolFlip:RightFlipper.RotateToEnd
+         PlaySound SoundFX("FlipperUpRight",DOFContactors):RightFlipper.RotateToEnd
      Else
-         PlaySoundAtVol SoundFX("FlipperDown",DOFContactors), RightFlipper, VolFlip:RightFlipper.RotateToStart
+         PlaySound SoundFX("FlipperDown",DOFContactors):RightFlipper.RotateToStart
      End If
  End Sub
 
@@ -597,7 +565,7 @@
   if enabled then
     AP = True
     Kicker1.Kick 1,48
-    PlaySoundAtVol SoundFX("Plunger",DOFContactors), Kicker1, VolKick
+    PlaySound SoundFX("Plunger",DOFContactors)
   End if
 End Sub
 
@@ -613,25 +581,25 @@
  '*********
 
 
-Sub sw27_Hit:Controller.Switch(27) = 1:sw27wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub    'shooterlane
+Sub sw27_Hit:Controller.Switch(27) = 1:sw27wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'shooterlane
 Sub sw27_UnHit:Controller.Switch(27) = 0:sw27wire.RotX = 0: End Sub
-Sub sw15_Hit:Controller.Switch(15) = 1:sw15wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'left outlane
+Sub sw15_Hit:Controller.Switch(15) = 1:sw15wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'left outlane
 Sub sw15_UnHit:Controller.Switch(15) = 0:sw15wire.RotX = 0:End Sub
-Sub sw16_Hit:Controller.Switch(16) = 1:sw16wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'left inlane
+Sub sw16_Hit:Controller.Switch(16) = 1:sw16wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'left inlane
 Sub sw16_UnHit:Controller.Switch(16) = 0:sw16wire.RotX = 0:End Sub
-Sub sw17_Hit:Controller.Switch(17) = 1:sw17wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'right inlane
+Sub sw17_Hit:Controller.Switch(17) = 1:sw17wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'right inlane
 Sub sw17_UnHit:Controller.Switch(17) = 0:sw17wire.RotX = 0:End Sub
-Sub sw18_Hit:Controller.Switch(18) = 1:sw18wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'right outlane
+Sub sw18_Hit:Controller.Switch(18) = 1:sw18wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'right outlane
 Sub sw18_UnHit:Controller.Switch(18) = 0:sw18wire.RotX = 0:End Sub
-Sub sw63_Hit:Controller.Switch(63) = 1:sw63wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'leftrollover
+Sub sw63_Hit:Controller.Switch(63) = 1:sw63wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'leftrollover
 Sub sw63_UnHit:Controller.Switch(63) = 0:sw63wire.RotX = 0:End Sub
-Sub sw64_Hit:Controller.Switch(64) = 1:sw64wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'centerrollover
+Sub sw64_Hit:Controller.Switch(64) = 1:sw64wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'centerrollover
 Sub sw64_UnHit:Controller.Switch(64) = 0:sw64wire.RotX = 0:End Sub
-Sub sw65_Hit:Controller.Switch(65) = 1:sw65wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'rightrollover
+Sub sw65_Hit:Controller.Switch(65) = 1:sw65wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'rightrollover
 Sub sw65_UnHit:Controller.Switch(65) = 0:sw65wire.RotX = 0:End Sub
-Sub sw48_Hit:Controller.Switch(48) = 1:sw48wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'Right freeway
+Sub sw48_Hit:Controller.Switch(48) = 1:sw48wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'Right freeway
 Sub sw48_UnHit:Controller.Switch(48) = 0:sw48wire.RotX = 0:End Sub
-Sub sw55_Hit:Controller.Switch(55) = 1:sw55wire.RotX = 15:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub  'leftloop
+Sub sw55_Hit:Controller.Switch(55) = 1:sw55wire.RotX = 15:PlaySound "metalhit_thin":End Sub   'leftloop
 Sub sw55_UnHit:Controller.Switch(55) = 0:sw55wire.RotX = 0:End Sub
 Sub sw86_Hit:Controller.Switch(86) = 1:End Sub    'upperleftflippergate
 Sub sw86_UnHit:Controller.Switch(86) = 0:End Sub
@@ -653,16 +621,16 @@
 Sub sw62_UnHit:Controller.Switch(62) = 0:End Sub
 
 
-Sub sw82_Hit:Controller.Switch(82) = 1:sw82wire.RotX = 75:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub    'Claw SuperJets
+Sub sw82_Hit:Controller.Switch(82) = 1:sw82wire.RotX = 75:PlaySound "metalhit_thin":End Sub   'Claw SuperJets
 Sub sw82_UnHit:Controller.Switch(82) = 0:sw82wire.RotX = 90: End Sub
 
-Sub sw83_Hit:Controller.Switch(83) = 1:sw83wire.RotX = 70:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub    'Claw PrisonBreak
+Sub sw83_Hit:Controller.Switch(83) = 1:sw83wire.RotX = 70:PlaySound "metalhit_thin":End Sub   'Claw PrisonBreak
 Sub sw83_UnHit:Controller.Switch(83) = 0:sw83wire.RotX = 90: End Sub
 
-Sub sw84_Hit:Controller.Switch(84) = 1:sw84wire.RotX = 75:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub    'Claw Freeze
+Sub sw84_Hit:Controller.Switch(84) = 1:sw84wire.RotX = 75:PlaySound "metalhit_thin":End Sub   'Claw Freeze
 Sub sw84_UnHit:Controller.Switch(84) = 0:sw84wire.RotX = 90: End Sub
 
-Sub sw85_Hit:Controller.Switch(85) = 1:sw85wire.RotX = 75:PlaySoundAtVol "metalhit_thin", ActiveBall, VolMetal:End Sub    'Claw ACMAG
+Sub sw85_Hit:Controller.Switch(85) = 1:sw85wire.RotX = 75:PlaySound "metalhit_thin":End Sub   'Claw ACMAG
 Sub sw85_UnHit:Controller.Switch(85) = 0:sw85wire.RotX = 90: End Sub
 
 
@@ -1253,74 +1221,67 @@
 Dim SoundBall
 
 Sub BallDropSoundCenter()
-  PlaySoundAtVol "BallDrop", BallP, 1
+  PlaySound "BallDrop"
 End Sub
 
 Sub BallDropSoundLeft()
-  PlaySoundAtVol "BallDrop", sw16, 1
+  PlaySound "BallDrop", 0, 1, -0.2
 End Sub
 
 Sub BallDropSoundRight()
-  PlaySoundAtVol "BallDrop", sw17, 1
+  PlaySound "BallDrop", 0, 1, 0.2
 End Sub
 
 
 Sub WireStartAcmag_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
 Sub WireStartLockFreeze_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
 Sub WireStartSuperJets_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                   '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                   '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
 Sub WireStartLeftRamp_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
 Sub WireStartRightRamp_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
 Sub WireStartMiddleRamp_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
 Sub WireStartBottomPopper_Hit()
  Set SoundBall = Activeball 'Ball-assignment
- ' playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
- playsoundAtVol "WireRamp", ActiveBall, 1
+ playsound "WireRamp",-1,0.6,pan(ActiveBall)                    '-1 = looping, Panning = -1 bis 1, je nach Position
 End Sub
 
 
-Sub WireStopLeft_Hit(): StopSound "WireRamp":PlaySoundAtVol "WireRamp_Stop", sw16, 1: vpmTimer.AddTimer 200, "BallDropSoundLeft'": End Sub
+Sub WireStopLeft_Hit(): StopSound "WireRamp":PlaySound "WireRamp_Stop": vpmTimer.AddTimer 200, "BallDropSoundLeft'": End Sub
 
-Sub WireStopRight_Hit(): StopSound "WireRamp":PlaySoundAtVol "WireRamp_Stop", sw17, 1: vpmTimer.AddTimer 200, "BallDropSoundRight'": End Sub
+Sub WireStopRight_Hit(): StopSound "WireRamp":PlaySound "WireRamp_Stop": vpmTimer.AddTimer 200, "BallDropSoundRight'": End Sub
 
-Sub WireStopRetina_Hit(): StopSound "WireRamp":PlaySoundAtVol "WireRamp_Stop", BallP, 1: vpmTimer.AddTimer 200, "BallDropSoundLeft'": End Sub
+Sub WireStopRetina_Hit(): StopSound "WireRamp":PlaySound "WireRamp_Stop": vpmTimer.AddTimer 200, "BallDropSoundLeft'": End Sub
 
-Sub WireStopSuperJets_Hit(): StopSound "WireRamp":PlaySoundAtVol "WireRamp_Stop", sw63, 1: vpmTimer.AddTimer 200, "BallDropSoundCenter'": End Sub
+Sub WireStopSuperJets_Hit(): StopSound "WireRamp":PlaySound "WireRamp_Stop": vpmTimer.AddTimer 200, "BallDropSoundCenter'": End Sub
 
 
 
@@ -1333,12 +1294,12 @@
         DiverterR.rotatetoend
     DiverterOff.isdropped = 1
     DiverterOn.isdropped = 0
-    PlaySoundAtVol SoundFX ("DiverterRight",DOFContactors), DiverterR, 1
+    PlaySound SoundFX ("DiverterRight",DOFContactors)
     Else
     DiverterOff.isdropped = 0
     DiverterOn.isdropped = 1
     DiverterR.rotatetostart
-    PlaysoundAtVol SoundFX ("DiverterRight",DOFContactors), DiverterR, 1
+    Playsound SoundFX ("DiverterRight",DOFContactors)
   End if
 End Sub
 
@@ -1346,26 +1307,26 @@
  ' StandupTargets
  '***************
 
-Sub Standup77_Hit: vpmTimer.pulseSw 77:Standup77p.RotX=Standup77p.RotX +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup77_Hit: vpmTimer.pulseSw 77:Standup77p.RotX=Standup77p.RotX +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup77_Timer:Standup77p.RotX=Standup77p.RotX -15:Me.TimerEnabled = 0: End Sub
 
-Sub Standup87_Hit: vpmTimer.pulseSw 87:Standup87p.RotX=Standup87p.RotX +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup87_Hit: vpmTimer.pulseSw 87:Standup87p.RotX=Standup87p.RotX +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup87_Timer:Standup87p.RotX=Standup87p.RotX -15:Me.TimerEnabled = 0: End Sub
 
 
-Sub Standup78_Hit: vpmTimer.pulseSw 78:Standup78p.RotZ=Standup78p.RotZ +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup78_Hit: vpmTimer.pulseSw 78:Standup78p.RotZ=Standup78p.RotZ +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup78_Timer:Standup78p.RotZ=Standup78p.RotZ -15:Me.TimerEnabled = 0: End Sub
 
-Sub Standup38_Hit: vpmTimer.pulseSw 38:Standup38p.RotZ=Standup38p.RotZ +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup38_Hit: vpmTimer.pulseSw 38:Standup38p.RotZ=Standup38p.RotZ +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup38_Timer:Standup38p.RotZ=Standup38p.RotZ -15:Me.TimerEnabled = 0: End Sub
 
-Sub Standup57_Hit: vpmTimer.pulseSw 57:Standup57p.RotZ=Standup57p.RotZ +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup57_Hit: vpmTimer.pulseSw 57:Standup57p.RotZ=Standup57p.RotZ +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup57_Timer:Standup57p.RotZ=Standup57p.RotZ -15:Me.TimerEnabled = 0: End Sub
 
-Sub Standup58_Hit: vpmTimer.pulseSw 58:Standup58p.RotZ=Standup58p.RotZ +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup58_Hit: vpmTimer.pulseSw 58:Standup58p.RotZ=Standup58p.RotZ +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup58_Timer:Standup58p.RotZ=Standup58p.RotZ -15:Me.TimerEnabled = 0: End Sub
 
-Sub Standup56_Hit: vpmTimer.pulseSw 56:Standup56p.RotZ=Standup56p.RotZ +15:PlaysoundAtVol SoundFX("target",DOFContactors), ActiveBall, VolTarg:Me.TimerEnabled = 1: End Sub
+Sub Standup56_Hit: vpmTimer.pulseSw 56:Standup56p.RotZ=Standup56p.RotZ +15:Playsound SoundFX("target",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Standup56_Timer:Standup56p.RotZ=Standup56p.RotZ -15:Me.TimerEnabled = 0: End Sub
 
 
@@ -1377,9 +1338,9 @@
  ' Bumper
  '*********
 
-Sub leftjetbumper_Hit:vpmTimer.PulseSw 43:PlaysoundAtVol SoundFX ("bumperleft",DOFContactors), leftjetbumper, VolBump:Me.TimerEnabled = 1: End Sub
+Sub leftjetbumper_Hit:vpmTimer.PulseSw 43:Playsound SoundFX ("bumperleft",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub Leftjetbumper_Timer:Me.TimerEnabled = 0:End Sub
-Sub rightjetbumper_Hit:vpmTimer.PulseSw 45:PlaysoundAtVol SoundFX ("bumperright",DOFContactors), rightjetbumper, VolBump:Me.TimerEnabled = 1: End Sub
+Sub rightjetbumper_Hit:vpmTimer.PulseSw 45:Playsound SoundFX ("bumperright",DOFContactors):Me.TimerEnabled = 1: End Sub
 Sub rightjetbumper_Timer:Me.TimerEnabled = 0: End Sub
 
 
@@ -1390,7 +1351,7 @@
 Dim RStep, LStep, TStep
 
 Sub RightSlingShot_Slingshot
-    PlaySoundAtVol SoundFX ("right_slingshot",DOFContactors), sling1, 1
+    PlaySound SoundFX ("right_slingshot",DOFContactors), 0, 1, 0.05, 0.05
     RSling.Visible = 0
     RSling1.Visible = 1
     sling1.TransZ = -20
@@ -1410,7 +1371,7 @@
 
 
 Sub LeftSlingShot_Slingshot
-    PlaySoundAtVol SoundFX ("left_slingshot",DOFContactors), sling2, 1
+    PlaySound SoundFX ("left_slingshot",DOFContactors),0,1,-0.05,0.05
     LSling.Visible = 0
     LSling1.Visible = 1
     sling2.TransZ = -20
@@ -1430,7 +1391,7 @@
 
 
 Sub TopSlingShot_Slingshot
-  PlaySoundAtVol SoundFX ("right_slingshot",DOFContactors), sling3, 1
+  PlaySound SoundFX ("right_slingshot",DOFContactors),0,1,0,0.05
   TSling.Visible = 0
   TSling1.Visible = 1
   Sling3.TransZ = -20
@@ -1494,79 +1455,14 @@
     End Select
 End Sub
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
-
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
 
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
 
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-'Set position as bumperX and Vol manually.
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
-'                     Supporting Ball & Sound Functions
-'*********************************************************************
-
-Function RndNum(min, max)
-    RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
-  Dim tmp
-  tmp = tableobj.y * 2 / table1.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  tmp = tableobj.x * 2 / table1.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
+' *********************************************************************
+'                      Supporting Ball & Sound Functions
+' *********************************************************************
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-    Vol = Csng(BallVel(ball) ^1.5 / VolDiv)
+    Vol = Csng(BallVel(ball) ^1.5 / 100)
 End Function
 
 Function Vol2(ball1, ball2) ' Calculates the Volume of the sound based on the speed of two balls
@@ -1620,21 +1516,16 @@
     If UBound(BOT) = -1 Then Exit Sub
 
     ' play the rolling sound for each ball
-
     For b = 0 to UBound(BOT)
-      If BallVel(BOT(b) ) > 1 Then
-        rolling(b) = True
-        if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-        Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-        End If
-      Else
-        If rolling(b) = True Then
-          StopSound("fx_ballrolling" & b)
-          rolling(b) = False
+        If BallVel(BOT(b) ) > 1 AND BOT(b).z < 30 Then
+            rolling(b) = TRUE
+            PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0
+        Else
+            If rolling(b) = TRUE Then
+                StopSound("fx_ballrolling" & b)
+                rolling(b) = FALSE
+            End If
         End If
-      End If
     Next
 End Sub
 
@@ -1646,7 +1537,7 @@
 
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, Pan(ball1), 0, Pitch(ball1), 0, 0
 End Sub
 
 
@@ -1688,35 +1579,35 @@
 
 
 Sub Pins_Hit (idx)
-  PlaySound "pinhit_low", 0, Vol(ActiveBall)*VolPi, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "pinhit_low", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0
 End Sub
 
 
 Sub Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall)*VolGates, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Spinner_Spin
-  PlaySoundAtVol "fx_spinner", Spinner, VolSpin
+  PlaySound "fx_spinner",0,.25,0,0.25
 End Sub
 
 Sub Rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1727,7 +1618,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolPo, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -1736,9 +1627,9 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
@@ -1752,22 +1643,16 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, 1, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "flip_hit_2", 0, 1, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "flip_hit_3", 0, 1, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
 Sub RandomSoundHole()
     Select Case Int(Rnd * 3) + 1
-        Case 1:PlaySound "Hole1", 0, 0.4, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-        Case 2:PlaySound "Hole2", 0, 0.4, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-        Case 3:PlaySound "Hole4", 0, 0.4, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+        Case 1:PlaySound "Hole1", 0, 0.4, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+        Case 2:PlaySound "Hole2", 0, 0.4, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+        Case 3:PlaySound "Hole4", 0, 0.4, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
     End Select
 End Sub
-
-' Thalamus : Exit in a clean and proper way
-Sub Table1_exit
-  Controller.Pause = False
-  Controller.Stop
-End Sub
