--- vpxtable_scripts/Earth Wind Fire (Zaccaria 1981).2.2.vbs
+++ todo_extracted_tables/Earth Wind Fire (Zaccaria 1981).2.2.vbs
@@ -3156,11 +3156,10 @@
     tmp =  - 7000
   End If
 
-' Thalamus, AudioFade patched
   If tmp > 0 Then
-    AudioFade = CSng(tmp ^ 5) 'was 10
+    AudioFade = CSng(tmp ^ 10)
   Else
-    AudioFade = CSng( - (( - tmp) ^ 5) ) 'was 10
+    AudioFade = CSng( - (( - tmp) ^ 10) )
   End If
 End Function
 
@@ -3174,11 +3173,10 @@
     tmp =  - 7000
   End If
 
-' Thalamus, AudioPan patched
   If tmp > 0 Then
-    AudioPan = CSng(tmp ^ 5) 'was 10
+    AudioPan = CSng(tmp ^ 10)
   Else
-    AudioPan = CSng( - (( - tmp) ^ 5) ) 'was 10
+    AudioPan = CSng( - (( - tmp) ^ 10) )
   End If
 End Function
 
