--- vpxtable_scripts/Gay 90's (Williams 1970) B2S MOD v2.vbs
+++ todo_extracted_tables/Gay 90's (Williams 1970) B2S MOD v2.vbs
@@ -1,32 +1,7 @@
 
 ' Thalamus 2018-07-23
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
-' Thalamus 2018-11-01 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-Randomize
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolKick   = 1    ' Kicker volume.
-Const VolSpin   = 1.5  ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
-
+' No special SSF tweaks yet.
 
 Dim Score(4)
 dim truesc(4)
@@ -230,7 +205,7 @@
   End If
 
     if keycode = AddCreditKey then
-  playsoundAtVol "coin3", drain, 1
+  playsound "coin3"
   coindelay.enabled=true
     credit=credit+4
   savehs
@@ -240,7 +215,7 @@
   end if
 
     if keycode = 5 then
-  playsoundAtVol "coin3", drain, 1
+  playsound "coin3"
   coindelay.enabled=true
   If B2SOn Then
     Controller.B2SSetCredits Credit
@@ -264,7 +239,7 @@
     pno3.state=lightstateoff
     pno4.state=lightstateoff
     play(currpl).state=lightstateon
-    playsound "Switch"
+    playsound "click"
     playsound "initialize"
     rst=0
     ballinplay=1
@@ -295,7 +270,7 @@
     if playno=2 then pno2.state=lightstateon
     if playno=3 then pno3.state=lightstateon
     if playno=4 then pno4.state=lightstateon
-    playsound "Switch"
+    playsound "click"
     If B2SOn Then
       Controller.B2SSetCredits Credit
       Controller.B2SSetCanPlay playno
@@ -306,14 +281,14 @@
 
   If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToEnd
-    PlaySoundAtVol "FlipperUp", LeftFlipper, VolFlip
-        playsoundAtVol "buzz", LeftFlipper, VolFlip
+    PlaySound "FlipperUp"
+        playsound "buzz"
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToEnd
-        PlaySoundAtVol "FlipperUp", RightFlipper, VolFlip
-        playsoundAtVol "buzz", RightFlipper, VolFlip
+        PlaySound "FlipperUp"
+        playsound "buzz"
   End If
 
   If keycode = LeftTiltKey Then
@@ -343,19 +318,19 @@
 
   If keycode = PlungerKey Then
     Plunger.Fire
-    playsoundAtVol "plunger", Plunger, 1
+    playsound "plunger"
   End If
 
   If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToStart
     stopsound "buzz"
-    if state=true and tilt=false then PlaySoundAtVol "FlipperDown", LeftFlipper, VolFlip
+    if state=true and tilt=false then PlaySound "FlipperDown"
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToStart
     stopsound "buzz"
-        if state=true and tilt=false then PlaySoundAtVol "FlipperDown", RightFlipper, VolFlip
+        if state=true and tilt=false then PlaySound "FlipperDown"
   End If
 
 End Sub
@@ -368,7 +343,7 @@
 
 
 sub coindelay_timer
-    playsound "Switch"
+    playsound "click"
     credit=credit+1
   credittxt.text=credit
     coindelay.enabled=false
@@ -488,7 +463,7 @@
     Controller.B2SSetCredits Credit
   end if
     rep(currpl)=1
-    playsound "Switch"
+    playsound "click"
     end if
     if score(currpl)=>replay2 and rep(currpl)=1 then
     credit=credit+1
@@ -498,7 +473,7 @@
     Controller.B2SSetCredits Credit
   end if
     rep(currpl)=2
-    playsound "Switch"
+    playsound "click"
     end if
     if score(currpl)=>replay3 and rep(currpl)=2 then
     credit=credit+1
@@ -508,7 +483,7 @@
     Controller.B2SSetCredits Credit
   end if
     rep(currpl)=3
-    playsound "Switch"
+    playsound "click"
     end if
 end sub
 
@@ -574,7 +549,7 @@
 
 
 Sub Drain_Hit()
-  playsoundAtVol "drainshorter", drain, 1
+  playsound "drainshorter"
   Drain.DestroyBall
     wv=0
     wheelcheck
@@ -772,7 +747,7 @@
   end if
     playsound "knocke"
     credittxt.text= credit
-    playsound "Switch"
+    playsound "click"
     end if
     next
 end sub
@@ -858,7 +833,7 @@
 End Sub
 
 sub bumper1_hit
-    if tilt=false then playsoundAtVol "jet1", bumper1, VolBump
+    if tilt=false then playsound "jet1"
     addscore 10
   me.timerenabled=1
 end sub
@@ -878,7 +853,7 @@
 End sub
 
 sub bumper2_hit
-    if tilt=false then playsoundAtVol "jet1", bumper2, VolBump
+    if tilt=false then playsound "jet1"
     addscore 10
   me.timerenabled=1
 end sub
@@ -898,7 +873,7 @@
 End sub
 
 sub bumper3_hit
-    if tilt=false then playsoundAtVol "jet1", bumper3, VolBump
+    if tilt=false then playsound "jet1"
     addscore 10
   me.timerenabled=1
 end sub
@@ -918,14 +893,14 @@
 End sub
 
 sub bumper4_hit
-    if tilt=false then playsoundAtVol "buzz", Bumper4, VolBump
+    if tilt=false then playsound "buzz"
     addscore 10
     wv=wv+1
     wheelcheck
 end sub
 
 sub bumper5_hit
-    if tilt=false then playsoundAtVol "buzz", Bumper5, VolBump
+    if tilt=false then playsound "buzz"
     addscore 10
     wv=wv+1
     wheelcheck
@@ -939,7 +914,7 @@
 sub Kicker1_Hit()
     if tilt=false then
     holecheck
-    playsoundAtVol "motorshort1s", Kicker1, VolKick
+    playsound "motorshort1s"
     end if
   kickstep=1
     me.timerenabled=1
@@ -948,7 +923,7 @@
 sub Kicker1_timer
   Select Case kickstep
     Case 4:
-    playsoundAtVol "kickerkick", Pkickarm, VolKick
+    playsound "kickerkick"
     Pkickarm.rotz=10
     kicker1.kick 240,10
     Case 5:
@@ -959,14 +934,14 @@
 end Sub
 
 Sub Kicker2_Hit()
-    playsoundAtVol "motorshort1s", Kicker2, VolKick
+    playsound "motorshort1s"
     me.timerenabled=1
     loutl.State=LightStateOff
     Flipper1Timer.enabled=True
 End Sub
 
 sub kicker2_timer
-    playsoundAtVol "kickerkick", Kicker2, VolKick
+    playsound "kickerkick"
     Kicker2.kick 0, 20
     me.timerenabled=0
 end sub
@@ -1080,7 +1055,7 @@
 Dim RStep, Lstep
 
 Sub RightSlingShot_Slingshot
-    PlaySoundAtVol "right_slingshot", sling1, 1
+    PlaySound "right_slingshot", 0, 1, 0.05, 0.05
     RSling.Visible = 0
     RSling1.Visible = 1
     sling1.TransZ = -20
@@ -1098,7 +1073,7 @@
 End Sub
 
 Sub LeftSlingShot_Slingshot
-    PlaySoundAtVol "left_slingshot", sling2, 1
+    PlaySound "left_slingshot",0,1,-0.05,0.05
     LSling.Visible = 0
     LSling1.Visible = 1
     sling2.TransZ = -20
@@ -1147,7 +1122,7 @@
 
 sub ballrel_hit
     if ballrelenabled=1 then
-    playsoundAtVol "launchball", ballrel, 1
+    playsound "launchball"
     ballrelenabled=0
     end if
 end sub
@@ -1411,38 +1386,38 @@
 ' depending of the speed of the collision.
 
 Sub a_Pins_Hit (idx)
-  PlaySound "pinhit_low", 0, Vol(ActiveBall)*VolPi, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "pinhit_low", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Targets_Hit (idx)
-  PlaySound "target", 0, Vol(ActiveBall)*VolTarg, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "target", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall)*VolGates, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub a_Spinner_Spin
-  PlaySoundAtVol "fx_spinner", a_Spinner, VolSpin
+  PlaySound "fx_spinner",0,.25,0,0.25
 End Sub
 
 Sub a_Rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1453,7 +1428,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1464,7 +1439,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolPo, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -1473,9 +1448,9 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -1489,9 +1464,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -1527,8 +1502,8 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volum)
-  PlaySound sound, 1, Volum, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
@@ -1588,7 +1563,7 @@
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -1651,13 +1626,10 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
-End Sub
-
-
-' Thalamus : Exit in a clean and proper way
-Sub table1_exit()
-  Controller.Pause = False
-  Controller.Stop
+  If Table1.VersionMinor > 3 OR Table1.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
 
