--- vpxtable_scripts/Goin' Nuts (Gottlieb 1983).vbs
+++ todo_extracted_tables/Goin' Nuts (Gottlieb 1983).vbs
@@ -1091,11 +1091,11 @@
     ' render the shadow for each ball
     For b = 0 to UBound(BOT)
         If BOT(b).X < Table1.Width/2 Then
-            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) + 6
+            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) + 6
         Else
-            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) - 6
+            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) - 6
         End If
-        ballShadow(b).Y = BOT(b).Y + 4
+        ballShadow(b).Y = BOT(b).Y + 12
         If BOT(b).Z > 20 Then
             BallShadow(b).visible = 1
         Else
@@ -1370,10 +1370,7 @@
     RealCOR = BallSpeed(aBall) / cor.ballvel(aBall.id)
     coef = desiredcor / realcor
 
-' Thalamus - patched : ' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
   End Sub
 
   Public Sub CopyCoef(aObj, aCoef) 'alternative addpoints, copy with coef
