--- vpxtable_scripts/Gold Ball (Bally 1983) 1.0.vbs
+++ todo_extracted_tables/Gold Ball (Bally 1983) 1.0.vbs
@@ -266,8 +266,6 @@
   Set gsTrough=new cvpmBallStack
   gsTrough.InitSaucer GoldKick,30,62,5.5
 ' gsTrough.InitExitSnd "fx_kicker", "Solenoid"
-  gsTrough.KickForceVar = 3
-  gsTrough.KickAngleVar = 3
   gsTrough.AddBall 0
   GoldKick.CreateBall.Image="Goldball"
 
@@ -1590,10 +1588,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched : ' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     if debugOn then TBPout.text = str
   End Sub
 
