--- vpxtable_scripts/Halloween 1978-1981 (Original 2022) 1.03.vbs
+++ todo_extracted_tables/Halloween 1978-1981 (Original 2022) 1.03.vbs
@@ -3403,8 +3403,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     if debugOn then TBPout.text = str
   End Sub
 
@@ -3414,8 +3413,7 @@
     RealCOR = BallSpeed(aBall) / (cor.ballvel(aBall.id)+0.0001)
     coef = desiredcor / realcor
     If abs(aball.velx) < 2 and aball.vely < 0 and aball.vely > -3.75 then
-' Thalamus - patched :       aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+      aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     End If
   End Sub
 
