--- vpxtable_scripts/harley quinn (HauntFreaks 2017).vbs
+++ todo_extracted_tables/harley quinn (HauntFreaks 2017).vbs
@@ -25,41 +25,15 @@
 ' RGB LED effects scripted by Ninuzzu
 ' Drop Target lighting by BorgDog
 '*********************************************************************************************************
-Option Explicit
-Randomize
 
 ' Thalamus 2018-07-23
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
 ' Changed UseSolenoids=1 to 2
-' Thalamus 2018-12-18 : Added FFv2
-' Thalamus 2018-11-01 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolRol    = 1    ' Rollovers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRB     = 1    ' Rubber bands volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolPlast  = 1    ' Plastics volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolWood   = 1    ' Woods volume.
-Const VolKick   = 1    ' Kicker volume.
-Const VolSpin   = 1.5  ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
+' No special SSF tweaks yet.
+' This is a JP table. He often uses walls as switches so I need to be careful of using PlaySoundAt
+
+Option Explicit
+Randomize
 
 'optReset = True  'Uncomment to reset to default options in case of error or keep all changes temporary. When done recomment again!
 
@@ -70,14 +44,10 @@
 
 LoadVPM "01210000", "sys80.vbs", 3.1
 
-' Thalamus - for Fast Flip v2
-NoUpperRightFlipper
-NoUpperLeftFlipper
-
 Dim bsTrough, dtLBank, dtRBank, dtMBank, dtCBank, bsTop, kickbackIM
 Dim DesktopMode:DesktopMode=Table1.ShowDT
 
-Const UseSolenoids = 2
+Const UseSolenoids = 1
 Const UseLamps = 0
 Const UseGI = 0
 Const UseSync = 0
@@ -94,10 +64,6 @@
 Const cGameName = "diamond"
 
 Sub table1_Init
-
-' Thalamus : Was missing 'vpminit me'
-  vpminit me
-
   Dim xx
     For each xx in aReels:xx.Visible = DesktopMode:Next
     lrail.Visible = DesktopMode
@@ -195,14 +161,14 @@
     If keycode = LeftTiltKey Then Nudge 90, 5:PlaySound SoundFX("fx_nudge", 0), 0, 1, -0.1, 0.25
     If keycode = RightTiltKey Then Nudge 270, 5:PlaySound SoundFX("fx_nudge", 0), 0, 1, 0.1, 0.25
     If keycode = CenterTiltKey Then Nudge 0, 6:PlaySound SoundFX("fx_nudge", 0), 0, 1, 0, 0.25
-    If keycode = PlungerKey Then PlaySoundAtVol "fx_PlungerPull",Plunger,1:Plunger.Pullback
+    If keycode = PlungerKey Then PlaySound "fx_PlungerPull", 0, 1, 0.1, 0.25:Plunger.Pullback
     If vpmKeyDown(keycode)Then Exit Sub
     If keycode = KeyRules Then Rules
 End Sub
 
 Sub table1_KeyUp(ByVal Keycode)
     If KeyCode = RightFlipperKey Then Controller.Switch(72) = 0
-    If keycode = PlungerKey Then PlaySoundAtVol "fx_plunger",Plunger,1:Plunger.Fire
+    If keycode = PlungerKey Then PlaySound "fx_plunger", 0, 1, 0.1, 0.25:Plunger.Fire
     If vpmKeyUp(keycode)Then Exit Sub
 End Sub
 
@@ -214,7 +180,7 @@
 Dim LStep, RStep
 
 Sub LeftSlingShot_Slingshot
-    PlaySoundAtVol SoundFX("fx_slingshot", DOFContactors), Lemk, 1
+    PlaySound SoundFX("fx_slingshot", DOFContactors), 0, 1, -0.05, 0.05
     DOF 103, DOFPulse
     LeftSling4.Visible = 1
     Lemk.RotX = 26
@@ -233,7 +199,7 @@
 End Sub
 
 Sub RightSlingShot_Slingshot
-    PlaySoundAtVol SoundFX("fx_slingshot", DOFContactors), Remk, 1
+    PlaySound SoundFX("fx_slingshot", DOFContactors), 0, 1, 0.05, 0.05
     DOF 104, DOFPulse
     RightSling4.Visible = 1
     Remk.RotX = 26
@@ -252,40 +218,40 @@
 End Sub
 
 ' Bumpers
-Sub Bumper1_Hit:vpmTimer.PulseSw 61:PlaySoundAtVol SoundFX("fx_bumper",DOFContactors),Bumper1,VolBump:DOF 105, DOFPulse:End Sub
-Sub Bumper2_Hit:vpmTimer.PulseSw 71:PlaySoundAtVol SoundFX("fx_bumper",DOFContactors),Bumper2,VolBump:DOF 106, DOFPulse:End Sub
-Sub Bumper3_Hit:vpmTimer.PulseSw 61:PlaySoundAtVol SoundFX("fx_bumper",DOFContactors),Bumper3,VolBump:DOF 107, DOFPulse:End Sub
+Sub Bumper1_Hit:vpmTimer.PulseSw 61:PlaySound SoundFX("fx_bumper",DOFContactors), 0, 1, 0, 0.1:DOF 105, DOFPulse:End Sub
+Sub Bumper2_Hit:vpmTimer.PulseSw 71:PlaySound SoundFX("fx_bumper",DOFContactors), 0, 1, 0, 0.1:End Sub
+Sub Bumper3_Hit:vpmTimer.PulseSw 61:PlaySound SoundFX("fx_bumper",DOFContactors), 0, 1, 0, 0.1:DOF 107, DOFPulse:End Sub
 
 ' Drain holes
-Sub Drain_Hit:PlaysoundAtVol "fx_drain",drain,1:bsTrough.AddBall Me:End Sub
+Sub Drain_Hit:Playsound "fx_drain":bsTrough.AddBall Me:End Sub
 
 ' Rollovers
-Sub sw43_Hit:Controller.Switch(43) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw43_Hit:Controller.Switch(43) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw43_UnHit:Controller.Switch(43) = 0:End Sub
 
-Sub sw53_Hit:Controller.Switch(53) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw53_Hit:Controller.Switch(53) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw53_UnHit:Controller.Switch(53) = 0:End Sub
 
-Sub sw63_Hit:Controller.Switch(63) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw63_Hit:Controller.Switch(63) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw63_UnHit:Controller.Switch(63) = 0:End Sub
 
-Sub sw73_Hit:Controller.Switch(73) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw73_Hit:Controller.Switch(73) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw73_UnHit:Controller.Switch(73) = 0:End Sub
 
-Sub sw65_Hit:Controller.Switch(65) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw65_Hit:Controller.Switch(65) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw65_UnHit:Controller.Switch(65) = 0:End Sub
 
-Sub sw70_Hit:Controller.Switch(70) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw70_Hit:Controller.Switch(70) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw70_UnHit:Controller.Switch(70) = 0:End Sub
 
-Sub sw45_Hit:Controller.Switch(45) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw45_Hit:Controller.Switch(45) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw45_UnHit:Controller.Switch(45) = 0:End Sub
 
-Sub sw55_Hit:Controller.Switch(55) = 1:PlaySoundAtVol "fx_sensor",ActiveBall,1:End Sub
+Sub sw55_Hit:Controller.Switch(55) = 1:PlaySound "fx_sensor", 0, 1, pan(ActiveBall):End Sub
 Sub sw55_UnHit:Controller.Switch(55) = 0:End Sub
 
 
-Sub Trigger1_Hit:Controller.Switch(70) = 1:PlaySoundAtVol "fx_ballrampdrop",ActiveBall,1:End Sub
+Sub Trigger1_Hit:Controller.Switch(70) = 1:PlaySound "fx_ballrampdrop", 0, 1, pan(ActiveBall):End Sub
 Sub Trigger1_UnHit:Controller.Switch(70) = 0:End Sub
 
 ' Ramp Switches
@@ -293,37 +259,42 @@
 Sub sw44_UnHit:Controller.Switch(44) = 0:End Sub
 
 'Spinners
-Sub sw64_Spin:vpmTimer.PulseSw 64:PlaySoundAtVol "fx_spinner",sw64,VolSpin:End Sub
-Sub sw54_Spin:vpmTimer.PulseSw 54:PlaySoundAtVol "fx_spinner",sw54,VolSpin:End Sub
+Sub sw64_Spin:vpmTimer.PulseSw 64:PlaySound "fx_spinner", 0, 1, 0.1:End Sub
+Sub sw54_Spin:vpmTimer.PulseSw 54:PlaySound "fx_spinner", 0, 1, -0.1:End Sub
 
 ' Droptargets
-Sub sw20_Hit:dtLBank.hit 1:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw30_Hit:dtLBank.hit 2:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw40_Hit:dtLBank.hit 3:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw50_Hit:dtLBank.hit 4:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw60_Hit:dtLBank.hit 5:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw22_Hit:dtRBank.hit 1:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw32_Hit:dtRBank.hit 2:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw42_Hit:dtRBank.hit 3:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw52_Hit:dtRBank.hit 4:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw62_Hit:dtRBank.hit 5:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw21_Hit:dtMBank.hit 1:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw31_Hit:dtMBank.hit 2:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw41_Hit:dtMBank.hit 3:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw51_Hit:dtMBank.hit 4:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
-Sub sw23_Hit:dtCBank.hit 1:PlaySoundAtVol SoundFX("fx_droptarget", DOFDropTargets),ActiveBall,VolTarg:End Sub
+Sub sw20_Hit:dtLBank.hit 1:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw30_Hit:dtLBank.hit 2:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw40_Hit:dtLBank.hit 3:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw50_Hit:dtLBank.hit 4:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw60_Hit:dtLBank.hit 5:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw22_Hit:dtRBank.hit 1:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw32_Hit:dtRBank.hit 2:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw42_Hit:dtRBank.hit 3:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw52_Hit:dtRBank.hit 4:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw62_Hit:dtRBank.hit 5:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw21_Hit:dtMBank.hit 1:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw31_Hit:dtMBank.hit 2:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw41_Hit:dtMBank.hit 3:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw51_Hit:dtMBank.hit 4:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
+Sub sw23_Hit:dtCBank.hit 1:PlaySound SoundFX("fx_droptarget", DOFDropTargets), 0, 1, pan(ActiveBall):End Sub
 
 ' Targets
-Sub sw24_Hit:vpmTimer.PulseSw 24:PlaySoundAtVol SoundFX("fx_target", DOFTargets),ActiveBall,VolTarg:End Sub
-Sub sw34_Hit:vpmTimer.PulseSw 34:PlaySoundAtVol SoundFX("fx_target", DOFTargets),ActiveBall,VolTarg:End Sub
-Sub sw26_Hit:vpmTimer.PulseSw 26:PlaySoundAtVol SoundFX("fx_target", DOFTargets),ActiveBall,VolTarg:End Sub
-Sub sw36_Hit:vpmTimer.PulseSw 36:PlaySoundAtVol SoundFX("fx_target", DOFTargets),ActiveBall,VolTarg:End Sub
-Sub sw25_Hit:vpmTimer.PulseSw 25:PlaySoundAtVol SoundFX("fx_target", DOFTargets),ActiveBall,VolTarg:End Sub
-Sub sw35_Hit:vpmTimer.PulseSw 35:PlaySoundAtVol SoundFX("fx_target", DOFTargets),ActiveBall,VolTarg:End Sub
+Sub sw24_Hit:vpmTimer.PulseSw 24:PlaySound SoundFX("fx_target", DOFTargets), 0, 1, pan(ActiveBall):End Sub
+
+Sub sw34_Hit:vpmTimer.PulseSw 34:PlaySound SoundFX("fx_target", DOFTargets), 0, 1, pan(ActiveBall):End Sub
+
+Sub sw26_Hit:vpmTimer.PulseSw 26:PlaySound SoundFX("fx_target", DOFTargets), 0, 1, pan(ActiveBall):End Sub
+
+Sub sw36_Hit:vpmTimer.PulseSw 36:PlaySound SoundFX("fx_target", DOFTargets), 0, 1, pan(ActiveBall):End Sub
+
+Sub sw25_Hit:vpmTimer.PulseSw 25:PlaySound SoundFX("fx_target", DOFTargets), 0, 1, pan(ActiveBall):End Sub
+
+Sub sw35_Hit:vpmTimer.PulseSw 35:PlaySound SoundFX("fx_target", DOFTargets), 0, 1, pan(ActiveBall):End Sub
 
 ' VUK & Holes
 
-Sub sw74_Hit:PlaySoundAtVol "fx_kicker_enter",sw74,VolKick:bsTop.AddBall 0:End Sub
+Sub sw74_Hit:PlaySound "fx_kicker_enter", 0, 1, 0.01:bsTop.AddBall 0:End Sub
 
 '********************
 'Solenoids & Flashers
@@ -348,13 +319,13 @@
 
 Sub SolLFlipper(Enabled)
     If Enabled Then
-        PlaySoundAtVol SoundFX("fx_flipperup", DOFFlippers), LeftFlipper1, VolFlip
+        PlaySound SoundFX("fx_flipperup", DOFFlippers), 0, 1, -0.1, 0.1
         LeftFlipper1.RotateToEnd
         LeftFlipper2.RotateToEnd
         LeftFlipper3.RotateToEnd
         LeftFlipper4.RotateToEnd
     Else
-        PlaySoundAtVol SoundFX("fx_flipperdown", DOFFlippers), LeftFlipper1, VolFlip
+        PlaySound SoundFX("fx_flipperdown", DOFFlippers), 0, 1, -0.1, 0.1
         LeftFlipper1.RotateToStart
         LeftFlipper2.RotateToStart
         LeftFlipper3.RotateToStart
@@ -364,13 +335,13 @@
 
 Sub SolRFlipper(Enabled)
     If Enabled Then
-        PlaySoundAtVol SoundFX("fx_flipperup", DOFFlippers), RightFlipper1, VolFlip
+        PlaySound SoundFX("fx_flipperup", DOFFlippers), 0, 1, 0.1, 0.1
         RightFlipper1.RotateToEnd
         RightFlipper2.RotateToEnd
         RightFlipper3.RotateToEnd
         RightFlipper4.RotateToEnd
     Else
-        PlaySoundAtVol SoundFX("fx_flipperdown", DOFFlippers), RightFlipper1, VolFlip
+        PlaySound SoundFX("fx_flipperdown", DOFFlippers), 0, 1, 0.1, 0.1
         RightFlipper1.RotateToStart
         RightFlipper2.RotateToStart
         RightFlipper3.RotateToStart
@@ -901,23 +872,23 @@
 ' Diverse Collection Hit Sounds
 '******************************
 
-Sub aMetals_Hit(idx):PlaySound "fx_MetalHit2", 0, Vol(ActiveBall)*VolMetal, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aRubber_Bands_Hit(idx):PlaySound "fx_rubber_band", 0, Vol(ActiveBall)*VolRB, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aRubber_Posts_Hit(idx):PlaySound "fx_rubber_post", 0, Vol(ActiveBall)*VolPo, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aRubber_Pins_Hit(idx):PlaySound "fx_rubber_pin", 0, Vol(ActiveBall)*VolPi, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aPlastics_Hit(idx):PlaySound "fx_PlasticHit", 0, Vol(ActiveBall)*VolPlast, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aGates_Hit(idx):PlaySound "fx_Gate", 0, Vol(ActiveBall)*VolGates, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
-Sub aWoods_Hit(idx):PlaySound "fx_Woodhit", 0, Vol(ActiveBall)*VolWood, pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aMetals_Hit(idx):PlaySound "fx_MetalHit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aRubber_Bands_Hit(idx):PlaySound "fx_rubber_band", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aRubber_Posts_Hit(idx):PlaySound "fx_rubber_post", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aRubber_Pins_Hit(idx):PlaySound "fx_rubber_pin", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aPlastics_Hit(idx):PlaySound "fx_PlasticHit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aGates_Hit(idx):PlaySound "fx_Gate", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
+Sub aWoods_Hit(idx):PlaySound "fx_Woodhit", 0, Vol(ActiveBall), pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall):End Sub
 
 ' Ramp Soundss
 Sub RHelp1_Hit()
     'StopSound "fx_metalrolling"
-    PlaySoundAtVol "fx_ballrampdrop", ActiveBall, 1
+    PlaySound "fx_ballrampdrop", 0, 1, pan(ActiveBall)
 End Sub
 
 Sub RHelp2_Hit()
     'StopSound "fx_metalrolling"
-    PlaySoundAtVol "fx_ballrampdrop", ActiveBall, 1
+    PlaySound "fx_ballrampdrop", 0, 1, pan(ActiveBall)
 End Sub
 
 Function RndNum(min,max)
@@ -1347,8 +1318,8 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volum)
-  PlaySound sound, 1, Volum, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
@@ -1408,7 +1379,7 @@
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -1471,7 +1442,11 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  If Table1.VersionMinor > 3 OR Table1.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
 
 ' Thalamus : Exit in a clean and proper way
