--- vpxtable_scripts/Harley Davidson (Sega 1999)MOD1.72.vbs
+++ todo_extracted_tables/Harley Davidson (Sega 1999)MOD1.72.vbs
@@ -301,28 +301,20 @@
     bsCycle.InitSw 0,44,43,42,41,0,0,0
     bsCycle.InitKick Kicker1,180,5
     bsCycle.InitExitSnd SoundFX("ballrelease",DOFContactors), SoundFX("Solenoid",DOFContactors)
-    bsCycle.KickForceVar = 3
-    bsCycle.KickAngleVar = 3
 
     Set bsScoop=New cvpmBallStack
     bsScoop.InitSw 0,46,0,0,0,0,0,0
     bsScoop.InitKick sw46,175,20
     bsScoop.InitExitSnd SoundFX("Popper",DOFContactors), SoundFX("Solenoid",DOFContactors)
-    bsScoop.KickForceVar = 3
-    bsScoop.KickAngleVar = 3
 
     Set bsLaunch=New cvpmBallStack
   bsLaunch.InitSaucer Kicker2,16,0,44
   bsLaunch.InitExitSnd SoundFX("solenoid",DOFContactors), SoundFX("Solenoid",DOFContactors)
-  bsLaunch.KickForceVar = 3
-  bsLaunch.KickAngleVar = 3
 
     Set bsSVUK=New cvpmBallStack
     bsSVUK.InitSw 0,45,0,0,0,0,0,0
     bsSVUK.InitKick sw45a,105,7
     bsSVUK.InitExitSnd SoundFX("ballrelease",DOFContactors), SoundFX("scoopexit",DOFContactors)
-    bsSVUK.KickForceVar = 3
-    bsSVUK.KickAngleVar = 3
 
 ' Set dtL=New cvpmDropTarget
 '   dtL.InitDrop Array(sw17,sw18,sw19,sw20),Array(17,18,19,20)
@@ -453,7 +445,7 @@
     FlipperActivate RightFlipper, RFPress
   End If
 
-  If KeyCode=PlungerKey or KeyCode=LockBarKey Then Controller.Switch(53)=1
+  If KeyCode=PlungerKey Then Controller.Switch(53)=1
   If KeyCode=LeftMagnaSave Then
     Controller.Switch(8)=1
     if DisableLUTSelector = 0 then
@@ -506,7 +498,7 @@
 '   SolRFlipper False           'This would be called by the solenoid callbacks if using a ROM
   End If
 
-  If KeyCode=PlungerKey or KeyCode=LockBarKey Then Controller.Switch(53)=0
+  If KeyCode=PlungerKey Then Controller.Switch(53)=0
   If KeyCode=LeftMagnaSave Then
     Controller.Switch(8)=0
   End If
@@ -1390,11 +1382,11 @@
     ' render the shadow for each ball
     For b = 0 to UBound(BOT)
         If BOT(b).X < Table1.Width/2 Then
-            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) + 6
+            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) + 6
         Else
-            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) - 6
+            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) - 6
         End If
-        ballShadow(b).Y = BOT(b).Y + 4
+        ballShadow(b).Y = BOT(b).Y + 12
         If BOT(b).Z > 20 Then
             BallShadow(b).visible = 1
         Else
@@ -2554,8 +2546,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     if debugOn then TBPout.text = str
   End Sub
 
