--- vpxtable_scripts/Hot Rod (Gottlieb 1977).vbs
+++ todo_extracted_tables/Hot Rod (Gottlieb 1977).vbs
@@ -304,14 +304,14 @@
 
   If keycode = LeftFlipperKey and InProgress=true and TableTilted=false Then
     LeftFlipper.RotateToEnd
-    PlaySoundAtVol SoundFXDOF("FlipperUp",101,DOFOn,DOFFlippers), LeftFlipper, 1
-    PlayLoopSoundAtVol "buzzL", LeftFlipper, 1
+    PlaySound SoundFXDOF("FlipperUp",101,DOFOn,DOFFlippers)
+    PlaySound "buzzL",-1
   End If
 
   If keycode = RightFlipperKey  and InProgress=true and TableTilted=false Then
     RightFlipper.RotateToEnd
-    PlaySoundAtVol SoundFXDOF("FlipperUp",102,DOFOn,DOFFlippers), RightFlipper, 1
-    PlayLoopSoundAtVol "buzz", RightFlipper, 1
+    PlaySound SoundFXDOF("FlipperUp",102,DOFOn,DOFFlippers)
+    PlaySound "buzz",-1
   End If
 
   If keycode = LeftTiltKey Then
@@ -449,7 +449,7 @@
       exit sub
     end if
 
-    PlaySoundAtVol "plungerrelease", Plunger, 1
+    PlaySound"plungerrelease"
     Plunger.Fire
   End If
 
@@ -461,13 +461,13 @@
 
   If keycode = LeftFlipperKey and InProgress=true and TableTilted=false Then
     LeftFlipper.RotateToStart
-    PlaySoundAtVol SoundFXDOF("FlipperDown",101,DOFOff,DOFFlippers), LeftFlipper, 1
+    PlaySound SoundFXDOF("FlipperDown",101,DOFOff,DOFFlippers)
     StopSound "buzzL"
   End If
 
   If keycode = RightFlipperKey and InProgress=true and TableTilted=false Then
     RightFlipper.RotateToStart
-    PlaySoundAtVol SoundFXDOF("FlipperDown",102,DOFOff,DOFFlippers), RightFlipper, 1
+    PlaySound SoundFXDOF("FlipperDown",102,DOFOff,DOFFlippers)
     StopSound "buzz"
   End If
 
@@ -476,7 +476,7 @@
 Sub Drain_Hit()
   DOF 122, DOFPulse
   Drain.DestroyBall
-  PlaySoundAtVol "fx_drain", Drain, 1
+  PlaySound "fx_drain"
   Pause4Bonustimer.enabled=1
 
 End Sub
@@ -504,7 +504,7 @@
 '
 
 Sub RightSlingShot_Slingshot
-    PlaySoundAtVol SoundFXDOF("right_slingshot",110,DOFPulse,DOFContactors), sling1, 1
+    PlaySound SoundFXDOF("right_slingshot",110,DOFPulse,DOFContactors), 0, 1, 0.05, 0.05
   DOF 112, DOFPulse
     RSling0.Visible = 0
     RSling1.Visible = 1
@@ -523,7 +523,7 @@
 End Sub
 
 Sub LeftSlingShot_Slingshot
-    PlaySoundAtVol SoundFXDOF("left_slingshot",109,DOFPulse,DOFContactors), sling2, 1
+    PlaySound SoundFXDOF("left_slingshot",109,DOFPulse,DOFContactors),0,1,-0.05,0.05
   DOF 111, DOFPulse
     LSling0.Visible = 0
     LSling1.Visible = 1
@@ -563,7 +563,7 @@
 
 Sub Bumper1_Hit
   If TableTilted=false then
-    PlaySoundAtVol SoundFXDOF("bumper1",103,DOFPulse,DOFcontactors), ActiveBall, 1
+    PlaySound SoundFXDOF("bumper1",103,DOFPulse,DOFcontactors)
     DOF 106, DOFPulse
     bump1 = 1
     AddScore(100)
@@ -573,7 +573,7 @@
 
 Sub Bumper2_Hit
   If TableTilted=false then
-    PlaySoundAtVol SoundFXDOF("bumper1",104,DOFPulse,DOFcontactors), ActiveBall, 1
+    PlaySound SoundFXDOF("bumper1",104,DOFPulse,DOFcontactors)
     DOF 107, DOFPulse
     bump3 = 1
     If BallsPerGame=5 then
@@ -587,7 +587,7 @@
 
 Sub Bumper3_Hit
   If TableTilted=false then
-    PlaySoundAtVol SoundFXDOF("bumper1",105,DOFPulse,DOFcontactors), ActiveBall, 1
+    PlaySound SoundFXDOF("bumper1",105,DOFPulse,DOFcontactors)
     DOF 108, DOFPulse
     bump2 = 1
     AddScore(100)
@@ -599,7 +599,7 @@
 
 Sub TriggerTopA_Hit()
   If TableTilted=false then
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     DOF 120, DOFPulse
     SetMotor(300)
     TopLightA.state=0
@@ -617,7 +617,7 @@
 Sub TriggerTopB_Hit()
   If TableTilted=false then
     DOF 119, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     SetMotor(300)
     TopLightB.state=0
     for each obj in LightsB
@@ -630,7 +630,7 @@
 
 Sub TriggerTopC_Hit()
   If TableTilted=false then
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     DOF 121, DOFPulse
     SetMotor(300)
     TopLightC.state=0
@@ -645,7 +645,7 @@
 Sub TriggerUpperRightRollover_Hit()
   If TableTilted=false then
     DOF 134, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     AddScore(100)
     If RightExtraBallLight.state=1 then
       ExtraBall=True
@@ -661,7 +661,7 @@
 Sub TriggerUpperLeftRollover_Hit()
   If TableTilted=false then
     DOF 133, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     AddScore(100)
     If LeftExtraBallLight.state=1 then
       ExtraBall=True
@@ -678,7 +678,7 @@
 Sub TriggerLeftInlane_Hit()
   If TableTilted=false then
     DOF 128, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     SetMotor(300)
     for each obj in LightsB
       obj.state=0
@@ -692,7 +692,7 @@
 Sub TriggerRightInlane_Hit()
   If TableTilted=false then
     DOF 129, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     SetMotor(300)
 
     for each obj in LightsC
@@ -706,7 +706,7 @@
 Sub TriggerLeftOutlane_Hit()
   If TableTilted=false then
     DOF 127, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     SetMotor(300)
     for each obj in LightsA
       obj.state=0
@@ -719,7 +719,7 @@
 Sub TriggerRightOutlane_Hit()
   If TableTilted=false then
     DOF 130, DOFPulse
-    PlaySoundAtVol "sensor", ActiveBall, 1
+    PlaySound "sensor"
     SetMotor(300)
     for each obj in LightsA
       obj.state=0
@@ -788,7 +788,7 @@
   Kicker1.kick 135,10
   Pkickarm1.rotz=15
   KickerArm1.enabled=true
-  PlaySoundAtVol SoundFXDOF("saucer",131,DOFPulse,DOFcontactors), KickerArm1, 1
+  PlaySound SoundFXDOF("saucer",131,DOFPulse,DOFcontactors)
   DOF 118, DOFPulse
 end sub
 
@@ -838,7 +838,7 @@
   Kicker2.kick 230,10
   Pkickarm2.rotz=15
   KickerArm2.enabled=true
-  PlaySoundAtVol SoundFXDOF("saucer",132,DOFPulse,DOFcontactors), KickerArm2, 1
+  PlaySound SoundFXDOF("saucer",132,DOFPulse,DOFcontactors)
   DOF 118, DOFPulse
 end sub
 
@@ -858,7 +858,7 @@
 Sub Kicker3_Timer()
   Kicker3.TIMERENABLED = FALSE
   Kicker3.Kick 0,20
-  PlaySoundAtVol SoundFXDOF("saucer",124,DOFPulse,DOFcontactors), Kicker3, 1
+  PlaySound SoundFXDOF("saucer",124,DOFPulse,DOFcontactors)
   DOF 118, DOFPulse
 End Sub
 
@@ -873,7 +873,7 @@
 Sub Kicker4_Timer()
   Kicker4.TIMERENABLED = FALSE
   Kicker4.Kick 0,20
-  PlaySoundAtVol SoundFXDOF("saucer",125,DOFPulse,DOFcontactors), Kicker4, 1
+  PlaySound SoundFXDOF("saucer",125,DOFPulse,DOFcontactors)
   DOF 118, DOFPulse
 end Sub
 '*****************************************
@@ -2408,173 +2408,61 @@
 
 end sub
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
+'*********************************************************************
+'                 Positional Sound Playback Functions
+'*********************************************************************
 
 ' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
 ' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
 ' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
 Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
   PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
 End Sub
 
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
+' Similar subroutines that are less complicated to use (e.g. simply use standard parameters for the PlaySound call)
 Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+    PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
 Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
+    PlaySoundAt soundname, ActiveBall
 End Sub
 
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
 
 '*********************************************************************
-'            Supporting Ball, Sound Functions and Math
+'                     Supporting Ball & Sound Functions
 '*********************************************************************
 
-Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Const Pi = 3.1415927
-
-Function dSin(degrees)
-  dsin = sin(degrees * Pi/180)
-End Function
-
-Function dCos(degrees)
-  dcos = cos(degrees * Pi/180)
-End Function
-
 Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
   Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / table1.height-1
-  If tmp > 0 Then
+    tmp = tableobj.y * 2 / table1.height-1
+    If tmp > 0 Then
     AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
+    Else
+        AudioFade = Csng(-((- tmp) ^10) )
+    End If
 End Function
 
 Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / table1.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / table1.width-1
-  If tmp > 0 Then
-    Pan = Csng(tmp ^10)
-  Else
-    Pan = Csng(-((- tmp) ^10) )
-  End If
+    Dim tmp
+    tmp = tableobj.x * 2 / table1.width-1
+    If tmp > 0 Then
+        AudioPan = Csng(tmp ^10)
+    Else
+        AudioPan = Csng(-((- tmp) ^10) )
+    End If
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
-End Function
-
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-End Function
-
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
-End Function
-
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
+    Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
+    Pitch = BallVel(ball) * 20
 End Function
 
 Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
-End Function
-
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
+    BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
 End Function
 
 '*****************************************
