--- vpxtable_scripts/Humpty Dumpty (Gottlieb 1947)v3_0.vbs
+++ todo_extracted_tables/Humpty Dumpty (Gottlieb 1947)v3_0.vbs
@@ -38,29 +38,7 @@
 
 ' Thalamus 2018-07-23
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
-' Thalamus 2018-11-01 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolSpin   = 1.5  ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
-
+' No special SSF tweaks yet.
 
 Sub DOF(dofevent, dofstate)
   If B2SOn=True Then
@@ -268,7 +246,7 @@
   if keycode = 6 then
         if Credit < 26 then
         Credit = Credit + 1
-        PlaysoundAtVol "coin", drainer, 1
+        Playsound "coin"
         end if
     CredBox.Text = Credit
   end if
@@ -280,13 +258,13 @@
   end if
   If keycode = PlungerKey Then
     Plunger.PullBack
-PlaySoundAtVol "plungerpull", Plunger, 1
+PlaySound "plungerpull",0,1,0.25,0.25
   End If
     if (keycode =3 or keycode =RightMagnaSave) and gameon = TRUE and BallsOut<= 5 and hold=0 then
   drop.Rotz = 0
 DOF 104,2
        BallsOut=BallsOut+1
-       PlaySoundAtVol "balloutr", Plunger, 1
+       PlaySound "balloutr"
        Plungekick.createball.image="JPBall-Dark2"
        Plungekick.kick 270,20
        hold=1
@@ -297,18 +275,14 @@
     Flipper1.RotateToEnd
     Flipper2.RotateToEnd
     Flipper3.RotateToEnd
-    PlaySoundAtVol "FlipperUp", Flipper1, VolFlip
-    PlaySoundAtVol "FlipperUp", Flipper2, VolFlip
-    PlaySoundAtVol "FlipperUp", Flipper3, VolFlip
+    PlaySound "FlipperUp"
   End If
   If keycode = RightFlipperKey and tilted = FALSE then
 DOF 102,1
     Flipper4.RotateToEnd
     Flipper5.RotateToEnd
     Flipper6.RotateToEnd
-    PlaySoundAtVol "FlipperUp", Flipper4, VolFlip
-    PlaySoundAtVol "FlipperUp", Flipper5, VolFlip
-    PlaySoundAtVol "FlipperUp", Flipper6, VolFlip
+    PlaySound "FlipperUp"
   End If
   If keycode = LeftTiltKey Then
     Nudge 90, 2
@@ -325,25 +299,19 @@
     playsound "nudge"
     TiltCheck
   End If
-  If keycode = MechanicalTilt Then
-    playsound "nudge"
-    TiltCheck
-  End If
 End Sub
 
 Sub Table1_KeyUp(ByVal keycode)
   If keycode = PlungerKey Then
     Plunger.Fire
-PlaySoundAtVol "plunger", Plunger, 1
+PlaySound "plunger",0,1,0.25,0.25
   End If
   If keycode = LeftFlipperKey Then
 DOF 101,0
     Flipper1.RotateToStart
     Flipper2.RotateToStart
     Flipper3.RotateToStart
-    PlaySoundAtVol "FlipperDown", Flipper1, VolFlip
-    PlaySoundAtVol "FlipperDown", Flipper2, VolFlip
-    PlaySoundAtVol "FlipperDown", Flipper3, VolFlip
+    PlaySound "FlipperDown"
 
   End If
   If keycode = RightFlipperKey Then
@@ -351,9 +319,7 @@
     Flipper4.RotateToStart
     Flipper5.RotateToStart
     Flipper6.RotateToStart
-    PlaySoundAtVol "FlipperDown", Flipper4, VolFlip
-    PlaySoundAtVol "FlipperDown", Flipper5, VolFlip
-    PlaySoundAtVol "FlipperDown", Flipper6, VolFlip
+    PlaySound "FlipperDown"
   End If
 End Sub
 
@@ -542,7 +508,7 @@
 
   bump1 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
 
   If lightb1.state = 0 then
 lightb1.state = 1: addscore (10000)
@@ -568,13 +534,13 @@
 DOF 148,0
 end sub
 
-sub r2_hit(): PlaySoundAtVol "fx_bumper", ActiveBall, 1
+sub r2_hit(): PlaySound "fx_bumper"
 if rsl2.state= 1 then
 Primitive74.image = "redshootcovertexture"
 DOF 125,0
 giLighttriangle2.state = 0
 AwardSpecial
-PlaySoundAtVol "fx_bumper", ActiveBall, 1
+PlaySound "fx_bumper"
 end if
 if lefto.state= 0 then
 addscore (10000)
@@ -592,7 +558,7 @@
 
   bump2 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
 
 If ali=1 and lightb2.state= 0 then
   lightb2.state = 1
@@ -623,7 +589,7 @@
 
   bump3 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   If ali=2 and lightb3.state= 0 then
   lightb3.state = 1
 DOF 149,1
@@ -655,7 +621,7 @@
 
   bump4 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
 If ali=3 and lightb4.state = 0 and moo=0 then
   lightb4.state = 1: addscore (10000)
 DOF 150,1
@@ -687,7 +653,7 @@
 DOF 150,0
 end sub
 
-sub r5_hit(): PlaySoundAtVol "fx_bumper", ActiveBall, 1
+sub r5_hit(): PlaySound "fx_bumper"
   if rsl1.state= 1 then
 DOF 144,1
   Primitive57.image = "redshootcovertextureon"
@@ -718,7 +684,7 @@
 
   bumptop = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   If Lightbtop.state = 1 then
 DOF 136,1
   addscore (10000)
@@ -754,7 +720,7 @@
 sub dbump2_hit()
   bumpcenter = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   if dbl2.state= 1 then
 DOF 149,1
 DOF 151,1
@@ -1057,39 +1023,37 @@
 
 '********** Rubber non-scoring wall animations
 
-' Thalamus : This sub is used twice - this means ... this one IS NOT USED
-' Not a issue though, they are the same
-' sub Rwall1_hit
-'   Rw1.visible=0
-'   RW1a.visible=1
-'   rw1step=1
-'   Me.timerenabled=1
-' end sub
-
-' sub Rwall1_timer
-'   select case rw1step
-'     Case 1: RW1a.visible=0: Rw1.visible=1
-'     case 2: RW1.visible=0: rw1b.visible=1
-'     Case 3: rw1b.visible=0: Rw1.visible=1: me.timerenabled=0
-'   end Select
-'   rw1step=rw1step+1
-' end sub
-
-' sub Rwall2_hit
-'   Rw2.visible=0
-'   RW2a.visible=1
-'   rw2step=1
-'   Me.timerenabled=1
-' end sub
-
-' sub Rwall2_timer
-'   select case rw2step
-'     Case 1: RW2a.visible=0: Rw2.visible=1
-'     case 2: RW2.visible=0: rw2b.visible=1
-'     Case 3: rw2b.visible=0: Rw2.visible=1: me.timerenabled=0
-'   end Select
-'   rw2step=rw2step+1
-' end sub
+sub Rwall1_hit
+  Rw1.visible=0
+  RW1a.visible=1
+  rw1step=1
+  Me.timerenabled=1
+end sub
+
+sub Rwall1_timer
+  select case rw1step
+    Case 1: RW1a.visible=0: Rw1.visible=1
+    case 2: RW1.visible=0: rw1b.visible=1
+    Case 3: rw1b.visible=0: Rw1.visible=1: me.timerenabled=0
+  end Select
+  rw1step=rw1step+1
+end sub
+
+sub Rwall2_hit
+  Rw2.visible=0
+  RW2a.visible=1
+  rw2step=1
+  Me.timerenabled=1
+end sub
+
+sub Rwall2_timer
+  select case rw2step
+    Case 1: RW2a.visible=0: Rw2.visible=1
+    case 2: RW2.visible=0: rw2b.visible=1
+    Case 3: rw2b.visible=0: Rw2.visible=1: me.timerenabled=0
+  end Select
+  rw2step=rw2step+1
+end sub
 
 sub Rwall1_hit
   Rw1.visible=0
@@ -1134,7 +1098,7 @@
 
   bumpbb1 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   If Lightbtopleft.state = 1 then
 DOF 129,1
   Primitive107.image = "basedark"
@@ -1164,7 +1128,7 @@
 
   bumpbb2 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   If Lightbtopright.state = 1 then
 DOF 139,1
   Primitive108.image = "basedark"
@@ -1193,7 +1157,7 @@
 
   bumpbb3 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   If Lightbbottomleft.state = 1 then
 DOF 132,1
   Primitive104.image = "basedark"
@@ -1222,7 +1186,7 @@
 
   bumpbb4 = 1
   Me.TimerEnabled = 1
-  PlaySoundAtVol "fx_bumper", ActiveBall, 1
+  PlaySound "fx_bumper"
   If Lightbbottomright.state = 1 then
 DOF 142,1
   Primitive106.image = "basedark"
@@ -1259,7 +1223,7 @@
 end sub
 
 sub AdvButton_hit()
-    playsoundAtVol "metal", ActiveBall, 1
+    playsound "metal"
     Addscore 10000
     wall7x.isdropped=True
   timer2.enabled=True
@@ -1605,7 +1569,7 @@
 
 Sub OutGate_Hit()
   Hold = 0
-    PlaySoundAtVol "gater", ActiveBall, VolGates
+    PlaySound "gater"
     If Tilted = TRUE  and Ball = Ballsout then
        GameOn = FALSE
        Exit Sub
@@ -1633,7 +1597,7 @@
   primitive95.rotz=gate.CurrentAngle
 End Sub
 
-Sub Gate_hit : PlaySoundAtVol "Gate", ActiveBall, VolGates : End Sub
+Sub Gate_hit : PlaySound "Gate" : End Sub
 
 
 ' High Score To Date Routines
@@ -2011,39 +1975,40 @@
 ' Last lines: If the distance between 2 balls is more than the radius of a ball,
 ' then there is no collision and then set the collision property of the ball to False (-1).
 
+
 Sub Pins_Hit (idx)
-  PlaySound "pinhit_low", 0, Vol(ActiveBall)*VolPi, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "pinhit_low", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Targets_Hit (idx)
-  PlaySound "target", 0, Vol(ActiveBall)*VolTarg, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "target", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
-Sub Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+Sub metal_Hit (idx)
+  PlaySound "metal", 10, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall)*VolGates, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Spinner_Spin
-  PlaySoundAtVol "fx_spinner", Spinner, VolSpin
+  PlaySound "fx_spinner",0,.25,0,0.25
 End Sub
 
 Sub rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 1 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 1 then
     RandomSoundRubber()
@@ -2054,7 +2019,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolPo, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -2063,9 +2028,9 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -2079,9 +2044,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -2116,8 +2081,8 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volum)
-  PlaySound sound, 1, Volum, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
@@ -2177,7 +2142,7 @@
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -2240,7 +2205,10 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  If Table1.VersionMinor > 3 OR Table1.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
 
-
