--- vpxtable_scripts/Jacks Open (Gottlieb 1977) v1.2.vbs
+++ todo_extracted_tables/Jacks Open (Gottlieb 1977) v1.2.vbs
@@ -14,30 +14,12 @@
 '   NEW -  New High Score Sticky and additional photo resources from GNance
 '   NEW -  Score routine fixes by BorgDog
 
-Option Explicit
-Randomize
-
 ' Thalamus 2018-07-23
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
 ' No special SSF tweaks yet.
-' Thalamus 2018-11-01 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolFlip   = 1    ' Flipper volume.
 
+Option Explicit
+Randomize
 
 Const cGameName = "jacksopen"
 
@@ -154,14 +136,14 @@
 Sub JacksOpen_KeyDown(ByVal keycode)
   If Gamestarted AND(NOT Tilted) Then
     If keycode = LeftFlipperKey Then
-      PlaySoundAtVol SoundFXDOF("PNK_MH_Flip_L_up",101,DOFOn,DOFContactors), LeftFlipper, VolFlip
+      PlaySound SoundFXDOF("PNK_MH_Flip_L_up",101,DOFOn,DOFContactors)
       LeftFlipper.RotateToEnd
-      PlaySoundAtVol "Buzz1", LeftFlipper, VolFlip
+      PlaySound "Buzz1",-1,1,.1
     end if
     If keycode = RightFlipperKey Then
-      PlaySoundAtVol SoundFXDOF("PNK_MH_Flip_R_up",102,DOFOn,DOFContactors), RightFlipper, VolFlip
+      PlaySound SoundFXDOF("PNK_MH_Flip_R_up",102,DOFOn,DOFContactors)
       RightFlipper.RotateToEnd
-      PlaySoundAtVol "Buzz", RightFlipper, VolFlip
+      PlaySound "Buzz",-1,1,.1
     end if
   End If
   If keycode = LeftTiltKey Then
@@ -192,12 +174,12 @@
   If Gamestarted AND(NOT Tilted) Then
     If keycode = LeftFlipperKey Then
       LeftFlipper.RotateToStart
-      PlaySoundAtVol SoundFXDOF("PNK_MH_Flip_L_down",101,DOFOff,DOFContactors), LeftFlipper, VolFlip
+      PlaySound SoundFXDOF("PNK_MH_Flip_L_down",101,DOFOff,DOFContactors)
       StopSound "Buzz1"
     end if
     If keycode = RightFlipperKey Then
       RightFlipper.RotateToStart
-      PlaySoundAtVol SoundFXDOF("PNK_MH_Flip_R_down",102,DOFOff,DOFContactors), RightFlipper, VolFlip
+      PlaySound SoundFXDOF("PNK_MH_Flip_R_down",102,DOFOff,DOFContactors)
       StopSound "Buzz"
     end if
   End If
@@ -267,7 +249,7 @@
 Sub AddCredit
   Credits = Credits + 1
   DOF 142, DOFOn
-  PlaySoundAtVol "coin", drain ,1
+  PlaySound "coin"
   If Credits > 15 Then Credits = 15
   If B2SOn then Controller.B2SSetCredits credits
   EMReelCredits.setvalue(Credits)
@@ -299,7 +281,7 @@
         If B2SOn then Controller.B2ssetballinplay 32, CurrentBall
         EMReelBall.SetValue(CurrentBall)
         Drain.Kick 65, 15
-          PlaySoundAtVol SoundFXDOF("ballrel",109,DOFPulse,DOFContactors), Drain, 1
+          PlaySound SoundFXDOF("ballrel",109,DOFPulse,DOFContactors)
       End If
     End If
   End If
@@ -307,7 +289,7 @@
   If DropResetPause > 0 Then
     DropResetPause = DropResetPause  - 1
     If DropResetPause = 0 Then
-      PlaySoundAtVol "dropResetCenter", Target5, VolTarg
+      PlaySound "dropResetCenter"
       For each obj in Drops
         obj.IsDropped=false
       Next
@@ -317,7 +299,7 @@
   If RedResetPause > 0 Then
     RedResetPause = RedResetPause  - 1
     If RedResetPause = 0 Then
-      PlaySoundAtVol "dropResetCenter", Target5, VolTarg
+      PlaySound "dropResetCenter"
       Target2.IsDropped = True:Target4.IsDropped = True:Target6.IsDropped = True:Target8.IsDropped = True
     End If
   End If
@@ -548,7 +530,7 @@
 '  OBJECT HANDLERS
 '****************************************
 
-Sub Gate_Hit():PlaysoundAtVol "PNK_MH_gate", Gate, 1:End Sub
+Sub Gate_Hit():Playsound "PNK_MH_gate":End Sub
 Sub LeftFlipper_Collide(parm):RandomSoundFlipper():End Sub
 Sub RightFlipper_Collide(parm):RandomSoundFlipper():End Sub
 
@@ -712,7 +694,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target1Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropLeft",111,DOFPulse,DOFDropTargets), Target1, VolTarg
+    playsound SoundFXDOF("dropLeft",111,DOFPulse,DOFDropTargets)
     AddScore (1000)
     if targetscore.enabled=false then
       TSCount=1
@@ -725,7 +707,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target2Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropLeft",111,DOFPulse,DOFDropTargets), Target2, VolTarg
+    playsound SoundFXDOF("dropLeft",111,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -738,7 +720,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target3Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropLeft",111,DOFPulse,DOFDropTargets), Target3, VolTarg
+    playsound SoundFXDOF("dropLeft",111,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -751,7 +733,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target4Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropCenter",112,DOFPulse,DOFDropTargets), Target4, VolTarg
+    playsound SoundFXDOF("dropCenter",112,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -764,7 +746,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target5Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropCenter",112,DOFPulse,DOFDropTargets), Target5, VolTarg
+    playsound SoundFXDOF("dropCenter",112,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -777,7 +759,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target6Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropCenter",112,DOFPulse,DOFDropTargets), Target6, VolTarg
+    playsound SoundFXDOF("dropCenter",112,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -790,7 +772,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target7Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropRight",113,DOFPulse,DOFDropTargets), Target7, VolTarg
+    playsound SoundFXDOF("dropRight",113,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -803,7 +785,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target8Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropRight",113,DOFPulse,DOFDropTargets), Target8, VolTarg
+    playsound SoundFXDOF("dropRight",113,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -816,7 +798,7 @@
   If Gamestarted AND(NOT Tilted) Then
     Target9Down=True
     SMFadePause=SMFadeResetValue
-    playsoundAtVol SoundFXDOF("dropRight",113,DOFPulse,DOFDropTargets), Target9, VolTarg
+    playsound SoundFXDOF("dropRight",113,DOFPulse,DOFDropTargets)
     AddScore (1000)
       if targetscore.enabled=false then
         TSCount=1
@@ -832,7 +814,7 @@
 '****************************************
 
 Sub SwTop1_Hit()
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 120, DOFPulse
   If Gamestarted AND(NOT Tilted) Then
     If LightTRO1.state=1 Then
@@ -851,7 +833,7 @@
 End Sub
 
 Sub SwTop2_Hit()
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 103, DOFPulse
   If Gamestarted AND(NOT Tilted) Then
     If LightTRO2.state=1 Then
@@ -870,7 +852,7 @@
 End Sub
 
 Sub SwTop3_Hit()
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 104, DOFPulse
   If Gamestarted AND(NOT Tilted) Then
     If LightTRO3.state=1 Then
@@ -889,7 +871,7 @@
 End Sub
 
 Sub SwTop4_Hit()
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 105, DOFPulse
   If Gamestarted AND(NOT Tilted) Then
     If LightTRO4.state=1 Then
@@ -926,7 +908,7 @@
 '****************************************
 
 Sub SwLRO_Hit()
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 131, DOFPulse
   If LightLROb.state=1 Then
     AddScore(1000)
@@ -936,7 +918,7 @@
 End Sub
 
 Sub SwRRO_Hit()
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 134, DOFPulse
   If LightRROb.state=1 Then
     AddScore(1000)
@@ -947,7 +929,7 @@
 
 Sub SwLIL_Hit()
   SwLILPrim.RotX = 135
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 132, DOFPulse
   If LightLILb.state=1 Then
     AddScore(1000)
@@ -962,7 +944,7 @@
 
 Sub SwRIL_Hit()
   SwRILPrim.RotX = 135
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 133, DOFPulse
   If LightRILb.state=1 Then
     AddScore(1000)
@@ -977,7 +959,7 @@
 
 Sub SwLOL_Hit()
   SwLOLPrim.RotX = 135
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 131, DOFPulse
   If LightLOLb.state=1 Then
     AddScore(5000)
@@ -992,7 +974,7 @@
 
 Sub SwROL_Hit()
   SwROLPrim.RotX = 135
-  PlaySoundAtVol "RollOver", ActiveBall, 1
+  PlaySound "RollOver"
   DOF 134, DOFPulse
   If LightROLb.state=1 Then
     AddScore(5000)
@@ -1010,17 +992,17 @@
 '****************************************
 
 Sub Bumper1_Hit()
-    PlaySoundAtVol SoundFXDOF("bumperLeft",103, DOFPulse,DOFContactors), Bumper1, VolBump
+    PlaySound SoundFXDOF("bumperLeft",103, DOFPulse,DOFContactors)
   If B1Lit=1 Then:AddScore(100):Else:AddScore(10):End If
 End Sub
 
 Sub Bumper2_Hit()
-    PlaySoundAtVol SoundFXDOF("bumperCenter",104, DOFPulse,DOFContactors), Bumper2, VolBump
+    PlaySound SoundFXDOF("bumperCenter",104, DOFPulse,DOFContactors)
   If B2Lit=1 Then:AddScore(1000):Else:AddScore(100):End If
 End Sub
 
 Sub Bumper3_Hit()
-    PlaySoundAtVol SoundFXDOF("bumperRight",105, DOFPulse,DOFContactors), Bumper3, VolBump
+    PlaySound SoundFXDOF("bumperRight",105, DOFPulse,DOFContactors)
   If B3Lit=1 Then:AddScore(100):Else:AddScore(10):End If
 End Sub
 
@@ -1051,9 +1033,9 @@
 Sub Drain_Hit:EndOfBall:End Sub 'RollingSoundsOff:SoundTimer.enabled=0:EndOfBall:End Sub
 
 Sub EndOfBall()
-  PlaySoundAtVol "drain", drain, 1
+  PlaySound "drain"
   DOF 141, DOFPulse
-  SMFadePause=SMFadeResetValue:PlaySoundAtVol "MekSound", drain, 1
+  SMFadePause=SMFadeResetValue:PlaySound "MekSound"
   'Enable again if Tilted
   If Tilted Then
     EMReelTILT.SetValue 0
@@ -1461,8 +1443,8 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volum)
-  PlaySound sound, 1, Volum, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
@@ -1522,7 +1504,7 @@
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -1585,5 +1567,9 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  If JacksOpen.VersionMinor > 3 OR JacksOpen.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
