--- vpxtable_scripts/KC Jones (Gottlieb 1949).vbs
+++ todo_extracted_tables/KC Jones (Gottlieb 1949).vbs
@@ -451,9 +451,9 @@
 
     If Keycode=LeftFlipperKey Then
       LeftFlipper.RotateToEnd
-      PlaySoundatvol "FlipperUp", LeftFlipper, 1
+      PlaySound"FlipperUp"
             If BuzzCount=0 Then
-                PlayLoopSoundAtVol "Buzz", LeftFlipper, 1          ' Start Flipper Solenoid Buzz
+                PlaySound "Buzz", -1          ' Start Flipper Solenoid Buzz
             End If
             BuzzCount=BuzzCount+1
       End If
@@ -461,9 +461,9 @@
 
     If Keycode=RightFlipperKey Then
       RightFlipper.RotateToEnd
-      PlaySoundatvol "FlipperUp", RightFlipper, 1
+      PlaySound"FlipperUp"
             If BuzzCount=0 Then
-                PlayLoopSoundAtVol "Buzz", RightFlipper, 1          ' Start Flipper Solenoid Buzz
+                PlaySound "Buzz", -1          ' Start Flipper Solenoid Buzz
             End If
             BuzzCount=BuzzCount+1
       End If
@@ -482,12 +482,6 @@
       Nudge 0,0.75
       CheckTilt()
     End If
-' Thalamus - added mechanicaltilt
-    If Keycode=MechanicalTilt Then
-      Nudge 0,0.75
-      CheckTilt()
-    End If
-
     '
     ' End flipper and Nudge
     '
@@ -578,7 +572,7 @@
 
     If Keycode=LeftFlipperKey Then
       LeftFlipper.RotateToStart
-      PlaySoundatvol "FlipperDown", LeftFlipper, 1
+      PlaySound"FlipperDown"
             BuzzCount=BuzzCount-1
             If BuzzCount <= 0 Then
                StopSound "Buzz"           ' End Flipper Solenoid Buzz
@@ -587,7 +581,7 @@
 
     If Keycode=RightFlipperKey Then
       RightFlipper.RotateToStart
-      PlaySoundatvol "FlipperDown", RightFlipper, 1
+      PlaySound"FlipperDown"
             BuzzCount=BuzzCount-1
             If BuzzCount <= 0 Then
                StopSound "Buzz"           ' End Flipper Solenoid Buzz
@@ -599,9 +593,9 @@
         ShootBall=True                              'Reverse plunger animation
     Plunger.Fire
         If BallAtPlunger = True Then
-           PlaySoundatvol  "PlungerRoll", Plunger, 1
+           PlaySound "PlungerRoll"
         Else
-           PlaySoundatVol "PlungerSpring", Plunger, 1
+           PlaySound"PlungerSpring"
         End If
   End If
 End Sub
@@ -676,14 +670,14 @@
 '
 ' Gate hit as ball enters playfield
 Sub Gate_Hit()                    'Just for the sound
-  PlaySoundatvol "BallRoll1", ActiveBall, 1
+  PlaySound"BallRoll1"
     RollBack=0                        ' raise Stopper at right end of ball trough
 End Sub
 '
 ' Hit gate going wrong way at top
 Sub Rebound_Hit()
     If FirstOut > 0 Then
-       PlaySoundAtVol "Gate5", ActiveBall, 1              ' Gate Bounce Sound
+       PlaySound "Gate5"              ' Gate Bounce Sound
     Else
        FirstOut=1
     End If
@@ -694,7 +688,7 @@
 ' -----------------
 '
 Sub Wheel_Hit()
-    PlaysoundAtVol "rubber", ActiveBall, 1
+    Playsound "rubber"
     If FirstOut < 2 Then
        FirstOut=2
        AddSound "RollMore"
@@ -715,7 +709,7 @@
 Sub LaneSwitch_Hit()
     If RollBack=1 Then
        ' Failed to Clear Gate on Plunger Release
-       PlaySoundAtVol "Roll9", ActiveBall, 1
+       PlaySound "Roll9"
     Else
        ' On way out of lane from release
        RollBack=1
@@ -1501,9 +1495,9 @@
 ' -------------------
 '
 Sub Bumper1_Hit
-    PlaySoundAtVol "Rubber", ActiveBall, 1
+    PlaySound "Rubber"
  If LightB1.state=0 Then AddScore 1, 2      ' 10,000 Pts + Spot Sound
- If LightB1.state=1 Then AddScore 1, 0     ' 10,000 Points
+ If LightB1.state=1 Then AddScore 1, 0      ' 10,000 Points
     LightB1.state=1
 
     '
@@ -1526,9 +1520,9 @@
 '
 Sub Bumper2_Hit
     If LightB1.state=1 Then LightB2.state=1
-    PlaySoundAtVOl "Rubber", ActiveBall, 1
+    PlaySound "Rubber"
  If LightB2.state=0 Then AddScore 1, 2      ' 10,000 Pts + Spot Sound
- If LightB2.state=1 Then AddScore 1, 0     ' 10,000 Points
+ If LightB2.state=1 Then AddScore 1, 0      ' 10,000 Points
 
     '
     '  Debounce Ball "Laying" on Passive Bumper
@@ -1550,9 +1544,9 @@
 '
 Sub Bumper3_Hit
     If LightB2.state=1 Then LightB3.state=1
-    PlaySoundAtVofl "Rubber", ActiveBall, 1
+    PlaySound "Rubber"
  If LightB3.state=0 Then AddScore 1, 2      ' 10,000 Pts + Spot Sound
- If LightB3.state=1 Then AddScore 1, 0     ' 10,000 Points
+ If LightB3.state=1 Then AddScore 1, 0      ' 10,000 Points
 
     '
     '  Debounce Ball "Laying" on Passive Bumper
@@ -1574,9 +1568,9 @@
 '
 Sub Bumper4_Hit
     If LightB3.state=1 Then LightB4.state=1
-    PlaySoundAtVol "Rubber", ActiveBall, 1
+    PlaySound "Rubber"
  If LightB4.state=0 Then AddScore 1, 2      ' 10,000 Pts + Spot Sound
- If LightB4.state=1 Then AddScore 1, 0     ' 10,000 Points
+ If LightB4.state=1 Then AddScore 1, 0      ' 10,000 Points
     '
     '  Debounce Ball "Laying" on Passive Bumper
     '
@@ -1597,9 +1591,9 @@
 '
 Sub Bumper5_Hit
     If LightB4.state=1 Then LightB5.state=1
-    PlaySoundAtVol "Rubber", ActiveBall, 1
+    PlaySound "Rubber"
  If LightB5.state=0 Then AddScore 1, 2      ' 10,000 Pts + Spot Sound
- If LightB5.state=1 Then AddScore 1, 0     ' 10,000 Points
+ If LightB5.state=1 Then AddScore 1, 0      ' 10,000 Points
     '
     '  Debounce Ball "Laying" on Passive Bumper
     '
@@ -1623,7 +1617,7 @@
      If ActiveBall.Y < 1425 OR ActiveBall.Y > 1660 Then
         AddSound "Rubber"       ' Hit Outside Contact Switch Zone
      Else
-          PlaySoundAtVol "Rubber", ActiveBall, 1
+          PlaySound "Rubber"
           If SpotNumber(6)=1 Then
              AddScore 1, 2        ' 10,000 Pts + Spot Sound
           Else
@@ -1655,7 +1649,7 @@
         AddSound "Rubber"       ' Hit Outside Contact Switch Zone
      Else
        If B7Top.UserValue=False Then
-          PlaySoundAtVol "Rubber", ActiveBall, 1
+          PlaySound "Rubber"
           If SpotNumber(7)=1 Then
              AddScore 1, 2          ' 10,000 Pts + Spot Sound
           Else
@@ -1684,7 +1678,7 @@
 '
 Sub Bumper8_Hit()
     LightB8.state=0
-    PlaySoundAtVol "Bumper", ActiveBall, 1
+    PlaySound "Bumper"
     AddScore 1, 0           ' 10,000 Points
     B8Rings()             ' Kick off Ring Animation
     B8Ring3.TimerEnabled=True
@@ -1748,7 +1742,7 @@
 '
 Sub Bumper9_Hit()
     LightB9.state=0
-    PlaySoundAtVol "Bumper", ActiveBall, 1
+    PlaySound "Bumper"
     AddScore 1, 0           '10,000 Points
     B9Rings()             ' Kick off Ring Animation
     B9Ring3.TimerEnabled=True
@@ -1811,7 +1805,7 @@
 ' -------------------------
 '
 Sub Bumper10_Hit
-       PlaySoundAtVol "Rubber", ActiveBall, 1
+       PlaySound "Rubber"
        If LightB10.State=1 Then AddScore 50, 0        ' 500,000 Pts When Lit
        If LightB10.State=0 Then AddScore 5, 0         ' 50,000 Points
 
@@ -1836,8 +1830,8 @@
 ' -------------------------
 '
 Sub Bumper11_Hit
-       PlaySoundAtVol "Rubber", ActiveBall, 1
-       If LightB11.State=1 Then AddScore 50, 0       ' 500,000 Pts When Lit
+       PlaySound "Rubber"
+       If LightB11.State=1 Then AddScore 50, 0        ' 500,000 Pts When Lit
        If LightB11.State=0 Then AddScore 5, 0         ' 50,000 Points
 
     EndZCheck(0)            ' See if Z state is over
@@ -1864,7 +1858,7 @@
      End If
     X=ActiveBall.X
     If X < 100 OR X > 275 Then
-       PlaySoundAtVol "Rubber", ActiveBall, 1
+       PlaySound "Rubber"
        Exit Sub
     End If
     If SW12T1Lit.IsDropped=True Then
@@ -1895,7 +1889,7 @@
      End If
     X=ActiveBall.X
     If X < 700 OR X > 875 Then
-       PlaySoundAtVol "Rubber", ActiveBall, 1
+       PlaySound "Rubber"
        Exit Sub
     End If
     If SW13T1Lit.IsDropped=True Then
@@ -1941,7 +1935,7 @@
        ActiveBall.VelY=-12        ' Kick Upward Force
        X=ActiveBall.VelX
        ActiveBall.VelX=-X       ' With Reflective Rebound Drift
-       PlaySoundAtVol "Bumper", ActiveBall, 1
+       PlaySound "Bumper"
        Exit Sub
     End If
     '
@@ -2068,7 +2062,7 @@
 Sub CtrKicker_Hit()
     CtrKickerTrig.TimerEnabled=False  ' Stop Attraction Timer
      InKicker=True            ' Say Ball In Kicker
-    PlaySoundAtVol "KickerLand", ActiveBall, 1
+    PlaySound "KickerLand"
     CtrKicker.UserValue=0       ' Assume not lit
     CtrKickerCover.IsDropped=True   ' Drop UnLit Cover
     If CtrKickerCoverLit.IsDropped=False Then
@@ -2556,7 +2550,7 @@
 '
 '-----------------------------------------------------------
 Sub Showscore()
-    Dim N1         ' Define Local Temp Variable
+    Dim N1          ' Define Local Temp Variable
     '
     For N1=71 to 77
           Controller.B2SSetData N1, 0   ' Set All Millions Lights Off
@@ -2566,7 +2560,7 @@
     End If
     '
     For N1=61 to 69
-          Controller.B2SSetData N1, 0    ' Set All Hundred Thousand Lights Off
+          Controller.B2SSetData N1, 0   ' Set All Hundred Thousand Lights Off
     Next
     If SReelTens > 0 Then
           Controller.B2SSetData 60+SReelTens, 1 'Set Correct Hundred Thousand Light On
@@ -2599,183 +2593,3 @@
  Case 9: Controller.B2sSetData 19,1
  End Select
  End Sub
-
-
-' Thalamus : Exit in a clean and proper way
-Sub Table_exit
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
-
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
-'            Supporting Ball, Sound Functions and Math
-'*********************************************************************
-
-Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Const Pi = 3.1415927
-
-Function dSin(degrees)
-  dsin = sin(degrees * Pi/180)
-End Function
-
-Function dCos(degrees)
-  dcos = cos(degrees * Pi/180)
-End Function
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / table1.height-1
-' Thalamus, AudioFade - Patched
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^5) 'was 10
-  Else
-    AudioFade = Csng(-((- tmp) ^5) ) 'was 10
-  End If
-End Function
-
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / table1.width-1
-' Thalamus, AudioPan - Patched
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^5) 'was 10
-  Else
-    AudioPan = Csng(-((- tmp) ^5) ) 'was 10
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / table1.width-1
-' Thalamus, Pan - Patched
-  If tmp > 0 Then
-    Pan = Csng(tmp ^5) 'was 10
-  Else
-    Pan = Csng(-((- tmp) ^5) ) ' was 10
-  End If
-End Function
-
-Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
-End Function
-
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-End Function
-
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
-End Function
-
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
-End Function
-
-Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
-End Function
-
-Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
-End Function
-
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
-End Function
-
