--- vpxtable_scripts/KISS Balutito MOD 1.3.vbs
+++ todo_extracted_tables/KISS Balutito MOD 1.3.vbs
@@ -144,7 +144,7 @@
 Dim bMultiBallMode
 Dim bMusicOn
 Dim bExtraBallWonThisBall
-' Dim PI: PI = Round(4 * Atn(1), 6)
+Dim PI: PI = Round(4 * Atn(1), 6)
 
 Dim plungerIM 'used mostly as an autofire plunger
 Dim ttable
@@ -275,7 +275,7 @@
     If Keycode = AddCreditKey Then
         Credits = Credits + 1
         DOF 125, DOFOn
-        PlaySoundAtVol "fx_coin", Drain, 1
+        PlaySound "fx_coin",0,1,0.25,0.2
         If NOT bGameInPlay then
           If(Tilted = False) Then
             AttractMode.enabled=False
@@ -294,7 +294,7 @@
     If keycode = PlungerKey Then
 '        PlungerIM.AutoFire
         Plunger.PullBack
-    PlaySoundAtVol "fx_plungerpull", Plunger, 1
+    PlaySound "fx_plungerpull",0,1,0.25,0.2
     End If
 
     If bGameInPlay AND NOT Tilted Then
@@ -302,7 +302,6 @@
     If keycode = LeftTiltKey Then   Nudge 90,  6:PlaySound "fx_nudge", 0, 1, -0.1, 0.25:CheckTilt
     If keycode = RightTiltKey Then  Nudge 270, 6:PlaySound "fx_nudge", 0, 1, 0.1, 0.25:CheckTilt
     If keycode = CenterTiltKey Then Nudge 0,   7:PlaySound "fx_nudge", 0, 1, 1, 0.25:CheckTilt
-    If keycode = MechanicalTilt Then CheckTilt
 
 dim xxx,yyy
 if keycode = LeftMagnaSave and 1=2 Then
@@ -484,7 +483,7 @@
     End If
   If keycode = PlungerKey Then
     Plunger.Fire
-    PlaySoundAtVol "fx_plunger", Plunger, 1
+    PlaySound "fx_plunger",0,1,0.25,0.25
   End If
 End Sub
 
@@ -577,12 +576,12 @@
 ' Ramp Sounds
 Sub RHelp1_Hit()
     StopSound "fx_metalrolling"
-    PlaySoundAtVol "fx_ballrampdrop", ActiveBall, 1
+    PlaySound "fx_ballrampdrop", 0, 1, pan(ActiveBall)
 End Sub
 
 Sub RHelp2_Hit()
     StopSound "fx_metalrolling"
-    PlaySoundAtVol "fx_ballrampdrop", ActiveBall, 1
+    PlaySound "fx_ballrampdrop", 0, 1, pan(ActiveBall)
 End Sub
 
 
@@ -711,7 +710,7 @@
     BallsOnPlayfield = BallsOnPlayfield + 1
 
     ' kick it out..
-    PlaySoundAtVol SoundFXDOF("fx_Ballrel",123,DOFPulse,DOFContactors), BallRelease, 1
+    PlaySound SoundFXDOF("fx_Ballrel",123,DOFPulse,DOFContactors), 0, 1, 0.1, 0.1
     BallRelease.Kick 90, 4
 
     ' if there is 2 or more balls then set the multibal flag (remember to check for locked balls and other balls used for animations)
@@ -1033,7 +1032,7 @@
     Drain.DestroyBall
     BallsOnPlayfield = BallsOnPlayfield - 1
     ' pretend to knock the ball into the ball storage mech
-    PlaySoundAtVol "fx_drain", Drain, 1
+    PlaySound "fx_drain"
 
     debug.print "Show Saved Song (c) #" & SaveSong
 
@@ -1096,7 +1095,7 @@
     debug.print "ball in plunger lane"
     debug.print "Show Saved Song (b) #" & SaveSong
     ' some sound according to the ball position
-    PlaySoundAtVol "fx_sensor", Plunger, 1
+    PlaySound "fx_sensor", 0, 1, 0.15, 0.25
     bBallInPlungerLane = True
     ' turn on Launch light is there is one
     LaunchLight.State = 2
@@ -1295,7 +1294,7 @@
 '                        Table Object Hit Events
 
 Sub sw47_hit ' star child Kicker1_Hits
-   PlaySoundAtVol "fx_hole-enter", sw47, 1
+   PlaySound "fx_hole-enter", 0, 1, -0.1
    If f116.state=1 then
      FlashForMs Flasher9, 1000, 50, 0
      FlashForMs Flasher10, 1000, 50, 0
@@ -1311,7 +1310,7 @@
 End Sub
 
 Sub sw47_unhit
-    PlaySoundAtVol SoundFXDOF("fx_popper",120,DOFPulse,DOFContactors), sw47, 1
+    PlaySound SoundFXDOF("fx_popper",120,DOFPulse,DOFContactors), 0, 1, -0.1, 0.25
     DOF 121, DOFPulse
 End Sub
 
@@ -1327,7 +1326,7 @@
 
 Sub LeftSlingShot_Slingshot
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_slingshot",103,DOFPulse,DOFcontactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_slingshot",103,DOFPulse,DOFcontactors), 0, 1, -0.05, 0.05
   DOF 105, DOFPulse
     LeftSling4.Visible = 1:LeftSling1.Visible = 0
     LStep = 0
@@ -1348,7 +1347,7 @@
 
 Sub RightSlingShot_Slingshot
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_slingshot",104,DOFPulse,DOFcontactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_slingshot",104,DOFPulse,DOFcontactors), 0, 1, 0.05, 0.05
   DOF 106, DOFPulse
     RightSling4.Visible = 1:RightSling1.Visible = 0
     RStep = 0
@@ -1371,7 +1370,7 @@
 
 Sub LeftSlingShot1_Slingshot
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_slingshot",143,DOFPulse,DOFContactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_slingshot",143,DOFPulse,DOFContactors), 0, 1, -0.05, 0.05
     Sling3a.Visible = 1:Sling3.Visible = 0
     Lemk1.RotX = 26
     LStep1 = 0
@@ -1394,7 +1393,7 @@
 
 Sub RightSlingShot1_Slingshot
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_slingshot",144,DOFPulse,DOFContactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_slingshot",144,DOFPulse,DOFContactors), 0, 1, 1, 0.05
     Sling2a.Visible = 1:Sling2.Visible = 0
     Remk1.RotX = 26
     RStep1 = 0
@@ -1424,7 +1423,7 @@
 Dim BallInHole, dBall, dZpos
 
 Sub sw43_Hit
-   PlaySoundAtVol "fx_hole-enter", sw43, 1
+   PlaySound "fx_hole-enter", 0, 1, -0.1, 0.25
    Set dBall = ActiveBall
    dZpos=35
    sw43.TimerInterval = 4
@@ -1457,7 +1456,7 @@
 '**********
 
 Sub Spinner_Spin()
-    PlaySoundAtVol "fx_spinner", Spinner, 1
+    PlaySound "fx_spinner", 0, 1, 0.1
   DOF 136, DOFPulse
     If Tilted Then Exit Sub
     FlashForMs SmallFlasher2, 500, 50, 0
@@ -1484,7 +1483,7 @@
 
 Sub Bumper1_Hit
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_bumper",109,DOFPulse,DOFContactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_bumper",109,DOFPulse,DOFContactors), 0, 1, pan(ActiveBall)
   DOF 138, DOFPulse
     RandomBD() ' If BD then move shot
     if NOT i61.state=LightStateOff then ' Super Bumpers score 200K for upto 50Hits
@@ -1514,7 +1513,7 @@
 
 Sub Bumper2_Hit
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_bumper",110,DOFPulse,DOFContactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_bumper",110,DOFPulse,DOFContactors), 0, 1, pan(ActiveBall)
   DOF 140, DOFPulse
     RandomBD() ' If BD then move shot
     if NOT i61.state=LightStateOff then ' Super Bumpers score 200K for upto 50Hits
@@ -1544,7 +1543,7 @@
 
 Sub Bumper3_Hit
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_bumper",107,DOFPulse,DOFContactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_bumper",107,DOFPulse,DOFContactors), 0, 1, pan(ActiveBall)
   DOF 137, DOFPulse
     RandomBD() ' If BD then move shot
     if NOT i61.state=LightStateOff then ' Super Bumpers score 200K for upto 50Hits
@@ -1574,7 +1573,7 @@
 
 Sub Bumper4_Hit
     If Tilted Then Exit Sub
-    PlaySoundAtVol SoundFXDOF("fx_bumper",108,DOFPulse,DOFContactors), ActiveBall, 1
+    PlaySound SoundFXDOF("fx_bumper",108,DOFPulse,DOFContactors), 0, 1, pan(ActiveBall)
   DOF 139, DOFPulse
     RandomBD() ' If BD then move shot
     if NOT i61.state=LightStateOff then ' Super Bumpers score 200K for upto 50Hits
@@ -1677,179 +1676,3 @@
 '        End If
 '    Next
 'End Sub
-
-' Thalamus : Exit in a clean and proper way
-Sub Table1_exit
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
-
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
-'            Supporting Ball, Sound Functions and Math
-'*********************************************************************
-
-Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Const Pi = 3.1415927
-
-Function dSin(degrees)
-  dsin = sin(degrees * Pi/180)
-End Function
-
-Function dCos(degrees)
-  dcos = cos(degrees * Pi/180)
-End Function
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / table1.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / table1.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / table1.width-1
-  If tmp > 0 Then
-    Pan = Csng(tmp ^10)
-  Else
-    Pan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
-End Function
-
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-End Function
-
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
-End Function
-
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
-End Function
-
-Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
-End Function
-
-Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
-End Function
-
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
-End Function
-
