--- vpxtable_scripts/Mary Shelleys Frankenstein (Sega 1995) v1.22.vbs
+++ todo_extracted_tables/Mary Shelleys Frankenstein (Sega 1995) v1.22.vbs
@@ -247,7 +247,7 @@
 ' keys
 ' ****************************************************
 Sub Frankenstein_KeyDown(ByVal keycode)
-  If keycode = PlungerKey or keycode = LockBarKey Then Controller.Switch(62) = True
+  If keycode = PlungerKey Then Controller.Switch(62) = True
   If keycode = LeftFlipperKey Then LFPress = 1 ' : Controller.Switch(63) = True
   If keycode = RightFlipperKey Then RFPress = 1 : RUFPress = 1 ' : Controller.Switch(64) = True
   If keycode = LeftTiltKey Then Nudge 90,2: Playsound SoundFX("fx_nudge",0)
@@ -265,7 +265,7 @@
 End Sub
 
 Sub Frankenstein_KeyUp(ByVal keycode)
-  If keycode = PlungerKey or keycode = LockBarKey Then Controller.Switch(62) = False
+  If keycode = PlungerKey Then Controller.Switch(62) = False
   If keycode = LeftFlipperKey Then
     LFPress = 0
     LeftFlipper.EOSTorqueAngle = EOSA
