--- vpxtable_scripts/Mata Hari (Bally 1978) v2.0.0.vbs
+++ todo_extracted_tables/Mata Hari (Bally 1978) v2.0.0.vbs
@@ -342,9 +342,9 @@
 
 End Sub
 
-Sub Table1_Paused:Controller.Pause = 1:End Sub
-Sub Table1_unPaused:Controller.Pause = 0:End Sub
-Sub Table1_exit()
+Sub Table_Paused:Controller.Pause = 1:End Sub
+Sub Table_unPaused:Controller.Pause = 0:End Sub
+Sub Table_exit()
   Controller.Pause = False
   Controller.Stop
 End Sub
