--- vpxtable_scripts/Rat Fink VPX v1.1.vbs
+++ todo_extracted_tables/Rat Fink VPX v1.1.vbs
@@ -3,9 +3,6 @@
 'VP10 table layout/code by rothbauerw
 'pop bumper code and mechanics by BorgDog
 'VP10 retheme by HauntFreaks (2016)
-Option Explicit
-Randomize
-
 
 'NOT FOR RESALE OR COMMERCIAL USE!!
 
@@ -13,36 +10,11 @@
 
 ' Thalamus 2018-07-24
 ' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
-' Thalamus 2018-11-01 : Improved directional sounds
-' Changed useSolenoids=2 - thanks Carny
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
-'
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = 2    ' Bumpers volume.
-Const VolRol    = 1    ' Rollovers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRB     = 1    ' Rubber bands volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolPlast  = 1    ' Plastics volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolWood   = 1    ' Woods volume.
-Const VolKick   = 1    ' Kicker volume.
-Const VolSpin   = 1.5  ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
+' No special SSF tweaks yet.
+' This is a JP table. He often uses walls as switches so I need to be careful of using PlaySoundAt
 
+Option Explicit
+Randomize
 
 Dim cBall1, cBall2, cBall3, direction1, direction2, direction3
 Dim bMusicOn
@@ -53,7 +25,7 @@
 If Err Then MsgBox "You need the controller.vbs in order to run this table, available in the vp10 package"
 On Error Goto 0
 
-Const cGameName="ratfink",UseSolenoids=2,UseLamps=1,UseGI=0,SSolenoidOn="SolOn",SSolenoidOff="SolOff",SFlipperOn="fx_Flipperup",SFlipperOff="fx_Flipperdown"
+Const cGameName="ratfink",UseSolenoids=1,UseLamps=1,UseGI=0,SSolenoidOn="SolOn",SSolenoidOff="SolOff",SFlipperOn="fx_Flipperup",SFlipperOff="fx_Flipperdown"
 'Const cGameName="startreb",UseSolenoids=1,UseLamps=1,UseGI=0,SSolenoidOn="SolOn",SSolenoidOff="SolOff",SFlipperOn="fx_Flipperup",SFlipperOff="fx_Flipperdown"
 Const SCoin="Coin",cCredits=""
 
@@ -185,7 +157,7 @@
  Sub Table1_KeyDown(ByVal keycode)
   If vpmKeyDown(KeyCode) Then Exit Sub
 
-    If keycode=PlungerKey Then Plunger.Pullback:playsoundAtVol"plungerpull",plunger,1
+    If keycode=PlungerKey Then Plunger.Pullback:playsound"plungerpull",0,1,0.25,0.25
 
   If keycode = LeftTiltKey Then
     LeftNudge 80, 1.2, 20
@@ -205,7 +177,7 @@
 
 Sub Table1_KeyUp(ByVal keycode)
   If vpmKeyUp(KeyCode) Then Exit Sub
-  If keycode=PlungerKey Then Plunger.Fire:playsoundAtVol"plunger",Plunger, 1
+  If keycode=PlungerKey Then Plunger.Fire:playsound"plunger",0,1,0.25,0.25
 
 End Sub
 
@@ -216,7 +188,7 @@
 Dim RStep, Lstep
 
 Sub RightSlingShot_Slingshot
-    PlaySoundAtVol SoundFX("right_slingshot",DOFContactors), sling1, 1
+    PlaySound SoundFX("right_slingshot",DOFContactors), 0, 1, 0.05, 0.05
     vpmTimer.PulseSw 36
     RightSling.Visible = 0
     RightSling1.Visible = 1
@@ -236,7 +208,7 @@
 End Sub
 
 Sub LeftSlingShot_Slingshot
-    PlaySoundAtVol SoundFX("left_slingshot",DOFContactors), sling2, 1
+    PlaySound SoundFX("left_slingshot",DOFContactors),0,1,-0.05,0.05
     vpmTimer.PulseSw 37
     LeftSling.Visible = 0
     LeftSling1.Visible = 1
@@ -354,7 +326,7 @@
 ' Bumpers
 Sub Bumper1_Hit
   vpmTimer.PulseSw 40
-  PlaySoundAtVol SoundFX("fx_bumper1",DOFContactors), Bumper1, VolBump
+  PlaySound SoundFX("fx_bumper1",DOFContactors)
   direction1=Int(Rnd*2)-1
   if direction1=0 then direction1=1
   cBall1.velx = (10+2*RND(1))*direction1
@@ -363,7 +335,7 @@
 
 Sub Bumper2_Hit
   vpmTimer.PulseSw 38
-  PlaySoundAtVol SoundFX("fx_bumper2",DOFContactors), Bumper2, VolBump
+  PlaySound SoundFX("fx_bumper2",DOFContactors)
   direction2=Int(Rnd*2)-1
   if direction2=0 then direction2=1
   cBall2.velx = (10+2*RND(1))*direction2
@@ -372,7 +344,7 @@
 
 Sub Bumper3_Hit
   vpmTimer.PulseSw 39
-  PlaySoundAtVol SoundFX("fx_bumper3",DOFContactors), Bumper3, VolBump
+  PlaySound SoundFX("fx_bumper3",DOFContactors)
   direction3=Int(Rnd*2)-1
   if direction2=0 then direction3=1
   cBall3.velx = (10+2*RND(1))*direction3
@@ -386,40 +358,40 @@
 End Sub
 
 'Saucer
-Sub Kicker_Hit:PlaySoundAtVol "metalhit2",Kicker,VolKick:bsSaucer.AddBall 0:End Sub
+Sub Kicker_Hit:PlaySound "metalhit2":bsSaucer.AddBall 0:End Sub
 
 ' Rollovers
-Sub sw35_Hit:Controller.Switch(35) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:PlaySoundAtVol "outlane",ActiveBall, 1:End Sub
+Sub sw35_Hit:Controller.Switch(35) = 1:PlaySound "sensor":PlaySound "outlane":End Sub
 Sub sw35_UnHit:Controller.Switch(35) = 0:End Sub
 
-Sub sw20_Hit:Controller.Switch(20) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw20_Hit:Controller.Switch(20) = 1:PlaySound "sensor":End Sub
 Sub sw20_UnHit:Controller.Switch(20) = 0:End Sub
 
-Sub sw19_Hit:Controller.Switch(19) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw19_Hit:Controller.Switch(19) = 1:PlaySound "sensor":End Sub
 Sub sw19_UnHit:Controller.Switch(19) = 0:End Sub
 
-Sub sw34_Hit:Controller.Switch(34) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:PlaySoundAtVol "outlane",ActiveBall, 1:End Sub
+Sub sw34_Hit:Controller.Switch(34) = 1:PlaySound "sensor":PlaySound "outlane":End Sub
 Sub sw34_UnHit:Controller.Switch(34) = 0:End Sub
 
-Sub sw5_Hit:Controller.Switch(5) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw5_Hit:Controller.Switch(5) = 1:PlaySound "sensor":End Sub
 Sub sw5_UnHit:Controller.Switch(5) = 0:End Sub
 
-Sub sw26_Hit:Controller.Switch(26) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw26_Hit:Controller.Switch(26) = 1:PlaySound "sensor":End Sub
 Sub sw26_UnHit:Controller.Switch(26) = 0:End Sub
 
-Sub sw25_Hit:Controller.Switch(25) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw25_Hit:Controller.Switch(25) = 1:PlaySound "sensor":End Sub
 Sub sw25_UnHit:Controller.Switch(25) = 0:End Sub
 
-Sub sw31_Hit:Controller.Switch(31) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw31_Hit:Controller.Switch(31) = 1:PlaySound "sensor":End Sub
 Sub sw31_UnHit:Controller.Switch(31) = 0:End Sub
 
-Sub sw30_Hit:Controller.Switch(30) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw30_Hit:Controller.Switch(30) = 1:PlaySound "sensor":End Sub
 Sub sw30_UnHit:Controller.Switch(30) = 0:End Sub
 
-Sub sw18_Hit:Controller.Switch(18) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw18_Hit:Controller.Switch(18) = 1:PlaySound "sensor":End Sub
 Sub sw18_UnHit:Controller.Switch(18) = 0:End Sub
 
-Sub sw22_Hit:Controller.Switch(22) = 1:PlaySoundAtVol "sensor",ActiveBall, 1:End Sub
+Sub sw22_Hit:Controller.Switch(22) = 1:PlaySound "sensor":End Sub
 Sub sw22_UnHit:Controller.Switch(22) = 0:End Sub
 
 ' Droptargets
@@ -429,11 +401,11 @@
 Sub sw4_Hit:dtbank.Hit 4:lightdt4.state = 1:End Sub
 
 ' Targets
-Sub sw27_Hit:vpmTimer.PulseSw 27:PlaySoundAtVol SoundFX("target",DOFContactors),ActiveBall, VolTarg:End Sub
-Sub sw28_Hit:vpmTimer.PulseSw 28:PlaySoundAtVol SoundFX("target",DOFContactors),ActiveBall, VolTarg:End Sub
-Sub sw29_Hit:vpmTimer.PulseSw 29:PlaySoundAtVol SoundFX("target",DOFContactors),ActiveBall, VolTarg:End Sub
-Sub sw21_Hit:vpmTimer.PulseSw 21:PlaySoundAtVol SoundFX("target",DOFContactors),ActiveBall, VolTarg:End Sub
-Sub sw24_Hit:vpmTimer.PulseSw 24:PlaySoundAtVol SoundFX("target",DOFContactors),ActiveBall, VolTarg:End Sub
+Sub sw27_Hit:vpmTimer.PulseSw 27:PlaySound SoundFX("target",DOFContactors):End Sub
+Sub sw28_Hit:vpmTimer.PulseSw 28:PlaySound SoundFX("target",DOFContactors):End Sub
+Sub sw29_Hit:vpmTimer.PulseSw 29:PlaySound SoundFX("target",DOFContactors):End Sub
+Sub sw21_Hit:vpmTimer.PulseSw 21:PlaySound SoundFX("target",DOFContactors):End Sub
+Sub sw24_Hit:vpmTimer.PulseSw 24:PlaySound SoundFX("target",DOFContactors):End Sub
 
 
  ' Extra Sounds
@@ -491,17 +463,17 @@
 
  Sub SolLFlipper(Enabled)
      If Enabled Then
-         PlaySoundAtVol SoundFX("fx_Flipperup",DOFContactors),LeftFlipper,VolFlip:LeftFlipper.RotateToEnd
+         PlaySound SoundFX("fx_Flipperup",DOFContactors):LeftFlipper.RotateToEnd
      Else
-         PlaySoundAtVol SoundFX("fx_Flipperdown",DOFContactors),LeftFlipper,VolFlip:LeftFlipper.RotateToStart
+         PlaySound SoundFX("fx_Flipperdown",DOFContactors):LeftFlipper.RotateToStart
      End If
   End Sub
 
  Sub SolRFlipper(Enabled)
      If Enabled Then
-         PlaySoundAtVol SoundFX("fx_Flipperup",DOFContactors),RightFlipper,VolFlip:RightFlipper.RotateToEnd
+         PlaySound SoundFX("fx_Flipperup",DOFContactors):RightFlipper.RotateToEnd
      Else
-         PlaySoundAtVol SoundFX("fx_Flipperdown",DOFContactors),RightFlipper,VolFlip:RightFlipper.RotateToStart
+         PlaySound SoundFX("fx_Flipperdown",DOFContactors):RightFlipper.RotateToStart
      End If
  End Sub
 
@@ -515,9 +487,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
@@ -662,38 +634,38 @@
 
 
 Sub Pins_Hit (idx)
-  PlaySound "pinhit_low", 0, Vol(ActiveBall)*VolPi, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "pinhit_low", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Targets_Hit (idx)
-  PlaySound "target", 0, Vol(ActiveBall)*VolTarg, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "target", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall)*VolGates, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
 End Sub
 
 Sub Spinner_Spin
-  PlaySoundAtVol "fx_spinner", Spinner, VolSpin
+  PlaySound "fx_spinner",0,.25,0,0.25
 End Sub
 
 Sub Rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -705,7 +677,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -716,7 +688,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -727,7 +699,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolPo, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -736,9 +708,9 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*2)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
 
@@ -889,8 +861,8 @@
 
 'Set position as table object and Vol manually.
 
-Sub PlaySoundAtVol(sound, tableobj, Volum)
-  PlaySound sound, 1, Volum, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
+Sub PlaySoundAtVol(sound, tableobj, Vol)
+  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
 End Sub
 
 'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
@@ -950,7 +922,7 @@
 End Function
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
+  Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
@@ -1013,9 +985,14 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  If Table1.VersionMinor > 3 OR Table1.VersionMajor > 10 Then
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  Else
+    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 200, Pan(ball1), 0, Pitch(ball1), 0, 0
+  End if
 End Sub
 
+
 ' Thalamus : Exit in a clean and proper way
 Sub Table1_exit()
   Controller.Pause = False
