--- vpxtable_scripts/Secret Service (DataEast 1988)v1.3.vbs
+++ todo_extracted_tables/Secret Service (DataEast 1988)v1.3.vbs
@@ -263,15 +263,11 @@
   Set bsSaucer=New cvpmBallStack
     bsSaucer.InitSaucer sw19,19,85,4
     bsSaucer.InitExitSnd"Popper","SolOn"
-    bsSaucer.KickForceVar = 3
-    bsSaucer.KickAngleVar = 3
 
   Set bsEater=New cvpmBallStack
     bsEater.InitSw 0,11,0,0,0,0,0,0
     bsEater.InitKick sw11,200,5
     bsEater.InitExitSnd"ballrelease","SolOn"
-    bsEater.KickForceVar = 3
-    bsEater.KickAngleVar = 3
 
   Set dtDrop=New cvpmDropTarget
     dtDrop.InitDrop Array(sw36,sw37,sw38,sw39,sw40),Array(36,37,38,39,40)
@@ -1089,11 +1085,11 @@
     ' render the shadow for each ball
     For b = 0 to UBound(BOT)
         If BOT(b).X < Table1.Width/2 Then
-            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) + 6
+            BallShadow(b).X = ((BOT(b).X) - (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) + 6
         Else
-            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/21)) - 6
+            BallShadow(b).X = ((BOT(b).X) + (Ballsize/6) + ((BOT(b).X - (Table1.Width/2))/7)) - 6
         End If
-        ballShadow(b).Y = BOT(b).Y + 4
+        ballShadow(b).Y = BOT(b).Y + 12
         If BOT(b).Z > 20 Then
             BallShadow(b).visible = 1
         Else
@@ -1199,16 +1195,13 @@
   End Select
 End Sub
 
+Sub LeftFlipper_Collide(parm)
+  RandomSoundFlipper()
+End Sub
 
-' Thalamus : This sub is used twice - this means ... this one IS NOT USED
-' disabling these as they has static volume
-' Sub LeftFlipper_Collide(parm)
-'   RandomSoundFlipper()
-' End Sub
-'
-' Sub RightFlipper_Collide(parm)
-'   RandomSoundFlipper()
-' End Sub
+Sub RightFlipper_Collide(parm)
+  RandomSoundFlipper()
+End Sub
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
@@ -1217,12 +1210,3 @@
     Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
   End Select
 End Sub
-
-' Thalamus - 2021-04-30 : added proper exit
-
-' Thalamus : Exit in a clean and proper way
-Sub Table1_exit()
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
