--- vpxtable_scripts/Skyway (Williams 1954).vbs
+++ todo_extracted_tables/Skyway (Williams 1954).vbs
@@ -2658,8 +2658,7 @@
 '   TB.text = coef
 
 '               Applies the coef to x and y velocities
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
   End Sub
 
 '***********This Sub sets the values for Sleeves (or any other future objects) to 85% (or whatever is passed in) of Posts
