--- vpxtable_scripts/SonicStarWars_VPX_1_1.vbs
+++ todo_extracted_tables/SonicStarWars_VPX_1_1.vbs
@@ -7,39 +7,16 @@
 '############################################################################################
 '############################################################################################
 ' Version 1.0 FS mfuegemann 2018
-Option Explicit
-Randomize
 
 ' Thanks to
 ' Akiles for providing the playfield image and plastics scans
 ' TAB for the VP8 version that gave me a good start with the solenoid, lamp and switch numbers
-
-' Thalamus 2018-07-24
-' Added/Updated "Positional Sound Playback Functions" and "Supporting Ball & Sound Functions"
-' Thalamus 2018-11-01 : Improved directional sounds
-' !! NOTE : Table not verified yet !!
-
-' Options
-' Volume devided by - lower gets higher sound
-
-Const VolDiv = 2000    ' Lower number, louder ballrolling/collition sound
-Const VolCol = 10      ' Ball collition divider ( voldiv/volcol )
-
-' The rest of the values are multipliers
 '
-'  .5 = lower volume
-' 1.5 = higher volume
-
-Const VolBump   = .5   ' Bumpers volume.
-Const VolGates  = 1    ' Gates volume.
-Const VolMetal  = 1    ' Metals volume.
-Const VolRH     = 1    ' Rubber hits volume.
-Const VolPo     = 1    ' Rubber posts volume.
-Const VolPi     = 1    ' Rubber pins volume.
-Const VolTarg   = 1    ' Targets volume.
-Const VolSpin   = .5   ' Spinners volume.
-Const VolFlip   = 1    ' Flipper volume.
+' version 1.1:
+' - metal post added to "support" the Cyclotron
+' - insert lights corrected (thanks to Akiles for the new playfield photos)
 
+Option Explicit
 
 On Error Resume Next
 ExecuteGlobal GetTextFile("controller.vbs")
@@ -134,7 +111,7 @@
 
 'Ramp Up
 Sub Sol6(enabled)
-  PlaysoundAtVol SoundFX("solon",DOFContactors), P_BridgeRamp, .5
+  Playsound SoundFX("solon",DOFContactors),0,0.5,-0.9,0.15
   Ramp_Down.collidable = False
   if RampLimit <> RampUp then
     P_BridgeRamp.Rotx = RampUp + 15
@@ -146,7 +123,7 @@
 
 'Ramp Down
 Sub Sol7(enabled)
-  PlaysoundAtVol SoundFX("solon",DOFContactors), Primitive_MechArm, .5
+  Playsound SoundFX("solon",DOFContactors),0,0.5,-0.9,0.15
   Ramp_Down.collidable = True
   RampLimit = RampDown
   CycloneRampTimer.enabled = True
@@ -255,7 +232,7 @@
 '------------------------------
 Sub Drain_Hit()
   bsTrough.AddBall Me
-  PlaySoundAtVol "drain5", drain, 1
+  PlaySound "drain5"
 End Sub
 
 
@@ -289,8 +266,8 @@
 Sub Target22_Hit:vpmtimer.pulsesw(17):End Sub
 Sub Target23_Hit:vpmtimer.pulsesw(10):End Sub
 
-Sub Spinner14_Spin:vpmtimer.pulsesw(6):PlaySoundAtVol "soloff", Spinner14, VolSpin:End Sub
-Sub Spinner18_Spin:vpmtimer.pulsesw(7):PlaySoundAtVol "soloff", Spinner18, VolSpin:End Sub
+Sub Spinner14_Spin:vpmtimer.pulsesw(6):PlaySound "soloff", 0,0.2,0.8,0.5:End Sub
+Sub Spinner18_Spin:vpmtimer.pulsesw(7):PlaySound "soloff", 0,0.2,0.9,0.5:End Sub
 
 Sub SW13_Hit:Controller.Switch(31) = 1:End Sub
 Sub SW13_unhit:Controller.Switch(31) = 0:End Sub
@@ -298,15 +275,15 @@
 Sub SW30_unhit:Controller.Switch(37) = 0:End Sub
 
 Sub SW3_Hit:Controller.Switch(27) = 1:End Sub
-Sub SW3_unhit:Controller.Switch(27) = 0:PlaySoundAtVol "WireRampR",ActiveBall,.5:End Sub
+Sub SW3_unhit:Controller.Switch(27) = 0:PlaySound "WireRampR", 0,0.5,0.2,0.5:End Sub
 Sub SW4_Hit:Controller.Switch(25) = 1:End Sub
 Sub SW4_unhit:Controller.Switch(25) = 0:End Sub
 Sub SW5_Hit:Controller.Switch(26) = 1:LightCyclotronLamps:End Sub
 Sub SW5_unhit:Controller.Switch(26) = 0:End Sub
 
-Sub Bumper1_Hit:vpmtimer.pulsesw(1):PlaySoundAtVol SoundFX("fx_bumper4",DOFContactors),ActiveBall,VolBump:End Sub
-Sub Bumper2_Hit:vpmtimer.pulsesw(2):PlaySoundAtVol SoundFX("fx_bumper4",DOFContactors),ActiveBall,VolBump:End Sub
-Sub Bumper3_Hit:vpmtimer.pulsesw(3):PlaySoundAtVol SoundFX("fx_bumper4",DOFContactors),ActiveBall,VolBump:End Sub
+Sub Bumper1_Hit:vpmtimer.pulsesw(1):PlaySound SoundFX("fx_bumper4",DOFContactors), 0,1,-0.1,0.5:End Sub
+Sub Bumper2_Hit:vpmtimer.pulsesw(2):PlaySound SoundFX("fx_bumper4",DOFContactors), 0,1,0.4,0.5:End Sub
+Sub Bumper3_Hit:vpmtimer.pulsesw(3):PlaySound SoundFX("fx_bumper4",DOFContactors), 0,1,0.4,0.5:End Sub
 
 Sub LInlane_Hit:Controller.Switch(35) = 1:End Sub
 Sub LInlane_Unhit:Controller.Switch(35) = 0:End Sub
@@ -319,7 +296,7 @@
 
 Sub LaunchTrigger_Hit
   if activeball.vely < -4 Then
-    PlaysoundAtVol "Launch", LaunchTrigger, 1
+    Playsound "Launch",0,0.6,0.8
   end if
 End Sub
 
@@ -330,7 +307,7 @@
 Sub SSW_KeyDown(ByVal keycode)
   If keycode = PlungerKey Then
     Plunger.PullBack
-    PlaysoundAtVol "PlungerPull", Plunger, 1
+    Playsound "PlungerPull",0,1,0.8
   End If
   if keycode = StartGameKey and FreePlay then
     vpmTimer.PulseSw -3
@@ -343,7 +320,7 @@
 Sub SSW_KeyUp(ByVal keycode)
   If keycode = PlungerKey Then
     Plunger.Fire
-    PlaysoundAtVol "Plunger", Plunger, 1
+    Playsound "Plunger",0,1,0.8
   End If
 
   If KeyCode=LeftFlipperKey Then Controller.Switch(103)=0
@@ -388,7 +365,7 @@
 
 Sub RightSlingShot_Slingshot
   vpmtimer.pulsesw 5
-    PlaySoundAtVol SoundFX("rsling",DOFContactors), sling1, 1
+    PlaySound SoundFX("rsling",DOFContactors), 0, 1, 0.05, 0.05
     RSling.Visible = 0
     RSling1.Visible = 1
     sling1.TransZ = -20
@@ -406,7 +383,7 @@
 
 Sub LeftSlingShot_Slingshot
   vpmtimer.pulsesw 4
-    PlaySoundAtVol SoundFX("lsling",DOFContactors), sling2, 1
+    PlaySound SoundFX("lsling",DOFContactors),0,1,-0.05,0.05
     LSling.Visible = 0
     LSling1.Visible = 1
     sling2.TransZ = -20
@@ -447,6 +424,86 @@
 End Sub
 Set vpmShowDips=GetRef("editDips")
 
+
+
+' *********************************************************************
+'                      Supporting Ball & Sound Functions
+' *********************************************************************
+
+Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
+    Vol = Csng(BallVel(ball) ^2 / 300) * VolumeMultiplier
+End Function
+
+Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
+    Dim tmp
+    tmp = ball.x * 2 / SSW.width-1
+    If tmp > 0 Then
+        Pan = Csng(tmp ^10)
+    Else
+        Pan = Csng(-((- tmp) ^10) )
+    End If
+End Function
+
+Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
+    Pitch = BallVel(ball) * 20
+End Function
+
+Function BallVel(ball) 'Calculates the ball speed
+    BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
+End Function
+
+'*****************************************
+'      JP's VP10 Rolling Sounds
+'*****************************************
+
+Const tnob = 2 ' total number of balls
+ReDim rolling(tnob)
+InitRolling
+
+Sub InitRolling
+    Dim i
+    For i = 0 to tnob
+        rolling(i) = False
+    Next
+End Sub
+
+Sub RollingTimer_Timer()
+    Dim BOT, b
+    BOT = GetBalls
+
+  ' stop the sound of deleted balls
+    For b = UBound(BOT) + 1 to tnob
+        rolling(b) = False
+        StopSound("fx_ballrolling" & b)
+    Next
+
+  ' exit the sub if no balls on the table
+    If UBound(BOT) = -1 Then Exit Sub
+
+  ' play the rolling sound for each ball
+    For b = 0 to UBound(BOT)
+        If BallVel(BOT(b) ) > 1 AND BOT(b).z < 30 Then
+            rolling(b) = True
+            PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0
+        Else
+            If rolling(b) = True Then
+                StopSound("fx_ballrolling" & b)
+                rolling(b) = False
+            End If
+        End If
+    Next
+End Sub
+
+'**********************
+' Ball Collision Sound
+'**********************
+
+Sub OnBallBallCollision(ball1, ball2, velocity)
+  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, Pan(ball1), 0, Pitch(ball1), 0, 0
+End Sub
+
+
+
 '************************************
 ' What you need to add to your table
 '************************************
@@ -485,38 +542,38 @@
 
 
 Sub Pins_Hit (idx)
-  PlaySound "pinhit_low", 0, Vol(ActiveBall)*VolPi, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound "pinhit_low", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0
 End Sub
 
 Sub Targets_Hit (idx)
-  PlaySound SoundFX("target",DOFTargets), 0, Vol(ActiveBall)*VolTarg, Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound SoundFX("target",DOFTargets), 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0
 End Sub
 
 Sub Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall)*VolMetal, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall)*VolGates, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Spinner_Spin
-  PlaySoundAtVol "fx_spinner", Spinner, VolSpin
+  PlaySound "fx_spinner",0,.25,0,0.25
 End Sub
 
 Sub Rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -527,7 +584,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall)*VolPo, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -536,204 +593,29 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
 Sub LeftFlipper_Collide(parm)
-  ' RandomSoundFlipper()
-  Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(LeftFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(LeftFlipper)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(LeftFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(LeftFlipper)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(LeftFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(LeftFlipper)
-  End Select
+  RandomSoundFlipper()
 End Sub
 
 Sub RightFlipper_Collide(parm)
-  ' RandomSoundFlipper()
-  Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(RightFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(RightFlipper)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(RightFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(RightFlipper)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(RightFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(RightFlipper)
-  End Select
+  RandomSoundFlipper()
 End Sub
 
 Sub URightFlipper_Collide(parm)
-  ' RandomSoundFlipper()
-  Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(URightFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(URightFlipper)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(URightFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(URightFlipper)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(URightFlipper), 0, Pitch(ActiveBall), 1, 0, AudioFade(URightFlipper)
-  End Select
+  RandomSoundFlipper()
 End Sub
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall)*VolRH, Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX
-' PlaySound sound, 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
-
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volum)
-  PlaySound sound, 1, Volum, Pan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-'Set position as bumperX and Vol manually.
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, Pan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
-'                     Supporting Ball & Sound Functions
-'*********************************************************************
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "SSW" is the name of the table
-  Dim tmp
-  tmp = tableobj.y * 2 / SSW.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "SSW" is the name of the table
-  Dim tmp
-  tmp = tableobj.x * 2 / SSW.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "SSW" is the name of the table
-    Dim tmp
-    tmp = ball.x * 2 / SSW.width-1
-    If tmp > 0 Then
-        Pan = Csng(tmp ^10)
-    Else
-        Pan = Csng(-((- tmp) ^10) )
-    End If
-End Function
-
-Function AudioFade(ball) ' Can this be together with the above function ?
-  Dim tmp
-  tmp = ball.y * 2 / SSW.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / VolDiv)
-End Function
-
-Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
-End Function
-
-Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-'*****************************************
-'      JP's VP10 Rolling Sounds
-'*****************************************
-
-Const tnob = 2 ' total number of balls
-ReDim rolling(tnob)
-InitRolling
-
-Sub InitRolling
-    Dim i
-    For i = 0 to tnob
-        rolling(i) = False
-    Next
-End Sub
-
-Sub RollingTimer_Timer()
-    Dim BOT, b
-    BOT = GetBalls
-
-  ' stop the sound of deleted balls
-    For b = UBound(BOT) + 1 to tnob
-        rolling(b) = False
-        StopSound("fx_ballrolling" & b)
-    Next
-
-  ' exit the sub if no balls on the table
-    If UBound(BOT) = -1 Then Exit Sub
-
-  ' play the rolling sound for each ball
-
-    For b = 0 to UBound(BOT)
-      If BallVel(BOT(b) ) > 1 Then
-        rolling(b) = True
-        if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-        Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, Pan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-        End If
-      Else
-        If rolling(b) = True Then
-          StopSound("fx_ballrolling" & b)
-          rolling(b) = False
-        End If
-      End If
-    Next
-End Sub
-
-'**********************
-' Ball Collision Sound
-'**********************
-
-Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / (VolDiv/VolCol), Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
-End Sub
-
-' Thalamus : Exit in a clean and proper way
-Sub SSW_exit()
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
