--- vpxtable_scripts/The Web_VPX_1_2.vbs
+++ todo_extracted_tables/The Web_VPX_1_2.vbs
@@ -86,12 +86,10 @@
 
 Sub TheWeb_Exit()
   If B2SOn = True Then Controller.Stop
-
   If UltraDMD.IsRendering Then
     UltraDMD.CancelRendering
   end if
     UltraDMD = NULL
-  'If Not FlexDMD is Nothing Then FlexDMD.Run = False
 
   '------ UltraDMD Replace Default Color preference ------
   'code provided by Gigalula
@@ -6218,7 +6216,6 @@
       If keycode = LeftTiltKey Then:Nudge 90, 2:checkNudge:end if
       If keycode = RightTiltKey Then:Nudge 270, 2:checkNudge:end if
       If keycode = CenterTiltKey Then:Nudge 0, 2:checkNudge:end if
-      If keycode = MechanicalTilt Then checkNudge:end if
     end if
   end if
 End Sub
@@ -6524,18 +6521,7 @@
 End Sub
 
 Sub LoadUltraDMD
-    ' Set UltraDMD = CreateObject("UltraDMD.DMDObject")
-
-    Dim FlexDMD
-    Set FlexDMD = CreateObject("FlexDMD.FlexDMD")
-    If FlexDMD is Nothing Then
-        MsgBox "No UltraDMD found.  This table will NOT run without it."
-        Exit Sub
-    End If
-    FlexDMD.GameName = cGameName
-    FlexDMD.RenderMode = 2
-    Set UltraDMD = FlexDMD.NewUltraDMD()
-
+    Set UltraDMD = CreateObject("UltraDMD.DMDObject")
     UltraDMD.Init
     Set fso = CreateObject("Scripting.FileSystemObject")
     curDir = fso.GetAbsolutePathName(".")
@@ -7279,7 +7265,7 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, Pan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(BOT(b))
 End Sub
 
 '************************************
