--- vpxtable_scripts/Tiki Bob's Atomic Beach Party 2.0 VR_FlexDMD.vbs
+++ todo_extracted_tables/Tiki Bob's Atomic Beach Party 2.0 VR_FlexDMD.vbs
@@ -2632,10 +2632,7 @@
                 "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
                 if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched : ' Thalamus - patched :                 aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+                aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
                 if debugOn then TBPout.text = str
         End Sub
 
