--- vpxtable_scripts/Tornado Rally (original 2024).vbs
+++ todo_extracted_tables/Tornado Rally (original 2024).vbs
@@ -200,11 +200,11 @@
 Dim xx, bsTrough, initContacts, dtRDrop, dtLDrop, bsLSaucer, bsMSaucer, bsRSaucer
 
 
-Sub Table1_Paused:Controller.Pause = 1:End Sub
+Sub OKIES_Paused:Controller.Pause = 1:End Sub
 
 Sub OKIESSpiderMan_unPaused:Controller.Pause = 0:End Sub
 
-Sub Table1_Exit
+Sub OKIES_Exit
     If b2son then controller.stop
 End Sub
 
@@ -3638,12 +3638,11 @@
     tmp =  - 7000
   End If
 
-' Thalamus, AudioFade patched
-	If tmp > 0 Then
-		AudioFade = CSng(tmp ^ 5) 'was 10
-	Else
-		AudioFade = CSng( - (( - tmp) ^ 5) ) 'was 10
-	End If
+  If tmp > 0 Then
+    AudioFade = CSng(tmp ^ 10)
+  Else
+    AudioFade = CSng( - (( - tmp) ^ 10) )
+  End If
 End Function
 
 Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
@@ -3656,12 +3655,11 @@
     tmp =  - 7000
   End If
 
-' Thalamus, AudioPan patched
-	If tmp > 0 Then
-		AudioPan = CSng(tmp ^ 5) 'was 10
-	Else
-		AudioPan = CSng( - (( - tmp) ^ 5) ) 'was 10
-	End If
+  If tmp > 0 Then
+    AudioPan = CSng(tmp ^ 10)
+  Else
+    AudioPan = CSng( - (( - tmp) ^ 10) )
+  End If
 End Function
 
 Function Vol(ball) ' Calculates the volume of the sound based on the ball speed
