--- vpxtable_scripts/Unreal_Tournament_99_Capture_the_Flag_1.0.vbs
+++ todo_extracted_tables/Unreal_Tournament_99_Capture_the_Flag_1.0.vbs
@@ -5148,7 +5148,7 @@
 
   If keycode = 19 then ScoreCard=1 : CardTimer.enabled=1
 
-  If keycode = PlungerKey or keycode = LockBarKey Then
+  If keycode = PlungerKey Then
     If AutoKick=0 and RespawnWaiting>0  Then
       AutoKick=1
       L061.timerenabled=0
@@ -5231,7 +5231,6 @@
     If keycode = LeftTiltKey   Then Nudge  90, 5 : SoundNudgeLeft()   : Tilting
     If keycode = RightTiltKey  Then Nudge 270, 5 : SoundNudgeRight()  : Tilting
     If keycode = CenterTiltKey Then Nudge   0, 3 : SoundNudgeCenter() : Tilting
-    If keycode = MechanicalTilt Then Tilting
   End If
 
     If keycode = AddCreditKey or keycode = AddCreditKey2 Then
@@ -6423,8 +6422,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     if debugOn then TBPout.text = str
   End Sub
 
