--- vpxtable_scripts/Vernes World (Spinball 1996) v600.vbs
+++ todo_extracted_tables/Vernes World (Spinball 1996) v600.vbs
@@ -98,15 +98,12 @@
         .Balls = 4
     End With
 
-    ' Thalamus, more randomness pls
     Set bsVUK = New cvpmBallStack
     With bsVUK
         .InitSaucer sw67, 67, 0, 58
         .InitExitSnd SoundFX("fx_kicker", DOFContactors), SoundFX("fx_Solenoid", DOFContactors)
         .InitAddSnd "fx_kicker_enter"
         .KickZ = 1.56
-        .KickForceVar = 3
-        .KickAngleVar = 3
     End With
 
     Set bsSVUK = New cvpmBallStack
@@ -117,8 +114,6 @@
         .InitExitSnd SoundFX("fx_kicker", DOFContactors), SoundFX("fx_Solenoid", DOFContactors)
         .InitAddSnd "fx_kicker_enter"
         .KickBalls = 1
-        .KickForceVar = 3
-        .KickAngleVar = 3
     End With
 
     ' Drop targets
@@ -176,7 +171,7 @@
     If Keycode = keyInsertCoin3 then vpmTimer.AddTimer 750, "vpmTimer.PulseSw swCoin1'"
     If keycode = keyFront Then Controller.Switch(75) = 1
     If vpmKeyDown(keycode) Then Exit Sub
-    If keycode = PlungerKey or keycode = LockBarKey Then Controller.Switch(86) = 1
+    If keycode = PlungerKey Then Controller.Switch(86) = 1
 End Sub
 
 Sub table1_KeyUp(ByVal Keycode)
@@ -184,7 +179,7 @@
     If Keycode = RightFlipperKey then Controller.Switch(131) = 0
     If keycode = keyFront Then Controller.Switch(75) = 0
     If vpmKeyUp(keycode) Then Exit Sub
-    If keycode = PlungerKey or keycode = LockBarKey Then Controller.Switch(86) = 0
+    If keycode = PlungerKey Then Controller.Switch(86) = 0
 End Sub
 
 '*************************
@@ -809,10 +804,3 @@
         Case 21:table1.ColorGradeImage = "LUT Warm 10"
     End Select
 End Sub
-
-' Thalamus : Exit in a clean and proper way
-Sub Table1_exit
-  Controller.Pause = False
-  Controller.Stop
-End Sub
-
