--- vpxtable_scripts/Vulcan (Gottlieb 1977).vbs
+++ todo_extracted_tables/Vulcan (Gottlieb 1977).vbs
@@ -1,9 +1,9 @@
 ' -------------------------------------------------
-'  VULCAN  Gottlieb 1977
-'  -------------------------------------------------
+' VULCAN  Gottlieb 1977
+' -------------------------------------------------
 '
 ' Original table by D. Gottlieb & Co., October 1977
-'  VP   Vulcan adaptation by Dave Sanders, December 2002
+' VP   Vulcan adaptation by Dave Sanders, December 2002
 ' VP10 Vulcan adaptation started by MaX and hauntfreaks, July 2015
 '   VP10 Fire Queen and Vulcan by hauntfreaks and BorgDog, Sept 2015
 ' DOF coded by BorgDog with instruction(and correction) by arngrim - thanks!
@@ -227,7 +227,7 @@
 Sub Vulcan_KeyDown(ByVal keycode)
 
   if keycode=AddCreditKey then
-    PlaySoundAtVol "coinin", Drain, 1
+    playsound "coinin"
     coindelay.enabled=true
     end if
 
@@ -272,7 +272,7 @@
 
   If keycode = PlungerKey Then
     Plunger.PullBack
-    PlaySoundAtVol "plungerpull", Plunger, 1
+    PlaySound "plungerpull",0,1,0.25,0.25
   End If
   If keycode=LeftFlipperKey and State = false and OperatorMenu=0 then
     OperatorMenuTimer.Enabled = true
@@ -327,14 +327,14 @@
   if tilt=false and state=true then
   If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToEnd
-    PlaySoundAtVol SoundFXDOF("flipperup",101,DOFOn,DOFContactors), LeftFlipper, 1
-    PlayLoopSoundAtVol "Buzz", LeftFlipper,1
+    PlaySound SoundFXDOF("flipperup",101,DOFOn,DOFContactors), 0, .67, -0.05, 0.05
+    PlaySound "Buzz",-1,.05,-0.05, 0.05
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToEnd
-    PlaySoundAtVol SoundFXDOF("flipperup",102,DOFOn,DOFContactors), ActiveBall, 1
-    PlayLoopSoundAtVol "Buzz1", RightFlipper, 1
+    PlaySound SoundFXDOF("flipperup",102,DOFOn,DOFContactors), 0, .67, 0.05, 0.05
+    PlaySound "Buzz1",-1,.05,0.05,0.05
   End If
 
   If keycode = LeftTiltKey Then
@@ -351,11 +351,6 @@
     Nudge 0, 2
     checktilt
   End If
-
-  If keycode = MechanicalTilt Then
-    checktilt
-  End If
-
   end if
 End Sub
 
@@ -383,7 +378,7 @@
 
   If keycode = PlungerKey Then
     Plunger.Fire
-    PlaySoundAtVol "plunger", Plunger, 1
+    PlaySound "plunger",0,1,0.25,0.25
   End If
 
     if keycode = LeftFlipperKey then
@@ -393,13 +388,13 @@
    If tilt=false and state=true then
   If keycode = LeftFlipperKey Then
     LeftFlipper.RotateToStart
-    PlaySoundAtVol SoundFXDOF("flipperdown",101,DOFOff,DOFContactors), LeftFlipper, 1
+    PlaySound SoundFXDOF("flipperdown",101,DOFOff,DOFContactors), 0, 1, -0.05, 0.05
     StopSound "Buzz"
   End If
 
   If keycode = RightFlipperKey Then
     RightFlipper.RotateToStart
-    PlaySoundAtVol SoundFXDOF("flipperdown",102,DOFOff,DOFContactors), LeftFlipper, 1
+    PlaySound SoundFXDOF("flipperdown",102,DOFOff,DOFContactors), 0, 1, 0.05, 0.05
     StopSound "Buzz1"
   End If
    End if
@@ -473,7 +468,7 @@
 
 Sub Drain_Hit()
   DOF 129, DOFPulse
-  PlaySoundAtVol "drain", Drain, 1
+  PlaySound "drain",0,1,0,0.25
   Drain.DestroyBall
   me.timerenabled=1
 End Sub
@@ -581,7 +576,7 @@
   matchtxt.text=" "
     BallRelease.CreateBall
     BallRelease.kick 135,4,0
-  PlaySoundAtVol SoundFXDOF("kickerkick",128,DOFPulse,DOFContactors), BallRelease, 1
+  playsound SoundFXDOF("kickerkick",128,DOFPulse,DOFContactors)
 end sub
 
 sub newball
@@ -599,13 +594,13 @@
 End Sub
 
 Sub resetWT
-  PlaySoundAtVol SoundFXDOF("BankReset",111,DOFPulse,DOFContactors), drop1, 1
+  playsound SoundFXDOF("BankReset",111,DOFPulse,DOFContactors)
   for each objekt in droptargetsWhite: objekt.isdropped=false: Next
   for each light in WDTlights:light.state=0:next
 end sub
 
 Sub resetDT
-  PlaySoundAtVol SoundFXDOF("BankReset",114,DOFPulse,DOFContactors), drop1, 1
+  playsound SoundFXDOF("BankReset",114,DOFPulse,DOFContactors)
   DOF 114, DOFPulse
   for each objekt in droptargets: objekt.isdropped=false: Next
   for each light in DTlights:light.state=0:next
@@ -697,7 +692,7 @@
 
 Sub Bumper1_Hit
    if tilt=false then
-  PlaySoundAtVol SoundFXDOF("fx_bumper4",107,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("fx_bumper4",107,DOFPulse,DOFContactors)
   DOF 108,DOFPulse
   if bumper1light.state=1 then
     addscore bumperlitscore
@@ -725,7 +720,7 @@
 
 Sub Bumper2_Hit
    if tilt=false then
-  PlaySoundAtVol SoundFXDOF("fx_bumper4",109,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("fx_bumper4",109,DOFPulse,DOFContactors)
   DOF 110,DOFPulse
   if bumper2light.state=1 then
     addscore bumperlitscore
@@ -771,14 +766,14 @@
 
 sub SpotTarget_hit
   spot5.state=0
-  PlaySoundAtVol SoundFXDOF("target",135,DOFPulse,DOFContactors), ActiveBall, 1
+  PlaySound SoundFXDOF("target",135,DOFPulse,DOFContactors),0,.5
   if balls = 3 then
     addscore 3000
     else
     addscore 500
   end if
   if extraball.state=1 then
-    PlaySoundAtVol SoundFXDOF("bell10",141,DOFPulse,DOFChimes), ActiveBall, 1
+    PlaySound SoundFXDOF("bell10",141,DOFPulse,DOFChimes),0,.5
     if ebcount=0 then
       shootagain.state=lightstateon
       extraball.state=0
@@ -800,7 +795,7 @@
 
 Sub TopHole_timer
   TopHole.Kick 205, 12
-  PlaySoundAtVol SoundFXDOF("ballrelease",132,DOFPulse,DOFContactors), TopHole, 1
+  playsound SoundFXDOF("ballrelease",132,DOFPulse,DOFContactors),0,1,0,0.25
   DOF 131, DOFPulse
   me.timerenabled=0
 End Sub
@@ -868,7 +863,7 @@
   DOF 124,DOFPulse
   flashbumpers
     if LeftSP.state=1 then
-    PlaySoundAtVol SoundFXDOF("knock",130,DOFPulse,DOFKnocker), ActiveBall, 1
+    playsound SoundFXDOF("knock",130,DOFPulse,DOFKnocker)
     DOF 131,DOFPulse
     addcredit
     end if
@@ -915,7 +910,7 @@
   DOF 126,DOFPulse
   flashbumpers
     if LeftSP.state=1 then
-    PlaySoundAtVol SoundFXDOF("knock",130,DOFPulse,DOFKnocker), ActiveBall, 1
+    playsound SoundFXDOF("knock",130,DOFPulse,DOFKnocker)
     DOF 131,DOFPulse
     addcredit
     end if
@@ -1006,7 +1001,7 @@
 End Sub
 
 Sub W1a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",112,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",112,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   If Drop1.State = 1 then
@@ -1018,7 +1013,7 @@
 End Sub
 
 Sub W2a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",112,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",112,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   If Drop2.State = 1 then
@@ -1030,7 +1025,7 @@
 End Sub
 
 Sub W3a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",112,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",112,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   If Drop3.State = 1 then
@@ -1043,7 +1038,7 @@
 End Sub
 
 Sub W4a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",112,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",112,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   If Drop4.State = 1 then
@@ -1055,7 +1050,7 @@
 End Sub
 
 Sub W5a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",112,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",112,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   If Drop5.State = 1 then
@@ -1067,7 +1062,7 @@
 End Sub
 
 Sub G1a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",113,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",113,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   Hole1.state =1
@@ -1075,7 +1070,7 @@
 End Sub
 
 Sub G2a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",113,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",113,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   Hole2.state =1
@@ -1084,7 +1079,7 @@
 End Sub
 
 Sub G3a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",113,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",113,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   Hole3.state =1
@@ -1093,7 +1088,7 @@
 End Sub
 
 Sub G4a_Hit
-  PlaySoundAtVol SoundFXDOF("drop1",113,DOFPulse,DOFContactors), ActiveBall, 1
+  playsound SoundFXDOF("drop1",113,DOFPulse,DOFContactors)
   addscore 500
   addbonus
   Hole4.state =1
@@ -1230,180 +1225,37 @@
   if bonus>10 then Bonuslight(10).state=1
 End sub
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
-
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
-
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
-'            Supporting Ball, Sound Functions and Math
-'*********************************************************************
-
-Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
-End Function
-
-Const Pi = 3.1415927
-
-Function dSin(degrees)
-  dsin = sin(degrees * Pi/180)
-End Function
-
-Function dCos(degrees)
-  dcos = cos(degrees * Pi/180)
-End Function
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "Vulcan" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / Vulcan.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "Vulcan" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / Vulcan.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "Vulcan" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / Vulcan.width-1
-  If tmp > 0 Then
-    Pan = Csng(tmp ^10)
-  Else
-    Pan = Csng(-((- tmp) ^10) )
-  End If
-End Function
+' *********************************************************************
+'                      Supporting Ball & Sound Functions
+' *********************************************************************
 
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
+    Vol = Csng(BallVel(ball) ^2 / 500)
 End Function
 
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-End Function
-
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
-End Function
-
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
+Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
+    Dim tmp
+    tmp = ball.x * 2 / Vulcan.width-1
+    If tmp > 0 Then
+        Pan = Csng(tmp ^10)
+    Else
+        Pan = Csng(-((- tmp) ^10) )
+    End If
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
+    Pitch = BallVel(ball) * 20
 End Function
 
 Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
-End Function
-
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
+    BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
 End Function
 
 '*****************************************
 '      JP's VP10 Rolling Sounds
 '*****************************************
 
-Const tnob = 5 ' total number of balls
+Const tnob = 2 ' total number of balls
 ReDim rolling(tnob)
 InitRolling
 
@@ -1428,23 +1280,16 @@
     If UBound(BOT) = -1 Then Exit Sub
 
   ' play the rolling sound for each ball
-
     For b = 0 to UBound(BOT)
-      If BallVel(BOT(b) ) > 1 Then
-        rolling(b) = True
-        if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-          ' PlaySound("fx_ballrolling" & b), -1, BallRollVol(BOT(b) )*.8, AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-        Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-          ' PlaySound("fx_ballrolling" & b), -1, BallRollVol(BOT(b) )*.2, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-        End If
-      Else
-        If rolling(b) = True Then
-          StopSound("fx_ballrolling" & b)
-          rolling(b) = False
+        If BallVel(BOT(b) ) > 1 AND BOT(b).z < 30 Then
+            rolling(b) = True
+            PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0
+        Else
+            If rolling(b) = True Then
+                StopSound("fx_ballrolling" & b)
+                rolling(b) = False
+            End If
         End If
-      End If
     Next
 End Sub
 
@@ -1453,10 +1298,9 @@
 '**********************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, Pan(ball1), 0, Pitch(ball1), 0, 0
 End Sub
 
-
 '************************************
 ' What you need to add to your table
 '************************************
@@ -1495,26 +1339,26 @@
 
 
 Sub Metals_Thin_Hit (idx)
-  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_thin", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Metals_Medium_Hit (idx)
-  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Metals2_Hit (idx)
-  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Gates_Hit (idx)
-  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Rubbers_Hit(idx)
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1525,7 +1369,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 20 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 20 then
     RandomSoundRubber()
@@ -1536,7 +1380,7 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 16 then
     RandomSoundRubber()
@@ -1545,9 +1389,9 @@
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "rubber_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "rubber_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
@@ -1561,9 +1405,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "flip_hit_2", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "flip_hit_3", 0, Vol(ActiveBall), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
