--- vpxtable_scripts/Whirlwind 4K 1.11.vbs
+++ todo_extracted_tables/Whirlwind 4K 1.11.vbs
@@ -2801,8 +2801,7 @@
     "actual cor: " & round(realCOR,4) & vbnewline & "ballspeed coef: " & round(coef, 3) & vbnewline
     if Print then debug.print Round(cor.ballvel(aBall.id),2) & ", " & round(desiredcor,3)
 
-' Thalamus - patched :     aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
-    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef : aBall.velz = aBall.velz * coef
+    aBall.velx = aBall.velx * coef : aBall.vely = aBall.vely * coef
     'playsound "fx_knocker"
     if debugOn then TBPout.text = str
   End Sub
