--- vpxtable_scripts/World_Cup_Soccer_1.2.vbs
+++ todo_extracted_tables/World_Cup_Soccer_1.2.vbs
@@ -4,7 +4,6 @@
 ' Thalamus 2018-07-24
 ' Table doesn't have "Positional Sound Playback Functions" or "Supporting Ball & Sound Functions"
 ' Changed UseSolenoids=1 to 2
-' Thalamus 2018-12-17 : Added FFv2
 ' No special SSF tweaks yet.
 
 Dim DesktopMode: DesktopMode = tablewpc94.ShowDT
@@ -25,9 +24,6 @@
 
 LoadVPM "01530000", "WPC.VBS", 3.10
 
-' Thalamus - for Fast Flip v2
-NoUpperRightFlipper
-NoUpperLeftFlipper
 
  Sub LoadVPM(VPMver, VBSfile, VBSver)   'Add new call to InitializeOptions to allow selection of controller through F6 menu
     On Error Resume Next
@@ -157,8 +153,8 @@
     Set bsTV = New cvpmBallStack
         bsTV.InitSw 0,swTVBallPopper,0,0,0,0,0,0
         bsTV.InitKick TVBallPopper, 213, 10
-        bsTV.KickAngleVar = 3
-        bsTV.KickForceVar = 3
+    bsTV.KickAngleVar = 3
+    bsTV.KickForceVar = 3
         bsTV.KickBalls = 2
         bsTV.InitExitSnd Popper,SolOn
 
@@ -850,12 +846,10 @@
  End Sub
 
 
-' Thalamus : This sub is used twice - this means ... this one IS NOT USED
-' Not a issue though, they are the same
-' Sub UpdateFlipperLogos
-'     LFLogo.RotAndTra2 = LeftFlipper.CurrentAngle
-'     RFlogo.RotAndTra2 = RightFlipper.CurrentAngle
-' End Sub
+Sub UpdateFlipperLogos
+    LFLogo.RotAndTra2 = LeftFlipper.CurrentAngle
+    RFlogo.RotAndTra2 = RightFlipper.CurrentAngle
+End Sub
 
 Sub SolRampDiverter(enabled)
     if enabled then
