--- vpxtable_scripts/Yello Pinball Cha Cha (2021).vbs
+++ todo_extracted_tables/Yello Pinball Cha Cha (2021).vbs
@@ -383,7 +383,7 @@
 
 Sub Table1_KeyDown(ByVal keycode)
   If vpmKeyDown(keycode) Then Exit Sub
-    If keycode = PlungerKey Then PlaySoundAtVol "plungerpull", Plunger, 1:Plunger.Pullback
+    If keycode = PlungerKey Then PlaySound "plungerpull", 0, 1, 0.1, 0.25:Plunger.Pullback
   If keycode = LeftFlipperKey Then flipnf 0, 1: exit sub : End If
   If keycode = RightFlipperKey Then flipnf 1, 1 : controller.Switch(59) = 1 : exit sub : End If
   if Keycode = KeyRules then
@@ -403,10 +403,10 @@
     If keycode = PlungerKey Then
     Plunger.Fire
     if BallInPlunger then
-      If keycode = PlungerKey Then PlaySoundAtVol "plunger3", Plunger, 1:Plunger.Fire
+      If keycode = PlungerKey Then PlaySound "plunger3", 0, 1, 0.1, 0.25:Plunger.Fire
       FlashForMs yBoxFlasher, 2000, 40, 0
     Else
-      If keycode = PlungerKey Then PlaySoundAtVol "plunger3", Plunger, 1:Plunger.Fire
+      If keycode = PlungerKey Then PlaySound "plunger3", 0, 1, 0.1, 0.25:Plunger.Fire
       FlashForMs yBoxFlasher, 2000, 40, 0
     end if
   End If
@@ -438,7 +438,7 @@
   if Enabled then
     sw46.Kick 58, 16 :
     Tgate.Open = True
-    If sw46.BallCntOver > 0 Then PlaySoundAtVol SoundFX("Trough1",DOFcontactors), ActiveBall, .5: End If
+    If sw46.BallCntOver > 0 Then Playsound SoundFX("Trough1",DOFcontactors), 0, LVL(0.5), 0.01 : End If
   Else
     Tgate.Open = False
     BallSearch
@@ -447,7 +447,7 @@
 
 Sub TroughOut(enabled)
   if Enabled then
-    sw47.Kick 58, 8 :PlaySoundAtVol SoundFX("BallRelease",DOFcontactors), ActiveBall, .4
+    sw47.Kick 58, 8 :Playsound SoundFX("BallRelease",DOFcontactors), 0, LVL(0.4), 0.02
   End If
 end Sub
 
@@ -455,9 +455,9 @@
 '****           Ball Ramp Trough Switches                 ****
 '***********************************************************************************
 
-Sub Sw46_hit():PlaySoundAtVol "fx_drain", Drain, 1:controller.Switch(46) = 1 : End Sub 'Drain
+Sub Sw46_hit():Playsound "fx_drain":controller.Switch(46) = 1 : End Sub 'Drain
 'Sub Drain_Hit:Playsound "fx_drain":bsTrough.AddBall Me:End Sub
-Sub TroughSFX_Hit(): PlaySoundAtVol "Trough2", ActiveBall, .2 : End Sub
+Sub TroughSFX_Hit(): Playsound "Trough2", 0, LVL(0.2), 0:: End Sub
 Sub Sw49_hit():controller.Switch(49) = 1 : UpdateTrough : End Sub
 Sub Sw48_hit():controller.Switch(48) = 1 : UpdateTrough : End Sub
 Sub Sw47_hit():controller.Switch(47) = 1 : UpdateTrough : End Sub
@@ -487,24 +487,24 @@
 
 'Primitives from Victory (Gottlieb 1987)
 
-Sub sw9_Hit():PlaySoundAtVol "Sensor", ActiveBall, 1:Controller.Switch(9) = 1:sw9p.Visible = 0:End Sub
+Sub sw9_Hit():PlaySound "Sensor":Controller.Switch(9) = 1:sw9p.Visible = 0:End Sub
 Sub sw9_UnHit():Controller.Switch(9) = 0:sw9p.Visible = 1:End Sub
-Sub sw10_Hit():PlaySoundAtVol "Sensor2", ActiveBall, 1:Controller.Switch(10) = 1:sw10p.Visible = 0:End Sub
+Sub sw10_Hit():PlaySound "Sensor2":Controller.Switch(10) = 1:sw10p.Visible = 0:End Sub
 Sub sw10_UnHit():Controller.Switch(10) = 0:sw10p.Visible = 1:End Sub
-Sub sw11_Hit():PlaySoundAtVol "Sensor3", ActiveBall, 1:Controller.Switch(11) = 1:sw11p.Visible = 0:End Sub
+Sub sw11_Hit():PlaySound "Sensor3":Controller.Switch(11) = 1:sw11p.Visible = 0:End Sub
 Sub sw11_UnHit():Controller.Switch(11) = 0:sw11p.Visible = 1:End Sub
-Sub sw12_Hit():PlaySoundAtVol "Sensor4", ActiveBall, 1:Controller.Switch(12) = 1:sw12p.Visible = 0:End Sub
+Sub sw12_Hit():PlaySound "Sensor4":Controller.Switch(12) = 1:sw12p.Visible = 0:End Sub
 Sub sw12_UnHit():Controller.Switch(12) = 0:sw12p.Visible = 1:End Sub
 
-Sub sw45_Hit():PlaySoundAtVol "Sensor5", ActiveBall, 1:Controller.Switch(45) = 1:sw45p.Visible = 0:End Sub
+Sub sw45_Hit():PlaySound "Sensor5":Controller.Switch(45) = 1:sw45p.Visible = 0:End Sub
 Sub sw45_UnHit():Controller.Switch(45) = 0:sw45p.Visible = 1:End Sub
-Sub sw51_Hit():PlaySoundAtVol "Sensor6", ActiveBall, 1:Controller.Switch(51) = 1:sw51p.Visible = 0:End Sub
+Sub sw51_Hit():PlaySound "Sensor6":Controller.Switch(51) = 1:sw51p.Visible = 0:End Sub
 Sub sw51_UnHit():Controller.Switch(51) = 0:sw51p.Visible = 1:End Sub
-Sub sw52_Hit():PlaySoundAtVol "Sensor7", ActiveBall, 1:Controller.Switch(52) = 1:sw52p.Visible = 0:End Sub
+Sub sw52_Hit():PlaySound "Sensor7":Controller.Switch(52) = 1:sw52p.Visible = 0:End Sub
 Sub sw52_UnHit():Controller.Switch(52) = 0:sw52p.Visible = 1:End Sub
-Sub sw53_Hit():PlaySoundAtVol "Sensor8", ActiveBall, 1:Controller.Switch(53) = 1:sw53p.Visible = 0:End Sub
+Sub sw53_Hit():PlaySound "Sensor8":Controller.Switch(53) = 1:sw53p.Visible = 0:End Sub
 Sub sw53_UnHit():Controller.Switch(53) = 0:sw53p.Visible = 1:End Sub
-Sub sw54_Hit():PlaySoundAtVol "Sensor9", ActiveBall, 1:Controller.Switch(54) = 1:sw54p.Visible = 0:End Sub
+Sub sw54_Hit():PlaySound "Sensor9":Controller.Switch(54) = 1:sw54p.Visible = 0:End Sub
 Sub sw64_Hit:Controller.Switch(64) = 1:PlaySound "guitarRiff",0,1,-0.15,0.25:End Sub
 Sub sw64_Unhit:Controller.Switch(64) = 0:ActiveBall.VelY= ActiveBall.VelY/3:End Sub
 
@@ -513,22 +513,22 @@
 '****               Stand Up Targets                  ****
 '***********************************************************************************
 
-Sub sw18_hit:vpmTimer.pulseSw 18:PlaySoundAtVol "target", ActiveBall,  .1:End Sub
-Sub sw19_hit:vpmTimer.pulseSw 19:PlaySoundAtVol "target2", ActiveBall, .1:End Sub
-Sub sw20_hit:vpmTimer.pulseSw 20:PlaySoundAtVol "target3", ActiveBall, .1:End Sub
-Sub sw21_hit:vpmTimer.pulseSw 21:PlaySoundAtVol "target4", ActiveBall, .1:End Sub
-Sub sw22_hit:vpmTimer.pulseSw 22:PlaySoundAtVol "target5", ActiveBall, .1:End Sub
-
-Sub sw28_hit:vpmTimer.pulseSw 28:PlaySoundAtVol "target6", ActiveBall, .1:End Sub
-Sub sw29_hit:vpmTimer.pulseSw 29:PlaySoundAtVol "target7", ActiveBall, .1:End Sub
-Sub sw30_hit:vpmTimer.pulseSw 30:PlaySoundAtVol "target8", ActiveBall, .1:End Sub
-Sub sw31_hit:vpmTimer.pulseSw 31:PlaySoundAtVol "target9", ActiveBall, .1:End Sub
-Sub sw32_hit:vpmTimer.pulseSw 32:PlaySoundAtVol "target10", ActiveBall, .1:End Sub
+Sub sw18_hit:vpmTimer.pulseSw 18:PlaySound "target", 0, LVL(0.1), 0.3:End Sub
+Sub sw19_hit:vpmTimer.pulseSw 19:PlaySound "target2", 0, LVL(0.1), 0.3:End Sub
+Sub sw20_hit:vpmTimer.pulseSw 20:PlaySound "target3", 0, LVL(0.1), 0.3:End Sub
+Sub sw21_hit:vpmTimer.pulseSw 21:PlaySound "target4", 0, LVL(0.1), 0.3:End Sub
+Sub sw22_hit:vpmTimer.pulseSw 22:PlaySound "target5", 0, LVL(0.1), 0.3:End Sub
+
+Sub sw28_hit:vpmTimer.pulseSw 28:PlaySound "target6", 0, LVL(0.1), 0.3:End Sub
+Sub sw29_hit:vpmTimer.pulseSw 29:PlaySound "target7", 0, LVL(0.1), 0.3:End Sub
+Sub sw30_hit:vpmTimer.pulseSw 30:PlaySound "target8", 0, LVL(0.1), 0.3:End Sub
+Sub sw31_hit:vpmTimer.pulseSw 31:PlaySound "target9", 0, LVL(0.1), 0.3:End Sub
+Sub sw32_hit:vpmTimer.pulseSw 32:PlaySound "target10", 0, LVL(0.1), 0.3:End Sub
 
-Sub sw36_hit:vpmTimer.pulseSw 36:PlaySoundAtVol "target11", ActiveBall, .1:End Sub
-Sub sw37_hit:vpmTimer.pulseSw 37:PlaySoundAtVol "target12", ActiveBall, .1:End Sub
+Sub sw36_hit:vpmTimer.pulseSw 36:PlaySound "target11", 0, LVL(0.1), 0.3:End Sub
+Sub sw37_hit:vpmTimer.pulseSw 37:PlaySound "target12", 0, LVL(0.1), 0.3:End Sub
 
-Sub sw41_hit:vpmTimer.pulseSw 41:PlaySoundAtVol "target13", ActiveBall, .1:End Sub
+Sub sw41_hit:vpmTimer.pulseSw 41:PlaySound "target13", 0, LVL(0.1), 0.3:End Sub
 
 
 '***********************************************************************************
@@ -558,14 +558,14 @@
 Sub sw39_Dropped:dtBPod.Hit 1 : End Sub
 Sub sw40_Dropped:dtTPod.Hit 1 : End Sub
 
-Sub Sw33a_Hit() : PlaySoundAtVol "soloff", ActiveBall, .2 : End Sub 'dont drop target but play sound
-Sub Sw34a_Hit() : PlaySoundAtVol "soloff2", ActiveBall, .2 : End Sub 'dont drop target but play sound
+Sub Sw33a_Hit() : Playsound "soloff", 0, LVL(0.2), 0:: End Sub 'dont drop target but play sound
+Sub Sw34a_Hit() : Playsound "soloff2", 0, LVL(0.2), 0:: End Sub 'dont drop target but play sound
 
 '***********************************************************************************
 '****               Switches                      ****
 '***********************************************************************************
 
-Sub sw38_hit:vpmTimer.pulseSw 38 : PlaySoundAtVol "soloff", ActiveBall, .1:End Sub
+Sub sw38_hit:vpmTimer.pulseSw 38 : PlaySound "soloff", 0, LVL(0.1), 0.3:End Sub
 
 '***********************************************************************************
 '****               Bumper                          ****
@@ -574,16 +574,16 @@
 Const swJet1 = 55 ' Left Jet Bumper
 Const swJet2 = 56 ' Right Jet Bumper
 
-Sub Bumper1_Hit():vpmTimer.PulseSwitch (swJet1), 0, "" : PlaySoundAtVol SoundFX("LeftBumper_Hit",DOFContactors), ActiveBall, 1
+Sub Bumper1_Hit():vpmTimer.PulseSwitch (swJet1), 0, "" : PlaySound SoundFX("LeftBumper_Hit",DOFContactors)
   FlashForMs yBoxFlasher, 100, 2, 0  : End Sub
-Sub Bumper2_Hit():vpmTimer.PulseSwitch (swJet2), 0, "" : PlaySoundAtVol SoundFX("RightBumper_Hit",DOFContactors), ActiveBall, 1
+Sub Bumper2_Hit():vpmTimer.PulseSwitch (swJet2), 0, "" : PlaySound SoundFX("RightBumper_Hit",DOFContactors)
   FlashForMs yBoxFlasher, 100, 2, 0 : End Sub
 
 '***********************************************************************************
 '****               Drain hole & Kicker               ****
 '***********************************************************************************
 
-Sub sw42_Hit:bsLockup.AddBall me : PlaySoundAtVol "popper_ball", sw42, 1: End Sub
+Sub sw42_Hit:bsLockup.AddBall me : playsound "popper_ball": End Sub
 
 '***********************************************************************************
 '****               Knocker                           ****
@@ -600,7 +600,7 @@
 Sub SolKickBack(enabled)
     If enabled Then
        Kickback.Fire
-       PlaySoundAtVol SoundFX("DiverterLeft",DOFContactors), Kickback, .5
+       PlaySound SoundFX("DiverterLeft",DOFContactors), 0, LVL(1), -0.02
     Else
        KickBack.PullBack
     End If
@@ -610,7 +610,7 @@
 '****               Shooter Lane                      ****
 '***********************************************************************************
 
-Sub sw50_Hit():PlaySoundAtVol "Sensor", ActiveBall, 1:Controller.Switch(50) = 1:sw50p.Visible = 0:End Sub
+Sub sw50_Hit():PlaySound "Sensor":Controller.Switch(50) = 1:sw50p.Visible = 0:End Sub
 Sub sw50_UnHit():Controller.Switch(50) = 0:sw50p.Visible = 1:End Sub
 
 dim BallInPlunger :BallInPlunger = False
@@ -810,9 +810,9 @@
 
 sub LeftFlipperSound(updown)'called along with the flipper, so feel free to add stuff, EOStorque tweaks, animation updates, upper flippers, whatever.
   if updown = 1 Then
-    PlaySoundAtVol SoundFX("FlipperUp",DOFFlippers), LeftFlipper, 1
+    playsound SoundFX("FlipperUp",DOFFlippers)  'flip
   Else
-    PlaySoundAtVol SoundFX("FlipperDown",DOFFlippers), LeftFlipper, 1
+    playsound SoundFX("FlipperDown",DOFFlippers)  'return
     LeftFlipper.eostorque = defaultEOS
     LeftFlipper1.eostorque = defaultEOS
       if FlipperReturnMod = 1 then
@@ -829,9 +829,9 @@
 end sub
 sub RightFlipperSound(updown)'called along with the flipper, so feel free to add stuff, EOStorque tweaks, animation updates, upper flippers, whatever.
   if updown = 1 Then
-    PlaySoundAtVol SoundFX("FlipperUp",DOFFlippers), RightFlipper, 1
+    playsound SoundFX("FlipperUp",DOFFlippers)  'flip
   Else
-    PlaySoundAtVol SoundFX("FlipperDown",DOFFlippers), RightFlipper, 1
+    playsound SoundFX("FlipperDown",DOFFlippers)  'return
     RightFlipper.eostorque = defaultEOS
     LeftFlipper1.eostorque = defaultEOS
       if FlipperReturnMod = 1 then
@@ -857,11 +857,11 @@
 Sub SolLFlipper(Enabled)
   if FastFlips = 0 then
     If Enabled Then
-      PlaySoundAtVol SoundFX("FlipperUp",DOFFlippers), LeftFlipper, 1
+      playsound SoundFX("FlipperUp",DOFFlippers), 0, LVL(1), -0.01  'flip
       LeftFlipper.RotateToEnd
       LeftFlipper1.RotateToEnd
     Else
-      PlaySoundAtVol SoundFX("FlipperDown",DOFFlippers), LeftFlipper, 1
+      playsound SoundFX("FlipperDown",DOFFlippers), 0, LVL(1), -0.01  'return
       LeftFlipper.RotateToStart
       LeftFlipper1.RotateToStart
       LeftFlipper.eostorque = defaulteos
@@ -883,10 +883,10 @@
 Sub SolRFlipper(Enabled)
   if FastFlips = 0 then
     If Enabled Then
-      PlaySoundAtVol SoundFX("FlipperUp",DOFFlippers), RightFlipper, 1
+      playsound SoundFX("FlipperUp",DOFFlippers), 0, LVL(1), 0.01 'flip
       RightFlipper.RotateToEnd
     Else
-      PlaySoundAtVol SoundFX("FlipperDown",DOFFlippers), RightFlipper, 1
+      playsound SoundFX("FlipperDown",DOFFlippers), 0, LVL(1), 0.01 'return
       RightFlipper.RotateToStart
       RightFlipper.eostorque = defaulteos
       if FlipperReturnMod = 1 then
@@ -908,7 +908,7 @@
 ' If FlipperEnabled = False Then Exit Sub
   LSling1.Visible = 0:LSling2.Visible = 0:LSling3.Visible = 0: LSling4.Visible = 1: LSling.TransZ = -27
   vpmTimer.PulseSw 57
-  PlaySoundAtVol SoundFX ("left_slingshot",DOFContactors), ActiveBall,1 : DOF dLeftSlingshot, 2
+  PlaySound SoundFX ("left_slingshot",DOFContactors),1:DOF dLeftSlingshot, 2
   SLPos = 0: Me.TimerEnabled = 1
 ' LightshowChangeSide
 End Sub
@@ -926,7 +926,7 @@
 ' If FlipperEnabled = False Then Exit Sub
   RSling1.Visible = 0:RSling2.Visible = 0:RSling3.Visible = 0: RSling4.Visible = 1: RSling.TransZ = -27
   vpmTimer.PulseSw 58
-  PlaySoundAtVol SoundFX ("right_slingshot",DOFContactors), Activeball, 1 : DOF dRightSlingshot, 2
+  PlaySound SoundFX ("right_slingshot",DOFContactors):DOF dRightSlingshot, 2
   SRPos = 0: Me.TimerEnabled = 1
 ' LightshowChangeSide
 End Sub
@@ -1408,184 +1408,53 @@
     Next
 End Sub
 
-' *******************************************************************************************************
-' Positional Sound Playback Functions by DJRobX, Rothbauerw, Thalamus and Herweh
-' PlaySound sound, 0, Vol(ActiveBall), AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-' *******************************************************************************************************
 
-' Play a sound, depending on the X,Y position of the table element (especially cool for surround speaker setups, otherwise stereo panning only)
-' parameters (defaults): loopcount (1), volume (1), randompitch (0), pitch (0), useexisting (0), restart (1))
-' Note that this will not work (currently) for walls/slingshots as these do not feature a simple, single X,Y position
 
-Sub PlayXYSound(soundname, tableobj, loopcount, volume, randompitch, pitch, useexisting, restart)
-  PlaySound soundname, loopcount, volume, AudioPan(tableobj), randompitch, pitch, useexisting, restart, AudioFade(tableobj)
-End Sub
-
-' Set position as table object (Use object or light but NOT wall) and Vol to 1
-
-Sub PlaySoundAt(soundname, tableobj)
-  PlaySound soundname, 1, 1, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-' set position as table object and Vol + RndPitch manually
-
-Sub PlaySoundAtVolPitch(sound, tableobj, Vol, RndPitch)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), RndPitch, 0, 0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed.
-
-Sub PlaySoundAtBall(soundname)
-  PlaySoundAt soundname, ActiveBall
-End Sub
-
-'Set position as table object and Vol manually.
-
-Sub PlaySoundAtVol(sound, tableobj, Volume)
-  PlaySound sound, 1, Volume, AudioPan(tableobj), 0,0,0, 1, AudioFade(tableobj)
-End Sub
-
-'Set all as per ball position & speed, but Vol Multiplier may be used eg; PlaySoundAtBallVol "sound",3
-
-Sub PlaySoundAtBallVol(sound, VolMult)
-  PlaySound sound, 0, Vol(ActiveBall) * VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-Sub PlaySoundAtBallAbsVol(sound, VolMult)
-  PlaySound sound, 0, VolMult, AudioPan(ActiveBall), 0, Pitch(ActiveBall), 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' requires rampbump1 to 7 in Sound Manager
-
-Sub RandomBump(voladj, freq)
-  Dim BumpSnd:BumpSnd= "rampbump" & CStr(Int(Rnd*7)+1)
-  PlaySound BumpSnd, 0, Vol(ActiveBall)*voladj, AudioPan(ActiveBall), 0, freq, 0, 1, AudioFade(ActiveBall)
-End Sub
-
-' set position as bumperX and Vol manually. Allows rapid repetition/overlaying sound
-
-Sub PlaySoundAtBumperVol(sound, tableobj, Vol)
-  PlaySound sound, 1, Vol, AudioPan(tableobj), 0,0,1, 1, AudioFade(tableobj)
-End Sub
-
-Sub PlaySoundAtBOTBallZ(sound, BOT)
-  PlaySound sound, 0, ABS(BOT.velz)/17, Pan(BOT), 0, Pitch(BOT), 1, 0, AudioFade(BOT)
-End Sub
-
-' play a looping sound at a location with volume
-Sub PlayLoopSoundAtVol(sound, tableobj, Vol)
-  PlaySound sound, -1, Vol, AudioPan(tableobj), 0, 0, 1, 0, AudioFade(tableobj)
-End Sub
-
-'*********************************************************************
-'            Supporting Ball, Sound Functions and Math
-'*********************************************************************
+'***********************************************************************************
+'****                   Sound routines                    ****
+'***********************************************************************************
+'****                      Supporting Ball & Sound Functions          ****
+'***********************************************************************************
 
 Function RndNum(min, max)
-  RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
+    RndNum = Int(Rnd() * (max-min + 1) ) + min ' Sets a random number between min and max
 End Function
 
-Const Pi = 3.1415927
-
 Function LVL(input)
   LVL = Input * SoundLevelMult
 End Function
 
-Function dSin(degrees)
-  dsin = sin(degrees * Pi/180)
-End Function
-
-Function dCos(degrees)
-  dcos = cos(degrees * Pi/180)
-End Function
-
-Function AudioFade(tableobj) ' Fades between front and back of the table (for surround systems or 2x2 speakers, etc), depending on the Y position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.y * 2 / table1.height-1
-  If tmp > 0 Then
-    AudioFade = Csng(tmp ^10)
-  Else
-    AudioFade = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function AudioPan(tableobj) ' Calculates the pan for a tableobj based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = tableobj.x * 2 / table1.width-1
-  If tmp > 0 Then
-    AudioPan = Csng(tmp ^10)
-  Else
-    AudioPan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
-Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
-  Dim tmp
-  On Error Resume Next
-  tmp = ball.x * 2 / table1.width-1
-  If tmp > 0 Then
-    Pan = Csng(tmp ^10)
-  Else
-    Pan = Csng(-((- tmp) ^10) )
-  End If
-End Function
-
 Function Vol(ball) ' Calculates the Volume of the sound based on the ball speed
-  Vol = Csng(BallVel(ball) ^2 / 2000)
-End Function
-
-Function VolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  VolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
+    Vol = Csng(BallVel(ball) ^2 / 2000)
 End Function
 
-Function DVolMulti(ball,Multiplier) ' Calculates the Volume of the sound based on the ball speed
-  DVolMulti = Csng(BallVel(ball) ^2 / 150 ) * Multiplier
-  debug.print DVolMulti
+Function Vol2(ball1, ball2) ' Calculates the Volume of the sound based on the speed of two balls
+    Vol2 = (Vol(ball1) + Vol(ball2) ) / 2
 End Function
 
-Function BallRollVol(ball) ' Calculates the Volume of the sound based on the ball speed
-  BallRollVol = Csng(BallVel(ball) ^2 / (80000 - (79900 * Log(RollVol) / Log(100))))
+Function Pan(ball) ' Calculates the pan for a ball based on the X position on the table. "table1" is the name of the table
+    Dim tmp
+    tmp = ball.x * 2 / Table1.width-1
+    If tmp> 0 Then
+        Pan = Csng(tmp ^10)
+    Else
+        Pan = Csng(-((- tmp) ^10) )
+    End If
 End Function
 
 Function Pitch(ball) ' Calculates the pitch of the sound based on the ball speed
-  Pitch = BallVel(ball) * 20
+    Pitch = BallVel(ball) * 20
 End Function
 
 Function BallVel(ball) 'Calculates the ball speed
-  BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
-End Function
-
-Function BallVelZ(ball) 'Calculates the ball speed in the -Z
-  BallVelZ = INT((ball.VelZ) * -1 )
+    BallVel = INT(SQR((ball.VelX ^2) + (ball.VelY ^2) ) )
 End Function
 
-Function VolZ(ball) ' Calculates the Volume of the sound based on the ball speed in the Z
-  VolZ = Csng(BallVelZ(ball) ^2 / 200)*1.2
-End Function
-
-'*** Determines if a Points (px,py) is inside a 4 point polygon A-D in Clockwise/CCW order
-
-Function InRect(px,py,ax,ay,bx,by,cx,cy,dx,dy)
-  Dim AB, BC, CD, DA
-  AB = (bx*py) - (by*px) - (ax*py) + (ay*px) + (ax*by) - (ay*bx)
-  BC = (cx*py) - (cy*px) - (bx*py) + (by*px) + (bx*cy) - (by*cx)
-  CD = (dx*py) - (dy*px) - (cx*py) + (cy*px) + (cx*dy) - (cy*dx)
-  DA = (ax*py) - (ay*px) - (dx*py) + (dy*px) + (dx*ay) - (dy*ax)
-
-  If (AB <= 0 AND BC <=0 AND CD <= 0 AND DA <= 0) Or (AB >= 0 AND BC >=0 AND CD >= 0 AND DA >= 0) Then
-    InRect = True
-  Else
-    InRect = False
-  End If
-End Function
-
-'*****************************************
-'      JP's VP10 Rolling Sounds
-'*****************************************
+'***********************************************************************************
+'****                 JP's VP10 Rolling Sounds                  ****
+'***********************************************************************************
 
-Const tnob = 5 ' total number of balls
+Const tnob = 9 ' total number of balls
 ReDim rolling(tnob)
 InitRolling
 
@@ -1596,7 +1465,9 @@
     Next
 End Sub
 
-Sub RollingTimer_Timer()
+Sub RollingTimer_Timer() : RollingUpdate() : End Sub
+
+Sub RollingUpdate()
     Dim BOT, b
     BOT = GetBalls
 
@@ -1610,53 +1481,45 @@
     If UBound(BOT) = -1 Then Exit Sub
 
   ' play the rolling sound for each ball
-
     For b = 0 to UBound(BOT)
-      If BallVel(BOT(b) ) > 1 Then
-        rolling(b) = True
-        if BOT(b).z < 30 Then ' Ball on playfield
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) ), AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-          ' PlaySound("fx_ballrolling" & b), -1, BallRollVol(BOT(b) )*.8, AudioPan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0, AudioFade(BOT(b) )
-        Else ' Ball on raised ramp
-          PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*.5, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-          ' PlaySound("fx_ballrolling" & b), -1, BallRollVol(BOT(b) )*.2, AudioPan(BOT(b) ), 0, Pitch(BOT(b) )+50000, 1, 0, AudioFade(BOT(b) )
-        End If
-      Else
-        If rolling(b) = True Then
-          StopSound("fx_ballrolling" & b)
-          rolling(b) = False
+        If BallVel(BOT(b) ) > 1 AND BOT(b).z < 30 Then
+            rolling(b) = True
+            PlaySound("fx_ballrolling" & b), -1, Vol(BOT(b) )*5, Pan(BOT(b) ), 0, Pitch(BOT(b) ), 1, 0
+        Else
+            If rolling(b) = True Then
+                StopSound("fx_ballrolling" & b)
+                rolling(b) = False
+            End If
         End If
-      End If
     Next
 End Sub
 
-'**********************
-' Ball Collision Sound
-'**********************
+'***********************************************************************************
+'****                 Ball Collision Sound                  ****
+'***********************************************************************************
 
 Sub OnBallBallCollision(ball1, ball2, velocity)
-    PlaySound("fx_collide"), 0, Csng(velocity) ^2 / 2000, AudioPan(ball1), 0, Pitch(ball1), 0, 0, AudioFade(ball1)
+  PlaySound("fx_collide"), 0, LVL(Csng(velocity) ^2 / 2000), Pan(ball1), 0, Pitch(ball1), 0, 0
 End Sub
 
-
 Sub zApron_Hit (idx)
-  PlaySound "woodhitaluminium", 0, LVL((Vol(ActiveBall)^2.5)*10), Pan(ActiveBall)/4, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "woodhitaluminium", 0, LVL((Vol(ActiveBall)^2.5)*10), Pan(ActiveBall)/4, 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub zInlanes_Hit (idx)
-  PlaySound "MetalHit2", 0, LVL(Vol(ActiveBall)*5), Pan(ActiveBall)*55, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "MetalHit2", 0, LVL(Vol(ActiveBall)*5), Pan(ActiveBall)*55, 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub RampEntry_Hit()
   If activeball.vely < -10 then
-    PlaySound "ramp_hit2", 0, LVL(Vol(ActiveBall)/5), Pan(ActiveBall), 0, Pitch(ActiveBall)*10, 1, 0, AudioFade(ActiveBall)
+    PlaySound "ramp_hit2", 0, LVL(Vol(ActiveBall)/5), Pan(ActiveBall), 0, Pitch(ActiveBall)*10, 1, 0
   Elseif activeball.vely > 3 then
-    PlaySound "PlayfieldHit", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "PlayfieldHit", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End If
 End Sub
 
 Sub RampHit2_Hit()
-  PlaySound "ramp_hit3", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "ramp_hit3", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub RampEntry2_Hit()
@@ -1665,23 +1528,23 @@
 
 Sub RampSound0_Hit()
   If Activeball.velx > 0 then
-    PlaySoundAtVol "pmaReriW", ActiveBall, 1
+    Playsound "pmaReriW", 0, LVL(1)
   Else
-    PlaySoundAtVol "WireRamp1", ActiveBall, 1
+    Playsound "WireRamp1", 0, LVL(1)
   End If
 End Sub
 
 Sub Metals_Hit (idx)
-  PlaySound "metalhit_medium", 0, LVL(Vol(ActiveBall)*30 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "metalhit_medium", 0, LVL(Vol(ActiveBall)*30 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
 End Sub
 
 Sub Targets_Hit (idx)
-  PlaySound SoundFX("target",DOFTargets), 0, LVL(0.2), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
-  PlaySound SoundFX("targethit",0), 0, LVL(Vol(ActiveBall)*18 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0, AudioFade(ActiveBall)
+  PlaySound SoundFX("target",DOFTargets), 0, LVL(0.2), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0
+  PlaySound SoundFX("targethit",0), 0, LVL(Vol(ActiveBall)*18 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 0, 0
 End Sub
 
 Sub Gates_Hit (idx)
-  PlaySound "gate4", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+  PlaySound "gate4", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   FlashForMs yBoxFlasher, 1000, 40, 0
 End Sub
 
@@ -1689,13 +1552,13 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 15 then
-    PlaySound "fx_rubber2", 0, LVL(Vol(ActiveBall)*30 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, LVL(Vol(ActiveBall)*30 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
 
   If finalspeed >= 6 AND finalspeed <= 15 then
     RandomSoundRubber()
   else
-    PlaySound "rubber_hit_3", 0, LVL(Vol(ActiveBall)*30 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "rubber_hit_3", 0, LVL(Vol(ActiveBall)*30 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0
   End If
 End Sub
 
@@ -1703,20 +1566,20 @@
   dim finalspeed
     finalspeed=SQR(activeball.velx * activeball.velx + activeball.vely * activeball.vely)
   If finalspeed > 16 then
-    PlaySound "fx_rubber2", 0, LVL(Vol(ActiveBall)*80 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "fx_rubber2", 0, LVL(Vol(ActiveBall)*80 ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End if
   If finalspeed >= 6 AND finalspeed <= 15 then
     RandomSoundRubber()
   else
-    PlaySound "rubber_hit_3", 0, LVL(Vol(ActiveBall)*1 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    PlaySound "rubber_hit_3", 0, LVL(Vol(ActiveBall)*1 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0
   End If
 End Sub
 
 Sub RandomSoundRubber()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "rubber_hit_1", 0, LVL(Vol(ActiveBall)*50 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "rubber_hit_2", 0, LVL(Vol(ActiveBall)*50 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "rubber_hit_3", 0, LVL(Vol(ActiveBall)*50 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "rubber_hit_1", 0, LVL(Vol(ActiveBall)*50 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "rubber_hit_2", 0, LVL(Vol(ActiveBall)*50 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "rubber_hit_3", 0, LVL(Vol(ActiveBall)*50 ), Pan(ActiveBall)*50, 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
@@ -1730,9 +1593,9 @@
 
 Sub RandomSoundFlipper()
   Select Case Int(Rnd*3)+1
-    Case 1 : PlaySound "flip_hit_1", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 2 : PlaySound "flip_hit_2", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
-    Case 3 : PlaySound "flip_hit_3", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0, AudioFade(ActiveBall)
+    Case 1 : PlaySound "flip_hit_1", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 2 : PlaySound "flip_hit_2", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
+    Case 3 : PlaySound "flip_hit_3", 0, LVL(Vol(ActiveBall) ), Pan(ActiveBall), 0, Pitch(ActiveBall), 1, 0
   End Select
 End Sub
 
@@ -1753,8 +1616,4 @@
 const dLeftSlingshot        = 211 ' Left Slingshot
 const dRightSlingshot         = 212 ' Right Slingshot
 
-Sub Table1_Exit()
-  Controller.Games(cGameName).Settings.Value("sound") = 1
-  Controller.Stop
-End Sub
-
+Sub Table1_Exit():Controller.Games(cGameName).Settings.Value("sound") = 1:Controller.Stop:End Sub
